<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=analysisDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=analysisArray;
            var docs_map=analysisArrayMap;
            var pos=analysisArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Pleiotropy analysis in UKBB</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Pleiotropy analysis in UKBB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../workflow.html">Workflow</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/dianacornejo/pleiotropy_UKB"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Analysis-of-hearing-impairment-phenotypes">Analysis of hearing impairment phenotypes<a class="anchor-link" href="#Analysis-of-hearing-impairment-phenotypes">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>This notebook applies the <code>Get_Job_Script.ipynb</code> to automatically generate the sbatch scripts to run in Yale's cluster. The end result is to apply <a href="https://github.com/statgenetics/UKBB_GWAS_dev/tree/master/workflow">various LMM workflows</a> to perform association analysis in different hearing impairment traits, do clumping analysis and extract associated regions.</p>
<p>The phenotypes analyzed are:</p>
<ol>
<li>Hearing aid f.3393</li>
<li>Hearing difficulty f.2247</li>
<li>Hearing difficulty with background noise f.2257</li>
<li>Combined phenotype f.2247 &amp; f.2257</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="File-paths-on-Yale-cluster">File paths on Yale cluster<a class="anchor-link" href="#File-paths-on-Yale-cluster">&#182;</a></h2><ul>
<li>Genotype files in PLINK format:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/pleiotropy_geneticfiles/UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv</code></li>
<li>Genotype files in bgen format:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/ukb39554_imputeddataset/</code></li>
<li>Summary stats for imputed variants BOLT-LMM:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/results/BOLTLMM_results/results_imputed_data</code></li>
<li>Summary stats for inputed variants FastGWA:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/results/FastGWA_results/results_imputed_data</code></li>
<li>Phenotype files:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/phenotype_files/hearing_impairment</code></li>
<li>Relationship file:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/pleiotropy_geneticfiles/unrelated_n307259/UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620</code></li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="08/31/20-analysis">08/31/20 analysis<a class="anchor-link" href="#08/31/20-analysis">&#182;</a></h2><p>On the cluster, open up this notebook using the JupyterLab server you set up via the ssh channel, then run the following cells,</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Bash-variables-for-workflow-configuration">Bash variables for workflow configuration<a class="anchor-link" href="#Bash-variables-for-workflow-configuration">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Common variables</span>
<span class="n">UKBB_PATH</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span>
<span class="n">USER_PATH</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">project</span>
<span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">bfile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_Caucasians_phenotypeindepqc120319_updated082020removedwithdrawnindiv</span><span class="o">.</span><span class="n">bed</span>
<span class="n">unrelated_samples</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">unrelated_n307259</span><span class="o">/</span><span class="n">UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620</span>
<span class="n">formatFile_fastgwa</span><span class="o">=</span><span class="err">$</span><span class="n">USER_PATH</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">fastGWA_template</span><span class="o">.</span><span class="n">yml</span>
<span class="n">formatFile_bolt</span><span class="o">=</span><span class="err">$</span><span class="n">USER_PATH</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">boltlmm_template</span><span class="o">.</span><span class="n">yml</span>
<span class="n">formatFile_saige</span><span class="o">=</span><span class="err">$</span><span class="n">USER_PATH</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">saige_template</span><span class="o">.</span><span class="n">yml</span>
<span class="n">formatFile_regenie</span><span class="o">=</span><span class="err">$</span><span class="n">USER_PATH</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">regenie_template</span><span class="o">.</span><span class="n">yml</span>
<span class="n">lmm_sos</span><span class="o">=</span><span class="err">$</span><span class="n">USER_PATH</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">LMM</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">hearing_pheno_path</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span>

<span class="c1"># Container</span>
<span class="n">container_lmm</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="n">container_marp</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">marp</span><span class="o">.</span><span class="n">sif</span>

<span class="c1"># Imputed data BGEN input</span>
<span class="c1">##genoFile=`echo $UKBB_PATH/genotype_files/ukb39554_imputeddataset/ukb_imp_chr{1..22}_v3.bgen`</span>
<span class="c1">##sampleFile=$UKBB_PATH/genotype_files/ukb39554_imputeddataset/ukb32285_imputedindiv.sample</span>
<span class="c1"># LMM directories</span>
<span class="n">lmm_imp_dir_fastgwa</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span>
<span class="n">lmm_imp_dir_bolt</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span>
<span class="n">lmm_imp_dir_saige</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">SAIGE_results</span><span class="o">/</span><span class="n">results_imputed_data</span>
<span class="n">lmm_imp_dir_regenie</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_imputed_data</span>

<span class="c1"># Exome data PLINK input</span>
<span class="n">genoFile</span><span class="o">=</span><span class="sb">`echo $UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_c{1..22}_b0_v1.bed`</span>
<span class="c1"># LMM directories</span>
<span class="n">lmm_exome_dir_fastgwa</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_exome_data</span>
<span class="n">lmm_exome_dir_bolt</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_exome_data</span>
<span class="n">lmm_exome_dir_saige</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">SAIGE_results</span><span class="o">/</span><span class="n">results_exome_data</span>
<span class="n">lmm_exome_dir_regenie</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span>

<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_lmm</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">lmm_sbatch_bolt</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_lmm</span><span class="o">-</span><span class="n">bolt</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">lmm_sbatch_saige</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_lmm</span><span class="o">-</span><span class="n">saige</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">lmm_sbatch_regenie</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_lmm</span><span class="o">-</span><span class="n">regenie</span><span class="o">.</span><span class="n">sbatch</span>

<span class="c1">## LMM variables </span>
<span class="n">LDscoresFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">LDSCORE</span><span class="o">.</span><span class="mi">1000</span><span class="n">G_EUR</span><span class="o">.</span><span class="n">tab</span><span class="o">.</span><span class="n">gz</span>
<span class="n">geneticMapFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genetic_map_hg19_withX</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span>
<span class="n">covarMaxLevels</span><span class="o">=</span><span class="mi">10</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">20</span>
<span class="n">bgenMinMAF</span><span class="o">=</span><span class="mf">0.001</span>
<span class="n">bgenMinINFO</span><span class="o">=</span><span class="mf">0.8</span>
<span class="n">lmm_job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">ylim</span><span class="o">=</span><span class="mi">0</span>
<span class="c1">### Specific to FastGWA</span>
<span class="n">grmFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv</span><span class="o">.</span><span class="n">grm</span><span class="o">.</span><span class="n">sp</span>
<span class="c1">### Specific to SAIGE</span>
<span class="n">bgenMinMAC</span><span class="o">=</span><span class="mi">4</span>
<span class="n">trait_type</span><span class="o">=</span><span class="n">binary</span>
<span class="n">loco</span><span class="o">=</span><span class="n">TRUE</span>
<span class="n">sampleCol</span><span class="o">=</span><span class="n">IID</span>
<span class="c1">### Specific to REGENIE</span>
<span class="n">bsize</span><span class="o">=</span><span class="mi">1000</span>
<span class="n">lowmem</span><span class="o">=~/</span><span class="n">scratch60</span><span class="o">/</span>
<span class="n">lowmem_prefix</span><span class="o">=~/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">predictions</span>
<span class="n">trait</span><span class="o">=</span><span class="n">bt</span>
<span class="n">minMAC</span><span class="o">=</span><span class="mi">4</span>
<span class="n">maf_filter</span><span class="o">=</span><span class="mf">0.01</span>
<span class="n">geno_filter</span><span class="o">=</span><span class="mf">0.01</span>
<span class="n">hwe_filter</span><span class="o">=</span><span class="mi">0</span>
<span class="n">mind_filter</span><span class="o">=</span><span class="mf">0.1</span>
<span class="n">reverse_log_p</span><span class="o">=</span><span class="bp">True</span>
<span class="c1"># LD clumping directories</span>
<span class="n">clumping_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span>
<span class="n">clumping_sos</span><span class="o">=</span><span class="err">$</span><span class="n">USER_PATH</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">LD_Clumping</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">clumping_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_ldclumping</span><span class="o">.</span><span class="n">sbatch</span>
<span class="c1">## LD clumping variables</span>
<span class="c1"># For sumtastsFiles if more than one provide each path</span>
<span class="n">bfile_ref</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span><span class="n">UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv</span><span class="o">.</span><span class="mf">1210.</span><span class="n">ref_geno</span><span class="o">.</span><span class="n">bed</span>
<span class="c1"># Changes dependending upon which traits are analyzed</span>
<span class="c1"># In this case tinnitus only</span>
<span class="n">sumstatsFiles</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span>
<span class="n">ld_sample_size</span><span class="o">=</span><span class="mi">1210</span>
<span class="n">clump_field</span><span class="o">=</span><span class="n">P</span>
<span class="n">clump_p1</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">clump_p2</span><span class="o">=</span><span class="mi">1</span>
<span class="n">clump_r2</span><span class="o">=</span><span class="mf">0.2</span>
<span class="n">clump_kb</span><span class="o">=</span><span class="mi">2000</span>
<span class="n">clump_annotate</span><span class="o">=</span><span class="n">BP</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">20</span>
<span class="n">clump_job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">clumpFile</span><span class="o">=</span> 
<span class="n">clumregionFile</span><span class="o">=</span>
<span class="c1"># Region extraction directories</span>
<span class="n">extract_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">region_extraction</span>
<span class="n">extract_sos</span><span class="o">=</span><span class="err">$</span><span class="n">USER_PATH</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Region_Extraction</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">extract_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_tinnitus</span><span class="o">-</span><span class="n">region</span><span class="o">.</span><span class="n">sbatch</span>
<span class="c1">## Region extraction variables</span>
<span class="n">region_file</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span>
<span class="n">geno_path</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">UKBB_bgenfilepath</span><span class="o">.</span><span class="n">txt</span>
<span class="n">sumstats_path</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span>
<span class="n">extract_job_size</span><span class="o">=</span><span class="mi">10</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="1.-Hearing-aid-user-f.3393">1. Hearing aid user f.3393<a class="anchor-link" href="#1.-Hearing-aid-user-f.3393">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="FastGWA-job-only-white-British">FastGWA job only white British<a class="anchor-link" href="#FastGWA-job-only-white-British">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=</span><span class="err">$</span><span class="n">lmm_imp_dir_fastgwa</span><span class="o">/</span><span class="n">f3393_hearing_aid</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_imp</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_aid_f3393</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_aid_f3393</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">hearing_aid_cat</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age_final_aid</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="FastGWA-all-whites">FastGWA all whites<a class="anchor-link" href="#FastGWA-all-whites">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=</span><span class="err">$</span><span class="n">lmm_imp_dir_fastgwa</span><span class="o">//</span><span class="n">f3393_hearing_aid_expandedwhite</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_expandedwhite_imp</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="mi">120120</span><span class="n">_UKBB_Hearing_aid_f3393_expandedwhite</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="mi">120120</span><span class="n">_UKBB_Hearing_aid_f3393_expandedwhite</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">hearing_aid_cat</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age_final_aid</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-12-02_f3393_expandedwhite_imp-fastgwa.sbatch</span>
INFO: Workflow farnam (ID=852e6b72805bc1ab) is executed successfully with 1 completed step.
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="FastGWA-exome-data">FastGWA exome data<a class="anchor-link" href="#FastGWA-exome-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=</span><span class="err">$</span><span class="n">lmm_exome_dir_fastgwa</span><span class="o">/</span><span class="n">f3393_hearing_aid_exomes</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_exomes</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="n">phenotypes_exome_data</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_aid_f3393_128254ind_exomes</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="n">phenotypes_exome_data</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_aid_f3393_128254ind_exomes</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">hearing_aid_cat</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age_final_aid</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --genoFile $genoFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-01-11_f3393_exomes-fastgwa.sbatch</span>
INFO: Workflow farnam (ID=3f949028f44f4152) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Regenie-exome-data">Regenie exome data<a class="anchor-link" href="#Regenie-exome-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[4]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_regenie</span><span class="o">=</span><span class="err">$</span><span class="n">lmm_exome_dir_regenie</span><span class="o">/</span><span class="n">f3393_hearing_aid_exomes_bfile</span>
<span class="n">lmm_sbatch_regenie</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_hearing_aid_exome_bfile</span><span class="o">-</span><span class="n">regenie</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="n">phenotypes_exome_data</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_aid_f3393_128254ind_exomes</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="n">phenotypes_exome_data</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_aid_f3393_128254ind_exomes</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">hearing_aid_cat</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age_final_aid</span>
<span class="c1">#Use original bed files from the UKBB exome data</span>
<span class="c1">#bfile=$UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/exome_files_snpsonly/ukb23155.filtered.merged.bed</span>
<span class="c1">#Use the original bed files for the genotype array on regenie step1</span>
<span class="n">bfile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_originalgenotypefilesdownloaded083019</span><span class="o">/</span><span class="n">UKB_genotypedatadownloaded083019</span><span class="o">.</span><span class="n">bed</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;regenie</span>
<span class="s2">    --cwd $lmm_dir_regenie </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --genoFile $genoFile</span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_regenie </span>
<span class="s2">    --phenoCol $phenoCol</span>
<span class="s2">    --covarCol $covarCol  </span>
<span class="s2">    --qCovarCol $qCovarCol</span>
<span class="s2">    --bsize $bsize</span>
<span class="s2">    --lowmem_prefix $lowmem_prefix</span>
<span class="s2">    --trait $trait </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO</span>
<span class="s2">    --maf_filter $maf_filter</span>
<span class="s2">    --geno_filter $geno_filter</span>
<span class="s2">    --hwe_filter $hwe_filter</span>
<span class="s2">    --mind_filter $mind_filter</span>
<span class="s2">    --minMAC $minMAC</span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --reverse_log_p $reverse_log_p</span>
<span class="s2">    --numThreads $numThreads</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_regenie</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-12_f3393_hearing_aid_exome_bfile-regenie.sbatch</span>
INFO: Workflow farnam (ID=w03430bc454ddd1e8) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Bolt-LMM-job">Bolt-LMM job<a class="anchor-link" href="#Bolt-LMM-job">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_bolt</span><span class="o">=</span><span class="err">$</span><span class="n">lmm_imp_dir_bolt</span><span class="o">/</span><span class="n">f3393_hearing_aid</span>
<span class="n">lmm_sbatch_bolt</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_hearing_aid_imp</span><span class="o">-</span><span class="n">bolt</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_aid_f3393</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_aid_f3393</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">hearing_aid_cat</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age_final_aid</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;boltlmm</span>
<span class="s2">    --cwd $lmm_dir_bolt </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_bolt </span>
<span class="s2">    --covarFile $covarFile </span>
<span class="s2">    --LDscoresFile $LDscoresFile </span>
<span class="s2">    --geneticMapFile $geneticMapFile </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp    </span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_bolt</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-10-20_f3393_hearing_aid_imp-bolt.sbatch</span>
INFO: Workflow farnam (ID=c482417dfcba9f23) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="LD-clumping-job">LD clumping job<a class="anchor-link" href="#LD-clumping-job">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Imputed-data">Imputed data<a class="anchor-link" href="#Imputed-data">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">clumping_dir</span><span class="o">=</span><span class="err">$</span><span class="n">clumping_dir</span><span class="o">/</span><span class="n">f3393_hearing_aid</span>
<span class="n">clumping_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_hearing_aid_ldclumping</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstatsFiles</span><span class="o">=</span><span class="err">$</span><span class="n">lmm_imp_dir_fastgwa</span><span class="o">/</span><span class="n">f3393_hearing_aid</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_aid_f3393_hearing_aid_cat</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>

<span class="n">clumping_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default </span>
<span class="s2">    --cwd $clumping_dir </span>
<span class="s2">    --bfile $bfile</span>
<span class="s2">    --bfile_ref $bfile_ref </span>
<span class="s2">    --genoFile $genofile</span>
<span class="s2">    --sampleFile $sampleFile </span>
<span class="s2">    --sumstatsFiles $sumstatsFiles </span>
<span class="s2">    --unrelated_samples $unrelated_samples </span>
<span class="s2">    --ld_sample_size $ld_sample_size </span>
<span class="s2">    --clump_field $clump_field</span>
<span class="s2">    --clump_p1 $clump_p1 </span>
<span class="s2">    --clump_p2 $clump_p2 </span>
<span class="s2">    --clump_r2 $clump_r2 </span>
<span class="s2">    --clump_kb $clump_kb </span>
<span class="s2">    --clump_annotate $clump_annotate </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --job_size $clump_job_size</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">clumping_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">clumping_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$clumping_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Exome-data:">Exome data:<a class="anchor-link" href="#Exome-data:">&#182;</a></h3><p>FIXME:</p>
<ol>
<li>Option: Is to create the reference bed from the exome data. However this bfile is used to calculate the LD between SNPs, therefore it is better to use a reference file created from the genotype array. One drawback will be that variants present in the exome and absent in the genotype array wont be selected as index SNPs.</li>
<li>Option: use the bfile_ref created from the imputed data </li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">clumping_dir</span><span class="o">=</span><span class="err">$</span><span class="n">clumping_dir</span><span class="o">/</span><span class="n">f3393_hearing_aid_exome</span>
<span class="n">clumping_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_hearing_aid_exome_ldclumping</span><span class="o">.</span><span class="n">sbatch</span>
<span class="c1">#clumping_sbatch=../output/$(date +&quot;%Y-%m-%d&quot;)_refbedfile2_exome_ldclumping.sbatch</span>
<span class="n">bfile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_Caucasians_phenotypeindepqc120319_updated082020removedwithdrawnindiv</span><span class="o">.</span><span class="n">bed</span>
<span class="n">sumstatsFiles</span><span class="o">=</span><span class="err">$</span><span class="n">lmm_exome_dir_regenie</span><span class="o">/</span><span class="n">f3393_hearing_aid_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_aid_f3393_128254ind_exomes_hearing_aid_cat</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">sampleFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">fam</span>
<span class="n">bfile_ref</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_s200631_chr1_22_exomedata</span><span class="o">.</span><span class="mf">1200.</span><span class="n">ref_geno</span><span class="o">.</span><span class="n">bed</span>
<span class="n">ld_sample_size</span><span class="o">=</span><span class="mi">1200</span>
<span class="n">clump_field</span><span class="o">=</span><span class="n">P</span>
<span class="n">clump_p1</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">clump_p2</span><span class="o">=</span><span class="mi">1</span>
<span class="n">clump_r2</span><span class="o">=</span><span class="mf">0.2</span>
<span class="n">clump_kb</span><span class="o">=</span><span class="mi">2000</span>
<span class="n">clump_annotate</span><span class="o">=</span><span class="n">BP</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">20</span>

<span class="c1"># Select samples filter_samples workflow &amp; create reference file with reference workflow</span>
<span class="c1"># Then use default workflow to run the LD clumping</span>
<span class="n">clumping_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default</span>
<span class="s2">    --cwd $clumping_dir</span>
<span class="s2">    --bfile $bfile</span>
<span class="s2">    --bfile_ref $bfile_ref </span>
<span class="s2">    --genoFile $genoFile</span>
<span class="s2">    --sampleFile $sampleFile </span>
<span class="s2">    --sumstatsFiles $sumstatsFiles </span>
<span class="s2">    --unrelated_samples $unrelated_samples </span>
<span class="s2">    --ld_sample_size $ld_sample_size </span>
<span class="s2">    --clump_field $clump_field</span>
<span class="s2">    --clump_p1 $clump_p1 </span>
<span class="s2">    --clump_p2 $clump_p2 </span>
<span class="s2">    --clump_r2 $clump_r2 </span>
<span class="s2">    --clump_kb $clump_kb </span>
<span class="s2">    --clump_annotate $clump_annotate </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --job_size $clump_job_size</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">clumping_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">clumping_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$clumping_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-07_f3393_hearing_aid_exome_ldclumping.sbatch</span>
INFO: Workflow farnam (ID=w64d7831c0c82bb7e) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Region-extraction">Region extraction<a class="anchor-link" href="#Region-extraction">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Imputed-data">Imputed data<a class="anchor-link" href="#Imputed-data">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[4]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_aid_f3393</span>
<span class="n">extract_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">region_extraction</span><span class="o">/</span><span class="n">f3393_hearing_aid</span>
<span class="n">extract_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">workflow</span><span class="o">/</span><span class="n">Region_Extraction_4</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">extract_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_hearing_aid_imp</span><span class="o">-</span><span class="n">region</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">region_file</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span><span class="n">f3393_hearing_aid</span><span class="o">/*.</span><span class="n">clumped_region</span>
<span class="n">geno_path</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">UKBB_bgenfilepath</span><span class="o">.</span><span class="n">txt</span>
<span class="n">sumstats_path</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f3393_hearing_aid</span><span class="o">/*.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>

<span class="n">extract_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default</span>
<span class="s2">    --cwd $extract_dir</span>
<span class="s2">    --region-file $region_file</span>
<span class="s2">    --pheno-path $phenoFile</span>
<span class="s2">    --geno-path $geno_path</span>
<span class="s2">    --bgen-sample-path $sampleFile</span>
<span class="s2">    --sumstats-path $sumstats_path</span>
<span class="s2">    --format-config-path $formatFile_fastgwa</span>
<span class="s2">    --unrelated-samples $unrelated_samples</span>
<span class="s2">    --job-size $extract_job_size</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">extract_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">extract_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$extract_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-10-28_f3393_hearing_aid_imp-region.sbatch</span>
INFO: Workflow farnam (ID=55af58288aa5cc61) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Exome-data">Exome data<a class="anchor-link" href="#Exome-data">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[15]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1">### Create the bedfilepath file</span>
<span class="n">cd</span> <span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span>
<span class="k">for</span> <span class="nb">file</span> <span class="ow">in</span> <span class="n">ukb23155_c</span><span class="p">{</span><span class="mf">1.</span><span class="o">.</span><span class="mi">22</span><span class="p">}</span><span class="n">_b0_v1</span><span class="o">.</span><span class="n">bed</span><span class="p">;</span>
    <span class="n">do</span> <span class="n">echo</span> <span class="sb">`pwd`</span><span class="o">/</span><span class="err">$</span><span class="nb">file</span><span class="p">;</span>
<span class="n">done</span> <span class="o">|</span> <span class="n">awk</span> <span class="s1">&#39;{print NR &quot; &quot; $s}&#39;</span> <span class="o">&gt;</span> <span class="n">UKBB_exome_plinkfilepath</span><span class="o">.</span><span class="n">txt</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="n">phenotypes_exome_data</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_aid_f3393_128254ind_exomes</span>
<span class="n">extract_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">region_extraction</span><span class="o">/</span><span class="n">f3393_hearing_aid_exomes</span>
<span class="c1"># Original region extraction pipeline</span>
<span class="n">extract_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Region_Extraction</span><span class="o">.</span><span class="n">ipynb</span> 
<span class="c1">#extract_sos=~/project/UKBB_GWAS_dev/workflow/Region_Extraction_4.ipynb</span>
<span class="n">extract_sbatch</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_hearing_aid_exome</span><span class="o">-</span><span class="n">regionextrac</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">region_file</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span><span class="n">f3393_hearing_aid_exome</span><span class="o">/*.</span><span class="n">clumped_region</span>
<span class="n">geno_path</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">UKBB_exome_plinkfilepath</span><span class="o">.</span><span class="n">txt</span>
<span class="n">sumstats_path</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f3393_hearing_aid_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_aid_f3393_128254ind_exomes_hearing_aid_cat</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">unrelated_samples</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">unrelated_n307259</span><span class="o">/</span><span class="n">UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620</span>
<span class="n">extract_job_size</span><span class="o">=</span><span class="mi">10</span>
<span class="n">sampleFile</span><span class="o">=</span>
<span class="n">formatFile_regenie</span><span class="o">=</span>
<span class="c1">#container_lmm=$UKBB_PATH/lmm.sif</span>

<span class="n">extract_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default</span>
<span class="s2">    --cwd $extract_dir</span>
<span class="s2">    --region-file $region_file</span>
<span class="s2">    --pheno-path $phenoFile</span>
<span class="s2">    --geno-path $geno_path</span>
<span class="s2">    --bgen-sample-path $sampleFile</span>
<span class="s2">    --sumstats-path $sumstats_path</span>
<span class="s2">    --format-config-path $formatFile_regenie</span>
<span class="s2">    --unrelated-samples $unrelated_samples</span>
<span class="s2">    --job-size $extract_job_size</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">extract_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">extract_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$extract_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">/home/dc2325/project/UKBB_GWAS_dev/output/2021-02-09_f3393_hearing_aid_exome-regionextrac.sbatch</span>
INFO: Workflow farnam (ID=w60dbbcede9750add) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Fine-mapping">Fine mapping<a class="anchor-link" href="#Fine-mapping">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">finemap_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">fine_mapping</span><span class="o">/</span><span class="n">f3393_hearing_aid</span>
<span class="n">finemap_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">workflow</span><span class="o">/</span><span class="n">SuSiE_test</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">finemap_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_hearing_aid_imp</span><span class="o">-</span><span class="n">finemap</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">region_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">region_extraction</span><span class="o">/</span><span class="n">f3393_hearing_aid</span>
<span class="n">region_file</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span><span class="n">f3393_hearing_aid</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_aid_f3393_hearing_aid_cat</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">clumped_region</span>
<span class="n">sumstats_path</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f3393_hearing_aid</span><span class="o">/*.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">N</span><span class="o">=</span><span class="mi">230411</span>
<span class="n">container_lmm</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">lmm_v_1_4</span><span class="o">.</span><span class="n">sif</span>

<span class="n">finemap_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default</span>
<span class="s2">    --cwd $finemap_dir</span>
<span class="s2">    --region_dir $region_dir</span>
<span class="s2">    --region_file $phenoFile</span>
<span class="s2">    --sumstats_path $sumstats_path</span>
<span class="s2">    --container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">finemap_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">finemap_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$finemap_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Post_GWAS-annotation-SNP-to-gene">Post_GWAS annotation SNP-to-gene<a class="anchor-link" href="#Post_GWAS-annotation-SNP-to-gene">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f3393_hearing_aid_exomes</span>
<span class="n">postgwa_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_postgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstatsFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f3393_hearing_aid_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_aid_f3393_128254ind_exomes_hearing_aid_cat</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">postgwa_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">workflow</span><span class="o">/</span><span class="n">snptogene</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">hg</span><span class="o">=</span><span class="mi">38</span>
<span class="n">postgwa_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default</span>
<span class="s2">    --cwd $lmm_dir</span>
<span class="s2">    --sumstatsFile $sumstatsFile</span>
<span class="s2">    --hg $hg</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">postgwa_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">postgwa_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$postgwa_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-01-28_f3393_postgwa.sbatch</span>
INFO: Workflow farnam (ID=w6b10aab5fa809918) is executed successfully with 1 completed step.
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">UKBB_PATH</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span>
<span class="n">ukbb</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span>
<span class="n">USER_PATH</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">project</span>
<span class="n">cwd</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">output</span><span class="o">/</span><span class="n">bfile_annovar</span>
<span class="n">sumstatsFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f3393_hearing_aid_exomes_bfile</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_aid_f3393_128254ind_exomes_hearing_aid_cat</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">hg</span><span class="o">=</span><span class="mi">38</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_annovar</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">annovar</span><span class="o">.</span><span class="n">sif</span>
<span class="n">bimfiles</span><span class="o">=</span><span class="sb">`echo /gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_c{1..22}_b0_v1.bim`</span>
<span class="n">bim_name</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">output</span><span class="o">/</span><span class="n">ukb23155_chr1_chr22</span><span class="o">.</span><span class="n">bim</span>
<span class="n">humandb</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">ysm</span><span class="o">/</span><span class="n">datasets</span><span class="o">/</span><span class="n">db</span><span class="o">/</span><span class="n">annovar</span><span class="o">/</span><span class="n">humandb</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">workflow</span><span class="o">/</span><span class="n">snptogene</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">annovar</span> \
    <span class="o">--</span><span class="n">cwd</span> <span class="err">$</span><span class="n">cwd</span> \
    <span class="o">--</span><span class="n">sumstatsFile</span> <span class="err">$</span><span class="n">sumstatsFile</span>\
    <span class="o">--</span><span class="n">bim_name</span> <span class="err">$</span><span class="n">bim_name</span> \
    <span class="o">--</span><span class="n">hg</span> <span class="err">$</span><span class="n">hg</span> \
    <span class="o">--</span><span class="n">job_size</span> <span class="err">$</span><span class="n">job_size</span> \
    <span class="o">--</span><span class="n">humandb</span> <span class="err">$</span><span class="n">humandb</span>\
    <span class="o">--</span><span class="n">ukbb</span> <span class="err">$</span><span class="n">ukbb</span> \
    <span class="o">--</span><span class="n">container_annovar</span> <span class="err">$</span><span class="n">container_annovar</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">annovar_1</span>: Get the list of significantly associated SNPs
INFO: <span class="ansi-green-fg">annovar_1</span> (index=0) is <span class="ansi-green-fg">ignored</span> due to saved signature
INFO: <span class="ansi-green-fg">annovar_1</span> output:   <span class="ansi-green-fg">/home/dc2325/scratch60/output/bfile_annovar/010421_UKBB_Hearing_aid_f3393_128254ind_exomes_hearing_aid_cat.regenie.snp_annotate</span>
INFO: Running <span class="ansi-green-fg">annovar_2</span>: Get chr, start, end, ref_allele, alt_allele format
INFO: <span class="ansi-green-fg">annovar_2</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">annovar_2</span> output:   <span class="ansi-green-fg">/home/dc2325/scratch60/output/bfile_annovar/010421_UKBB_Hearing_aid_f3393_128254ind_exomes_hearing_aid_cat.regenie.avinput</span>
INFO: Running <span class="ansi-green-fg">annovar_3</span>: Annotate variants file using ANNOVAR
INFO: <span class="ansi-green-fg">annovar_3</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">annovar_3</span> output:   <span class="ansi-green-fg">/home/dc2325/scratch60/output/bfile_annovar/010421_UKBB_Hearing_aid_f3393_128254ind_exomes_hearing_aid_cat.regenie.hg38_multianno.csv</span>
INFO: Workflow annovar (ID=wbd6286248f401819) is executed successfully with 2 completed steps and 1 ignored step.
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Post-GWAS-annotation-ANNOVAR">Post-GWAS annotation ANNOVAR<a class="anchor-link" href="#Post-GWAS-annotation-ANNOVAR">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">annovar_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">annovar_exome</span>
<span class="n">bedfiles</span><span class="o">=</span><span class="sb">`echo $UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_c{1..22}_b0_v1.bed`</span>
<span class="n">bimfiles</span><span class="o">=</span><span class="sb">`echo $UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_c{1..22}_b0_v1.bim`</span>
<span class="n">bim_name</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_chr1_chr22_exomedata</span><span class="o">.</span><span class="n">bim</span>
<span class="n">annovar_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_annovar_chr1_22_exomes_postgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">annovar_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">workflow</span><span class="o">/</span><span class="n">QC_Exome_UKBB</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">humandb</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">ysm</span><span class="o">/</span><span class="n">datasets</span><span class="o">/</span><span class="n">db</span><span class="o">/</span><span class="n">annovar</span><span class="o">/</span><span class="n">humandb</span>
<span class="n">ukbb</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span>
<span class="n">container_annovar</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">annovar</span><span class="o">.</span><span class="n">sif</span> 
<span class="n">name_prefix</span><span class="o">=</span><span class="n">ukb23155_chr1_chr22</span>

<span class="c1"># Use the bim_merge workflow first and then the annovar workflow</span>
<span class="n">annovar_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;annovar</span>
<span class="s2">    --cwd $annovar_dir </span><span class="se">\</span>
<span class="s2">    --bedfiles $bedfiles</span><span class="se">\</span>
<span class="s2">    --bimfiles $bimfiles </span><span class="se">\</span>
<span class="s2">    --bim_name $bim_name </span><span class="se">\</span>
<span class="s2">    --humandb $humandb </span><span class="se">\</span>
<span class="s2">    --ukbb $ukbb </span><span class="se">\</span>
<span class="s2">    --job_size $job_size </span><span class="se">\</span>
<span class="s2">    --name_prefix $name_prefix </span><span class="se">\</span>
<span class="s2">    --container_annovar $container_annovar</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">annovar_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">annovar_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$annovar_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-07_annovar_chr1_22_exomes_postgwa.sbatch</span>
INFO: Workflow farnam (ID=w696e79ce57c148dd) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="2.-Hearing-difficulty/problems-f.2247">2. Hearing difficulty/problems f.2247<a class="anchor-link" href="#2.-Hearing-difficulty/problems-f.2247">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="FastGWA-job-white-British">FastGWA job white British<a class="anchor-link" href="#FastGWA-job-white-British">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2247_hearing_difficulty</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_imp</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_difficulty_f2247</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_difficulty_f2247</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">hearing_diff_new</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age_final_diff</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="FastGWA-all-white">FastGWA all white<a class="anchor-link" href="#FastGWA-all-white">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2247_hearing_difficulty_expandedwhite</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_expanded_white_imp</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">120120</span><span class="n">_UKBB_Hearing_difficulty_f2247_expandedwhite</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">120120</span><span class="n">_UKBB_Hearing_difficulty_f2247_expandedwhite</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">hearing_diff_new</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age_final_diff</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-12-02_f2247_expanded_white_imp-fastgwa.sbatch</span>
INFO: Workflow farnam (ID=9da5aed9605f3f9f) is executed successfully with 1 completed step.
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="FastGWA-exome-data">FastGWA exome data<a class="anchor-link" href="#FastGWA-exome-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2247_hearing_difficulty_exomes</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_exomes</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">bfile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">exome_files_snpsonly</span><span class="o">/</span><span class="n">ukb23155</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">merged</span><span class="o">.</span><span class="n">bed</span>
<span class="n">genoFile</span><span class="o">=</span><span class="sb">`echo $UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_c{1..22}_b0_v1.bed`</span>
<span class="n">sampleFile</span><span class="o">=</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="n">phenotypes_exome_data</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_difficulty_f2247_171970ind_exomes</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="n">phenotypes_exome_data</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_difficulty_f2247_171970ind_exomes</span>
<span class="n">formatFile_fastgwa</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">fastGWA_template</span><span class="o">.</span><span class="n">yml</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="s2">&quot;hearing_diff_new&quot;</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age_final_diff</span>
<span class="n">bgenMinMAF</span><span class="o">=</span><span class="mf">0.001</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --genoFile $genoFile </span>
<span class="s2">    --phenoFile $phenoFile</span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-01-12_f2247_exomes-fastgwa.sbatch</span>
INFO: Workflow farnam (ID=372a90dd555a65fb) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Regenie-exome-data">Regenie exome data<a class="anchor-link" href="#Regenie-exome-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[5]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_regenie</span><span class="o">=</span><span class="err">$</span><span class="n">lmm_exome_dir_regenie</span><span class="o">/</span><span class="n">f2247_hearing_difficulty_exomes_bfile</span>
<span class="n">lmm_sbatch_regenie</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_hearing_difficulty_exome_bfile</span><span class="o">-</span><span class="n">regenie</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="n">phenotypes_exome_data</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_difficulty_f2247_171970ind_exomes</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="n">phenotypes_exome_data</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_difficulty_f2247_171970ind_exomes</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">hearing_diff_new</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age_final_diff</span>
<span class="c1">#Use original bed files from the UKBB exome data</span>
<span class="c1">#bfile=$UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/exome_files_snpsonly/ukb23155.filtered.merged.bed</span>
<span class="c1">#Use the original bed files for the genotype array on regenie step1</span>
<span class="n">bfile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_originalgenotypefilesdownloaded083019</span><span class="o">/</span><span class="n">UKB_genotypedatadownloaded083019</span><span class="o">.</span><span class="n">bed</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;regenie</span>
<span class="s2">    --cwd $lmm_dir_regenie </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --genoFile $genoFile</span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_regenie </span>
<span class="s2">    --phenoCol $phenoCol</span>
<span class="s2">    --covarCol $covarCol  </span>
<span class="s2">    --qCovarCol $qCovarCol</span>
<span class="s2">    --bsize $bsize</span>
<span class="s2">    --lowmem_prefix $lowmem_prefix</span>
<span class="s2">    --trait $trait </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --maf_filter $maf_filter</span>
<span class="s2">    --geno_filter $geno_filter</span>
<span class="s2">    --hwe_filter $hwe_filter</span>
<span class="s2">    --mind_filter $mind_filter</span>
<span class="s2">    --minMAC $minMAC</span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --reverse_log_p $reverse_log_p</span>
<span class="s2">    --numThreads $numThreads</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_regenie</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-12_f2247_hearing_difficulty_exome_bfile-regenie.sbatch</span>
INFO: Workflow farnam (ID=w6ce6b23ca9b23cce) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="LD-clumping-job">LD clumping job<a class="anchor-link" href="#LD-clumping-job">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Imputed-data">Imputed data<a class="anchor-link" href="#Imputed-data">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">clumping_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span><span class="n">f2247_hearing_difficulty</span>
<span class="n">clumping_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">LD_Clumping</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">clumping_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_hearing_difficulty_ldclumping</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstatsFiles</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2247_hearing_difficulty</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_difficulty_f2247_hearing_diff_new</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>

<span class="n">clumping_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default </span>
<span class="s2">    --cwd $clumping_dir </span>
<span class="s2">    --bfile $bfile</span>
<span class="s2">    --bfile_ref $bfile_ref </span>
<span class="s2">    --bgenFile $bgenFile</span>
<span class="s2">    --sampleFile $sampleFile </span>
<span class="s2">    --sumstatsFiles $sumstatsFiles </span>
<span class="s2">    --unrelated_samples $unrelated_samples </span>
<span class="s2">    --ld_sample_size $ld_sample_size </span>
<span class="s2">    --clump_field $clump_field</span>
<span class="s2">    --clump_p1 $clump_p1 </span>
<span class="s2">    --clump_p2 $clump_p2 </span>
<span class="s2">    --clump_r2 $clump_r2 </span>
<span class="s2">    --clump_kb $clump_kb </span>
<span class="s2">    --clump_annotate $clump_annotate </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --job_size $clump_job_size</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">clumping_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">clumping_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$clumping_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Exome-data">Exome data<a class="anchor-link" href="#Exome-data">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[4]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">clumping_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span><span class="n">f2247_hearing_difficulty_exome</span>
<span class="n">clumping_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">LD_Clumping</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">clumping_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_hearing_diff_exome_ldclumping</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">bfile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_Caucasians_phenotypeindepqc120319_updated082020removedwithdrawnindiv</span><span class="o">.</span><span class="n">bed</span>
<span class="n">sumstatsFiles</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2247_hearing_difficulty_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_difficulty_f2247_171970ind_exomes_hearing_diff_new</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">genoFile</span><span class="o">=</span><span class="sb">`echo $UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_c{1..22}_b0_v1.bed`</span>
<span class="n">sampleFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">fam</span>
<span class="n">unrelated_samples</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">unrelated_n307259</span><span class="o">/</span><span class="n">UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620</span>
<span class="n">bfile_ref</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_s200631_chr1_22_exomedata</span><span class="o">.</span><span class="mf">1200.</span><span class="n">ref_geno</span><span class="o">.</span><span class="n">bed</span>
<span class="n">container_lmm</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="n">ld_sample_size</span><span class="o">=</span><span class="mi">1200</span>
<span class="n">clump_field</span><span class="o">=</span><span class="n">P</span>
<span class="n">clump_p1</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">clump_p2</span><span class="o">=</span><span class="mi">1</span>
<span class="n">clump_r2</span><span class="o">=</span><span class="mf">0.2</span>
<span class="n">clump_kb</span><span class="o">=</span><span class="mi">2000</span>
<span class="n">clump_annotate</span><span class="o">=</span><span class="n">BP</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">20</span>
<span class="n">clump_job_size</span><span class="o">=</span><span class="mi">1</span>

<span class="c1"># Select samples filter_samples workflow &amp; create reference file with reference workflow</span>
<span class="c1"># Then use default workflow to run the LD clumping</span>
<span class="n">clumping_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default</span>
<span class="s2">    --cwd $clumping_dir</span>
<span class="s2">    --bfile $bfile</span>
<span class="s2">    --bfile_ref $bfile_ref </span>
<span class="s2">    --genoFile $genoFile</span>
<span class="s2">    --sampleFile $sampleFile </span>
<span class="s2">    --sumstatsFiles $sumstatsFiles </span>
<span class="s2">    --unrelated_samples $unrelated_samples </span>
<span class="s2">    --ld_sample_size $ld_sample_size </span>
<span class="s2">    --clump_field $clump_field</span>
<span class="s2">    --clump_p1 $clump_p1 </span>
<span class="s2">    --clump_p2 $clump_p2 </span>
<span class="s2">    --clump_r2 $clump_r2 </span>
<span class="s2">    --clump_kb $clump_kb </span>
<span class="s2">    --clump_annotate $clump_annotate </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --job_size $clump_job_size</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">clumping_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">clumping_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$clumping_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-07_f2247_hearing_diff_exome_ldclumping.sbatch</span>
INFO: Workflow farnam (ID=w923c2b21e51a953e) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Post-GWAS-annotation">Post-GWAS annotation<a class="anchor-link" href="#Post-GWAS-annotation">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[8]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2247_hearing_difficulty_exomes</span>
<span class="n">postgwa_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_postgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstatsFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2247_hearing_difficulty_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_difficulty_f2247_171970ind_exomes_hearing_diff_new</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">postgwa_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">workflow</span><span class="o">/</span><span class="n">snptogene</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">hg</span><span class="o">=</span><span class="mi">38</span>
<span class="n">postgwa_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default</span>
<span class="s2">    --cwd $lmm_dir</span>
<span class="s2">    --sumstatsFile $sumstatsFile</span>
<span class="s2">    --hg $hg</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">postgwa_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">postgwa_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$postgwa_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-01-28_f2247_postgwa.sbatch</span>
INFO: Workflow farnam (ID=w50d0cee396725518) is executed successfully with 1 completed step.
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="3.-Hearing-difficulty-with-background-noise-f.2257">3. Hearing difficulty with background noise f.2257<a class="anchor-link" href="#3.-Hearing-difficulty-with-background-noise-f.2257">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="FastGWA-job-white-British">FastGWA job white British<a class="anchor-link" href="#FastGWA-job-white-British">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2257_hearing_background_noise</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2257_imp</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_background_noise_f2257</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_background_noise_f2257</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">hearing_noise_cat</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age_final_noise</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="FastGWA-job-all-white">FastGWA job all white<a class="anchor-link" href="#FastGWA-job-all-white">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[6]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2257_hearing_background_noise_expandedwhite</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2257_expandedwhite_imp</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">120120</span><span class="n">_UKBB_Hearing_background_noise_f2257_expandedwhite</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">120120</span><span class="n">_UKBB_Hearing_background_noise_f2257_expandedwhite</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">hearing_noise_cat</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age_final_noise</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">dewan</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">dewan</span>: Configuration for Yale `pi_dewan` partition cluster
INFO: <span class="ansi-green-fg">dewan</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">dewan</span> output:   <span class="ansi-green-fg">../output/2020-12-02_f2257_expandedwhite_imp-fastgwa.sbatch</span>
INFO: Workflow dewan (ID=5e423268c0ec2f3b) is executed successfully with 1 completed step.
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Regenie-exome-data">Regenie exome data<a class="anchor-link" href="#Regenie-exome-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[7]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_regenie</span><span class="o">=</span><span class="err">$</span><span class="n">lmm_exome_dir_regenie</span><span class="o">/</span><span class="n">f2257_hearing_noise_exomes_bfile</span>
<span class="n">lmm_sbatch_regenie</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2257_hearing_noise_exome_bfile</span><span class="o">-</span><span class="n">regenie</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="n">phenotypes_exome_data</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_background_noise_f2257_175531ind_exomes</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="n">phenotypes_exome_data</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_background_noise_f2257_175531ind_exomes</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">hearing_noise_cat</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age_final_noise</span>
<span class="c1">#Use original bed files from the UKBB exome data</span>
<span class="c1">#bfile=$UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/exome_files_snpsonly/ukb23155.filtered.merged.bed</span>
<span class="c1">#Use the original bed files for the genotype array on regenie step1</span>
<span class="n">bfile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_originalgenotypefilesdownloaded083019</span><span class="o">/</span><span class="n">UKB_genotypedatadownloaded083019</span><span class="o">.</span><span class="n">bed</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;regenie</span>
<span class="s2">    --cwd $lmm_dir_regenie </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --genoFile $genoFile</span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_regenie </span>
<span class="s2">    --phenoCol $phenoCol</span>
<span class="s2">    --covarCol $covarCol  </span>
<span class="s2">    --qCovarCol $qCovarCol</span>
<span class="s2">    --bsize $bsize</span>
<span class="s2">    --lowmem_prefix $lowmem_prefix</span>
<span class="s2">    --trait $trait </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --maf_filter $maf_filter</span>
<span class="s2">    --geno_filter $geno_filter</span>
<span class="s2">    --hwe_filter $hwe_filter</span>
<span class="s2">    --mind_filter $mind_filter</span>
<span class="s2">    --minMAC $minMAC</span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --reverse_log_p $reverse_log_p</span>
<span class="s2">    --numThreads $numThreads</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_regenie</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-12_f2257_hearing_noise_exome_bfile-regenie.sbatch</span>
INFO: Workflow farnam (ID=wd8efa7feebdc6a0e) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="LD-clumping-job">LD clumping job<a class="anchor-link" href="#LD-clumping-job">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Imputed-data">Imputed data<a class="anchor-link" href="#Imputed-data">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">clumping_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span><span class="n">f2257_hearing_background_noise</span>
<span class="n">clumping_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">LD_Clumping</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">clumping_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2257_hearing_background_noise_ldclumping</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstatsFiles</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2257_hearing_background_noise</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_background_noise_f2257_hearing_noise_cat</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>

<span class="n">clumping_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default </span>
<span class="s2">    --cwd $clumping_dir </span>
<span class="s2">    --bfile $bfile</span>
<span class="s2">    --bfile_ref $bfile_ref </span>
<span class="s2">    --bgenFile $bgenFile</span>
<span class="s2">    --sampleFile $sampleFile </span>
<span class="s2">    --sumstatsFiles $sumstatsFiles </span>
<span class="s2">    --unrelated_samples $unrelated_samples </span>
<span class="s2">    --ld_sample_size $ld_sample_size </span>
<span class="s2">    --clump_field $clump_field</span>
<span class="s2">    --clump_p1 $clump_p1 </span>
<span class="s2">    --clump_p2 $clump_p2 </span>
<span class="s2">    --clump_r2 $clump_r2 </span>
<span class="s2">    --clump_kb $clump_kb </span>
<span class="s2">    --clump_annotate $clump_annotate </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --job_size $clump_job_size</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">clumping_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">clumping_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$clumping_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Exome-data">Exome data<a class="anchor-link" href="#Exome-data">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[5]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">clumping_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span><span class="n">f2257_hearing_background_noise_exome</span>
<span class="n">clumping_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">LD_Clumping</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">clumping_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2257_hearing_background_noise_exome_ldclumping</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">bfile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_Caucasians_phenotypeindepqc120319_updated082020removedwithdrawnindiv</span><span class="o">.</span><span class="n">bed</span>
<span class="n">sumstatsFiles</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2257_hearing_noise_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_background_noise_f2257_175531ind_exomes_hearing_noise_cat</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">genoFile</span><span class="o">=</span><span class="sb">`echo $UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_c{1..22}_b0_v1.bed`</span>
<span class="n">sampleFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">fam</span>
<span class="n">unrelated_samples</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">unrelated_n307259</span><span class="o">/</span><span class="n">UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620</span>
<span class="n">bfile_ref</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_s200631_chr1_22_exomedata</span><span class="o">.</span><span class="mf">1200.</span><span class="n">ref_geno</span><span class="o">.</span><span class="n">bed</span>
<span class="n">container_lmm</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="n">ld_sample_size</span><span class="o">=</span><span class="mi">1200</span>
<span class="n">clump_field</span><span class="o">=</span><span class="n">P</span>
<span class="n">clump_p1</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">clump_p2</span><span class="o">=</span><span class="mi">1</span>
<span class="n">clump_r2</span><span class="o">=</span><span class="mf">0.2</span>
<span class="n">clump_kb</span><span class="o">=</span><span class="mi">2000</span>
<span class="n">clump_annotate</span><span class="o">=</span><span class="n">BP</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">20</span>
<span class="n">clump_job_size</span><span class="o">=</span><span class="mi">1</span>

<span class="c1"># Select samples filter_samples workflow &amp; create reference file with reference workflow</span>
<span class="c1"># Then use default workflow to run the LD clumping</span>
<span class="n">clumping_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default</span>
<span class="s2">    --cwd $clumping_dir</span>
<span class="s2">    --bfile $bfile</span>
<span class="s2">    --bfile_ref $bfile_ref </span>
<span class="s2">    --genoFile $genoFile</span>
<span class="s2">    --sampleFile $sampleFile </span>
<span class="s2">    --sumstatsFiles $sumstatsFiles </span>
<span class="s2">    --unrelated_samples $unrelated_samples </span>
<span class="s2">    --ld_sample_size $ld_sample_size </span>
<span class="s2">    --clump_field $clump_field</span>
<span class="s2">    --clump_p1 $clump_p1 </span>
<span class="s2">    --clump_p2 $clump_p2 </span>
<span class="s2">    --clump_r2 $clump_r2 </span>
<span class="s2">    --clump_kb $clump_kb </span>
<span class="s2">    --clump_annotate $clump_annotate </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --job_size $clump_job_size</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">clumping_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">clumping_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$clumping_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-07_f2257_hearing_diff_exome_ldclumping.sbatch</span>
INFO: Workflow farnam (ID=w7c7b3789d732c7e0) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Post-GWAS-annotation">Post-GWAS annotation<a class="anchor-link" href="#Post-GWAS-annotation">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[9]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2257_hearing_noise_exomes</span>
<span class="n">postgwa_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2257_postgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstatsFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2257_hearing_noise_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_background_noise_f2257_175531ind_exomes_hearing_noise_cat</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">postgwa_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">workflow</span><span class="o">/</span><span class="n">snptogene</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">hg</span><span class="o">=</span><span class="mi">38</span>
<span class="n">postgwa_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default</span>
<span class="s2">    --cwd $lmm_dir</span>
<span class="s2">    --sumstatsFile $sumstatsFile</span>
<span class="s2">    --hg $hg</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">postgwa_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">postgwa_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$postgwa_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-01-28_f2257_postgwa.sbatch</span>
INFO: Workflow farnam (ID=w9856930ebcb27a9d) is executed successfully with 1 completed step.
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="4.-Combined-phenotype-f.2247-&amp;-f.2257">4. Combined phenotype f.2247 &amp; f.2257<a class="anchor-link" href="#4.-Combined-phenotype-f.2247-&amp;-f.2257">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="FastGWA-job-white-British">FastGWA job white British<a class="anchor-link" href="#FastGWA-job-white-British">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2247_f2257_combined</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_f2257_imp</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_f2247_f2257</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_f2247_f2257</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">f2247_f2257</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="FastGWA-job-all-white">FastGWA job all white<a class="anchor-link" href="#FastGWA-job-all-white">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[7]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2247_f2257_combined_expandedwhite</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_f2257_expandedwhite_imp</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">120120</span><span class="n">_UKBB_f2247_f2257_expandedwhite</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">120120</span><span class="n">_UKBB_f2247_f2257_expandedwhite</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">f2247_f2257</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">dewan</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">dewan</span>: Configuration for Yale `pi_dewan` partition cluster
INFO: <span class="ansi-green-fg">dewan</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">dewan</span> output:   <span class="ansi-green-fg">../output/2020-12-02_f2247_f2257_expandedwhite_imp-fastgwa.sbatch</span>
INFO: Workflow dewan (ID=cca88c31ebd37730) is executed successfully with 1 completed step.
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Bolt-LMM-job">Bolt-LMM job<a class="anchor-link" href="#Bolt-LMM-job">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_bolt</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2247_f2257_combined</span>
<span class="n">lmm_sbatch_bolt</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_f2257_imp</span><span class="o">-</span><span class="n">bolt</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_f2247_f2257</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_f2247_f2257</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">f2247_f2257</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age</span>
<span class="n">lmm_option</span><span class="o">=</span><span class="s1">&#39;lmmForceNonInf&#39;</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;boltlmm</span>
<span class="s2">    --cwd $lmm_dir_bolt </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_bolt </span>
<span class="s2">    --covarFile $covarFile </span>
<span class="s2">    --LDscoresFile $LDscoresFile </span>
<span class="s2">    --geneticMapFile $geneticMapFile </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --lmm_option $lmm_option</span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp    </span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_bolt</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-10-28_f2247_f2257_imp-bolt.sbatch</span>
INFO: Workflow farnam (ID=6f1d2712738ba187) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Regenie-exome-data">Regenie exome data<a class="anchor-link" href="#Regenie-exome-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[8]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_regenie</span><span class="o">=</span><span class="err">$</span><span class="n">lmm_exome_dir_regenie</span><span class="o">/</span><span class="n">f2247_f2257_combined_exomes_bfile</span>
<span class="n">lmm_sbatch_regenie</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_f2257_combined_exome_bfile</span><span class="o">-</span><span class="n">regenie</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="n">phenotypes_exome_data</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_f2247_f2257_136862ind_exomes</span>
<span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="n">hearing_pheno_path</span><span class="o">/</span><span class="n">phenotypes_exome_data</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_f2247_f2257_136862ind_exomes</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="n">f2247_f2257</span>
<span class="n">covarCol</span><span class="o">=</span><span class="n">sex</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">age_combined</span>
<span class="c1">#Use original bed files from the UKBB exome data</span>
<span class="c1">#bfile=$UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/exome_files_snpsonly/ukb23155.filtered.merged.bed</span>
<span class="c1">#Use the original bed files for the genotype array on regenie step1</span>
<span class="n">bfile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_originalgenotypefilesdownloaded083019</span><span class="o">/</span><span class="n">UKB_genotypedatadownloaded083019</span><span class="o">.</span><span class="n">bed</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;regenie</span>
<span class="s2">    --cwd $lmm_dir_regenie </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --genoFile $genoFile</span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_regenie </span>
<span class="s2">    --phenoCol $phenoCol</span>
<span class="s2">    --covarCol $covarCol  </span>
<span class="s2">    --qCovarCol $qCovarCol</span>
<span class="s2">    --bsize $bsize</span>
<span class="s2">    --lowmem_prefix $lowmem_prefix</span>
<span class="s2">    --trait $trait </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --maf_filter $maf_filter</span>
<span class="s2">    --geno_filter $geno_filter</span>
<span class="s2">    --hwe_filter $hwe_filter</span>
<span class="s2">    --mind_filter $mind_filter</span>
<span class="s2">    --minMAC $minMAC</span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --reverse_log_p $reverse_log_p</span>
<span class="s2">    --numThreads $numThreads</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_regenie</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-12_f2247_f2257_combined_exome_bfile-regenie.sbatch</span>
INFO: Workflow farnam (ID=wbd137a88958a3c38) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="LD-clumping-job">LD clumping job<a class="anchor-link" href="#LD-clumping-job">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Imputed-data">Imputed data<a class="anchor-link" href="#Imputed-data">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">clumping_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span><span class="n">f2247_f2257_combined</span>
<span class="n">clumping_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">LD_Clumping</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">clumping_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_f2257_combined_ldclumping</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstatsFiles</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2247_f2257_combined</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_f2247_f2257_f2247_f2257</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>

<span class="n">clumping_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default </span>
<span class="s2">    --cwd $clumping_dir </span>
<span class="s2">    --bfile $bfile</span>
<span class="s2">    --bfile_ref $bfile_ref </span>
<span class="s2">    --bgenFile $bgenFile</span>
<span class="s2">    --sampleFile $sampleFile </span>
<span class="s2">    --sumstatsFiles $sumstatsFiles </span>
<span class="s2">    --unrelated_samples $unrelated_samples </span>
<span class="s2">    --ld_sample_size $ld_sample_size </span>
<span class="s2">    --clump_field $clump_field</span>
<span class="s2">    --clump_p1 $clump_p1 </span>
<span class="s2">    --clump_p2 $clump_p2 </span>
<span class="s2">    --clump_r2 $clump_r2 </span>
<span class="s2">    --clump_kb $clump_kb </span>
<span class="s2">    --clump_annotate $clump_annotate </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --job_size $clump_job_size</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">clumping_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">clumping_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$clumping_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Exome-data">Exome data<a class="anchor-link" href="#Exome-data">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[6]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">clumping_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span><span class="n">f2247_f2257_combined_exome</span>
<span class="n">clumping_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">LD_Clumping</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">clumping_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_f2257_combined_exome_ldclumping</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">bfile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_Caucasians_phenotypeindepqc120319_updated082020removedwithdrawnindiv</span><span class="o">.</span><span class="n">bed</span>
<span class="n">sumstatsFiles</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2247_f2257_combined_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_f2247_f2257_136862ind_exomes_f2247_f2257</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">genoFile</span><span class="o">=</span><span class="sb">`echo $UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_c{1..22}_b0_v1.bed`</span>
<span class="n">sampleFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">fam</span>
<span class="n">unrelated_samples</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">unrelated_n307259</span><span class="o">/</span><span class="n">UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620</span>
<span class="n">bfile_ref</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_s200631_chr1_22_exomedata</span><span class="o">.</span><span class="mf">1200.</span><span class="n">ref_geno</span><span class="o">.</span><span class="n">bed</span>
<span class="n">container_lmm</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="n">ld_sample_size</span><span class="o">=</span><span class="mi">1200</span>
<span class="n">clump_field</span><span class="o">=</span><span class="n">P</span>
<span class="n">clump_p1</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">clump_p2</span><span class="o">=</span><span class="mi">1</span>
<span class="n">clump_r2</span><span class="o">=</span><span class="mf">0.2</span>
<span class="n">clump_kb</span><span class="o">=</span><span class="mi">2000</span>
<span class="n">clump_annotate</span><span class="o">=</span><span class="n">BP</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">20</span>
<span class="n">clump_job_size</span><span class="o">=</span><span class="mi">1</span>

<span class="c1"># Select samples filter_samples workflow &amp; create reference file with reference workflow</span>
<span class="c1"># Then use default workflow to run the LD clumping</span>
<span class="n">clumping_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default</span>
<span class="s2">    --cwd $clumping_dir</span>
<span class="s2">    --bfile $bfile</span>
<span class="s2">    --bfile_ref $bfile_ref </span>
<span class="s2">    --genoFile $genoFile</span>
<span class="s2">    --sampleFile $sampleFile </span>
<span class="s2">    --sumstatsFiles $sumstatsFiles </span>
<span class="s2">    --unrelated_samples $unrelated_samples </span>
<span class="s2">    --ld_sample_size $ld_sample_size </span>
<span class="s2">    --clump_field $clump_field</span>
<span class="s2">    --clump_p1 $clump_p1 </span>
<span class="s2">    --clump_p2 $clump_p2 </span>
<span class="s2">    --clump_r2 $clump_r2 </span>
<span class="s2">    --clump_kb $clump_kb </span>
<span class="s2">    --clump_annotate $clump_annotate </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --job_size $clump_job_size</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">clumping_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">clumping_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$clumping_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-07_f2247_f2257_combined_exome_ldclumping.sbatch</span>
INFO: Workflow farnam (ID=wf81e3f1aaede6aab) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Post-GWAS-annotation-Snp-to-gene">Post-GWAS annotation Snp-to-gene<a class="anchor-link" href="#Post-GWAS-annotation-Snp-to-gene">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[10]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2247_f2257_combined_exomes</span>
<span class="n">postgwa_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_f2257_postgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstatsFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2247_f2257_combined_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_f2247_f2257_136862ind_exomes_f2247_f2257</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">postgwa_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">workflow</span><span class="o">/</span><span class="n">snptogene</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">hg</span><span class="o">=</span><span class="mi">38</span>

<span class="n">postgwa_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default</span>
<span class="s2">    --cwd $lmm_dir</span>
<span class="s2">    --sumstatsFile $sumstatsFile</span>
<span class="s2">    --hg $hg</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">postgwa_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">postgwa_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$postgwa_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-01-28_f2247_f2257_postgwa.sbatch</span>
INFO: Workflow farnam (ID=w311a9815d7e57ba3) is executed successfully with 1 completed step.
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Post-GWAS-annotation-ANNOVAR">Post-GWAS annotation ANNOVAR<a class="anchor-link" href="#Post-GWAS-annotation-ANNOVAR">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2247_f2257_combined_exomes</span>
<span class="n">postgwa_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_f2257_postgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">postgwa_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">workflow</span><span class="o">/</span><span class="n">snptogene</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">sumstatsFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2247_f2257_combined_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_f2247_f2257_136862ind_exomes_f2247_f2257</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">hg</span><span class="o">=</span><span class="mi">38</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_annovar</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">annovar</span><span class="o">.</span><span class="n">sif</span>
<span class="n">bimfiles</span><span class="o">=</span><span class="sb">`echo $UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_c{1..22}_b0_v1.bim`</span>
<span class="n">bim_name</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">output</span><span class="o">/</span><span class="n">ukb23155_chr1_chr22</span><span class="o">.</span><span class="n">bim</span>
<span class="n">humandb</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">ysm</span><span class="o">/</span><span class="n">datasets</span><span class="o">/</span><span class="n">db</span><span class="o">/</span><span class="n">annovar</span><span class="o">/</span><span class="n">humandb</span>

<span class="n">annovar_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;annovar</span>
<span class="s2">    --cwd $lmm_dir</span>
<span class="s2">    --hg $hg</span>
<span class="s2">    --bimfiles $bimfiles</span>
<span class="s2">    --bim_name $bim_name</span>
<span class="s2">    --sumstatsFile $sumstatsFile</span>
<span class="s2">    --hg $hg</span>
<span class="s2">    --humandb $humandb</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --container_annovar $container_annovar</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">postgwa_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">postgwa_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$postgwa_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="5.-Hudson-plot">5. Hudson plot<a class="anchor-link" href="#5.-Hudson-plot">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="f2247_f2257-combined-vs-Hdiff-wells-paper">f2247_f2257 combined vs Hdiff wells paper<a class="anchor-link" href="#f2247_f2257-combined-vs-Hdiff-wells-paper">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">hudson_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Hudson_plot</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">hudson_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">hudson_plots</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span>
<span class="n">hudson_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_f2257_vs_hdiff_hudson</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstats_1</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2247_f2257_combined</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_f2247_f2257_f2247_f2257</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">sumstats_2</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">project</span><span class="o">/</span><span class="n">HI_UKBB</span><span class="o">/</span><span class="mi">2019</span><span class="n">_Wells_sumstats</span><span class="o">/</span><span class="n">HD_EA_gwas_sumstats</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span>
<span class="n">toptitle</span><span class="o">=</span><span class="s2">&quot;f2247_f2257_combined&quot;</span>
<span class="n">bottomtitle</span><span class="o">=</span><span class="s2">&quot;Hdiff_Wells_GWAS&quot;</span>
<span class="n">highlight_p_top</span><span class="o">=</span><span class="mf">0.0</span>
<span class="n">highlight_p_bottom</span><span class="o">=</span><span class="mf">0.0</span>
<span class="n">pval_filter</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">highlight_snp</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">project</span><span class="o">/</span><span class="n">HI_UKBB</span><span class="o">/</span><span class="mi">2019</span><span class="n">_Wells_sumstats</span><span class="o">/</span><span class="n">hdiff_snps_wells</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_lmm</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>

<span class="n">hudson_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;hudson</span>
<span class="s2">    --cwd $hudson_dir</span>
<span class="s2">    --sumstats_1 $sumstats_1</span>
<span class="s2">    --sumstats_2 $sumstats_2</span>
<span class="s2">    --toptitle $toptitle</span>
<span class="s2">    --bottomtitle $bottomtitle</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --highlight_p_top $highlight_p_top</span>
<span class="s2">    --highlight_p_bottom $highlight_p_bottom</span>
<span class="s2">    --pval_filter $pval_filter</span>
<span class="s2">    --highlight_snp $highlight_snp</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">hudson_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">hudson_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$hudson_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-10-26_f2247_f2257_vs_hdiff_hudson.sbatch</span>
INFO: Workflow farnam (ID=f34e7c33d379875d) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="f3393-vs-haid-wells-paper">f3393 vs haid wells paper<a class="anchor-link" href="#f3393-vs-haid-wells-paper">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">hudson_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Hudson_plot</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">hudson_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">hudson_plots</span><span class="o">/</span><span class="n">hearing_impairment</span><span class="o">/</span>
<span class="n">hudson_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_vs_Haid_hudson</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstats_1</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f3393_hearing_aid</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_aid_f3393_hearing_aid_cat</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">sumstats_2</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">project</span><span class="o">/</span><span class="n">HI_UKBB</span><span class="o">/</span><span class="mi">2019</span><span class="n">_Wells_sumstats</span><span class="o">/</span><span class="n">HAID_EA_gwas_sumstats</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span>
<span class="n">toptitle</span><span class="o">=</span><span class="s2">&quot;f_3393_hearing_aid&quot;</span>
<span class="n">bottomtitle</span><span class="o">=</span><span class="s2">&quot;Haid_Wells_GWAS&quot;</span>
<span class="n">highlight_p_top</span><span class="o">=</span><span class="mf">0.0</span>
<span class="n">highlight_p_bottom</span><span class="o">=</span><span class="mf">0.0</span>
<span class="n">pval_filter</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">highlight_snp</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">project</span><span class="o">/</span><span class="n">HI_UKBB</span><span class="o">/</span><span class="mi">2019</span><span class="n">_Wells_sumstats</span><span class="o">/</span><span class="n">haid_snps_wells</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_lmm</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>

<span class="n">hudson_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;hudson</span>
<span class="s2">    --cwd $hudson_dir</span>
<span class="s2">    --sumstats_1 $sumstats_1</span>
<span class="s2">    --sumstats_2 $sumstats_2</span>
<span class="s2">    --toptitle $toptitle</span>
<span class="s2">    --bottomtitle $bottomtitle</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --highlight_p_top $highlight_p_top</span>
<span class="s2">    --highlight_p_bottom $highlight_p_bottom</span>
<span class="s2">    --pval_filter $pval_filter</span>
<span class="s2">    --highlight_snp $highlight_snp</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">hudson_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">hudson_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$hudson_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-10-22_f3393_vs_Haid_hudson.sbatch</span>
INFO: Workflow farnam (ID=f48b56e3e091ef16) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="f.2247-exome-and-inputed-data">f.2247 exome and inputed data<a class="anchor-link" href="#f.2247-exome-and-inputed-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[7]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">hudson_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Hudson_plot</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">hudson_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">hudson_plots</span>
<span class="n">hudson_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_imp_exome_hudson</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstats_1</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2247_hearing_difficulty</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_difficulty_f2247_hearing_diff_new</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">sumstats_2</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2247_hearing_difficulty_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_difficulty_f2247_171970ind_exomes_hearing_diff_new</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">toptitle</span><span class="o">=</span><span class="s2">&quot;f2247_imputed&quot;</span>
<span class="n">bottomtitle</span><span class="o">=</span><span class="s2">&quot;f2247_exome&quot;</span>
<span class="n">phenocol1</span><span class="o">=</span><span class="s2">&quot;f2247_imputed&quot;</span>
<span class="n">phenocol2</span><span class="o">=</span><span class="s2">&quot;f2247_exome&quot;</span>
<span class="n">highlight_p_top</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">highlight_p_bottom</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">pval_filter</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_lmm</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>

<span class="n">hudson_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;hudson</span>
<span class="s2">    --cwd $hudson_dir</span>
<span class="s2">    --sumstats_1 $sumstats_1</span>
<span class="s2">    --sumstats_2 $sumstats_2</span>
<span class="s2">    --toptitle $toptitle</span>
<span class="s2">    --bottomtitle $bottomtitle</span>
<span class="s2">    --phenocol1 $phenoCol1</span>
<span class="s2">    --phenocol2 $phenoCol2</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --highlight_p_top $highlight_p_top</span>
<span class="s2">    --highlight_p_bottom $highlight_p_bottom</span>
<span class="s2">    --pval_filter $pval_filter</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">hudson_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">hudson_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$hudson_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-02_f2247_imp_exome_hudson.sbatch</span>
INFO: Workflow farnam (ID=w2d252da39eb028af) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="f.2257-exome-and-imputed-data">f.2257 exome and imputed data<a class="anchor-link" href="#f.2257-exome-and-imputed-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[8]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">hudson_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Hudson_plot</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">hudson_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">hudson_plots</span>
<span class="n">hudson_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2257_imp_exome_hudson</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstats_1</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2257_hearing_background_noise</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_background_noise_f2257_hearing_noise_cat</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">sumstats_2</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2257_hearing_noise_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_background_noise_f2257_175531ind_exomes_hearing_noise_cat</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">toptitle</span><span class="o">=</span><span class="s2">&quot;f2257_imputed&quot;</span>
<span class="n">bottomtitle</span><span class="o">=</span><span class="s2">&quot;f2257_exome&quot;</span>
<span class="n">phenocol1</span><span class="o">=</span><span class="s2">&quot;f2257_imputed&quot;</span>
<span class="n">phenocol2</span><span class="o">=</span><span class="s2">&quot;f2257_exome&quot;</span>
<span class="n">highlight_p_top</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">highlight_p_bottom</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">pval_filter</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_lmm</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>

<span class="n">hudson_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;hudson</span>
<span class="s2">    --cwd $hudson_dir</span>
<span class="s2">    --sumstats_1 $sumstats_1</span>
<span class="s2">    --sumstats_2 $sumstats_2</span>
<span class="s2">    --toptitle $toptitle</span>
<span class="s2">    --bottomtitle $bottomtitle</span>
<span class="s2">    --phenocol1 $phenocol1</span>
<span class="s2">    --phenocol2 $phenocol2</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --highlight_p_top $highlight_p_top</span>
<span class="s2">    --highlight_p_bottom $highlight_p_bottom</span>
<span class="s2">    --pval_filter $pval_filter</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">hudson_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">hudson_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$hudson_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-02_f2257_imp_exome_hudson.sbatch</span>
INFO: Workflow farnam (ID=w0c5959e07610ebcd) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Combined-phenotype">Combined phenotype<a class="anchor-link" href="#Combined-phenotype">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[8]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">hudson_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Hudson_plot</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">hudson_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">hudson_plots</span>
<span class="n">hudson_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f2247_f2257_imp_exome_hudson</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstats_1</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f2247_f2257_combined</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_f2247_f2257_f2247_f2257</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">sumstats_2</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f2247_f2257_combined_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_f2247_f2257_136862ind_exomes_f2247_f2257</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">toptitle</span><span class="o">=</span><span class="s2">&quot;Combined_f2247_f2257_imputed&quot;</span>
<span class="n">bottomtitle</span><span class="o">=</span><span class="s2">&quot;Combined_f2247_f2257_exome&quot;</span>
<span class="n">phenocol1</span><span class="o">=</span><span class="s2">&quot;f2247_f2257_imputed&quot;</span>
<span class="n">phenocol2</span><span class="o">=</span><span class="s2">&quot;f2247_f2257_exome&quot;</span>
<span class="n">highlight_p_top</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">highlight_p_bottom</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">pval_filter</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_lmm</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>

<span class="n">hudson_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;hudson</span>
<span class="s2">    --cwd $hudson_dir</span>
<span class="s2">    --sumstats_1 $sumstats_1</span>
<span class="s2">    --sumstats_2 $sumstats_2</span>
<span class="s2">    --toptitle $toptitle</span>
<span class="s2">    --bottomtitle $bottomtitle</span>
<span class="s2">    --phenocol1 $phenocol1</span>
<span class="s2">    --phenocol2 $phenocol2</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --highlight_p_top $highlight_p_top</span>
<span class="s2">    --highlight_p_bottom $highlight_p_bottom</span>
<span class="s2">    --pval_filter $pval_filter</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">hudson_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">hudson_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$hudson_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-03_f2247_f2257_imp_exome_hudson.sbatch</span>
INFO: Workflow farnam (ID=w093ec0ac945b39c0) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="f3393-exome-and-imputed-data">f3393 exome and imputed data<a class="anchor-link" href="#f3393-exome-and-imputed-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">hudson_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Hudson_plot</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">hudson_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">hudson_plots</span>
<span class="n">hudson_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_imp_exome_hudson</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstats_1</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">f3393_hearing_aid</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_aid_f3393_hearing_aid_cat</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">sumstats_2</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_exome_data</span><span class="o">/</span><span class="n">f3393_hearing_aid_exomes</span><span class="o">/</span><span class="mo">010421</span><span class="n">_UKBB_Hearing_aid_f3393_128254ind_exomes_hearing_aid_cat</span><span class="o">.</span><span class="n">regenie</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">toptitle</span><span class="o">=</span><span class="s2">&quot;f3393_imputed&quot;</span>
<span class="n">bottomtitle</span><span class="o">=</span><span class="s2">&quot;f3393_exome&quot;</span>
<span class="n">phenocol1</span><span class="o">=</span><span class="s2">&quot;f3393_imputed&quot;</span>
<span class="n">phenocol2</span><span class="o">=</span><span class="s2">&quot;f3393_exome&quot;</span>
<span class="n">highlight_p_top</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">highlight_p_bottom</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">pval_filter</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_lmm</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>

<span class="n">hudson_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;hudson</span>
<span class="s2">    --cwd $hudson_dir</span>
<span class="s2">    --sumstats_1 $sumstats_1</span>
<span class="s2">    --sumstats_2 $sumstats_2</span>
<span class="s2">    --toptitle $toptitle</span>
<span class="s2">    --bottomtitle $bottomtitle</span>
<span class="s2">    --phenocol1 $phenocol1</span>
<span class="s2">    --phenocol2 $phenocol2</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --highlight_p_top $highlight_p_top</span>
<span class="s2">    --highlight_p_bottom $highlight_p_bottom</span>
<span class="s2">    --pval_filter $pval_filter</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">hudson_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">hudson_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$hudson_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-03_f3393_imp_exome_hudson.sbatch</span>
INFO: Workflow farnam (ID=w7da36fe5bdb68b67) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="6.-Fine-mapping">6. Fine mapping<a class="anchor-link" href="#6.-Fine-mapping">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="f.3391-hearing-aid">f.3391 hearing aid<a class="anchor-link" href="#f.3391-hearing-aid">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">finemap_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">SuSiE_RSS</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">finemap_dir</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">fine_mapping</span><span class="o">/</span><span class="n">f3393_hearing_aid</span>
<span class="n">finemap_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_f3393_hearing_aid_susie</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstatFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">region_extraction</span><span class="o">/</span><span class="n">f3393_hearing_aid</span><span class="o">/</span><span class="mi">10</span><span class="n">_126783170_126813028</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_aid_f3393_hearing_aid_cat</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats_10_126783170_126813028</span><span class="o">.</span><span class="n">sumstats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">ldFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">region_extraction</span><span class="o">/</span><span class="n">f3393_hearing_aid</span><span class="o">/</span><span class="mi">10</span><span class="n">_126783170_126813028</span><span class="o">/</span><span class="mi">200828</span><span class="n">_UKBB_Hearing_aid_f3393_hearing_aid_cat</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats_10_126783170_126813028</span><span class="o">.</span><span class="n">sample_ld</span><span class="o">.</span><span class="n">gz</span>
<span class="n">N</span><span class="o">=</span><span class="mi">230411</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_lmm</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>

<span class="n">finemap_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;default</span>
<span class="s2">    --cwd $finemap_dir</span>
<span class="s2">    --sumstatFile $sumstatFile</span>
<span class="s2">    --ldFile $ldFile</span>
<span class="s2">    --N $N</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">finemap_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">finemap_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$finemap_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-10-15_f3393_hearing_aid_susie.sbatch</span>
INFO: Workflow farnam (ID=fdf18b3eb0fe563d) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span>
</pre></div>

    </div>
</div>
</div>

</div>
<hr>
&copy 2020 Diana Cornejo at Center for Statistical Genetics, Sergievsky Center, Columbia University VP&S
<p><small>Exported from <a href="http://github.com/dianacornejo/pleiotropy_UKB/blob/804a49e20903d8a6a7eafcb7472453c5ea2dcde8/analysis/cluster_scripts/Hearing_impairment_scripts.ipynb"><code>analysis/cluster_scripts/Hearing_impairment_scripts.ipynb</code></a> committed by dc2325 on Fri Feb 19 22:22:36 2021 <a href="http://github.com/dianacornejo/pleiotropy_UKB/commit/804a49e20903d8a6a7eafcb7472453c5ea2dcde8">revision 1, 804a49e</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
