<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=analysisDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=analysisArray;
            var docs_map=analysisArrayMap;
            var pos=analysisArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Pleiotropy analysis in UKBB</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Pleiotropy analysis in UKBB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../workflow.html">Workflow</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/dianacornejo/pleiotropy_UKB"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Analyses-of-the-lipids-and-other-blood-traits-from-UKBB">Analyses of the lipids and other blood traits from UKBB<a class="anchor-link" href="#Analyses-of-the-lipids-and-other-blood-traits-from-UKBB">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>This notebook applies the <code>Get_Job_Script.ipynb</code> to automatically generate the sbatch scripts to run in Yale's cluster. The end result is to apply <a href="https://github.com/statgenetics/UKBB_GWAS_dev/tree/master/workflow">various LMM workflows</a> to perform association analysis of different lipid traits (cholesterol, HDL, LDL, triglycerides), do clumping analysis and extract associated regions.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="File-paths-on-Yale-cluster">File paths on Yale cluster<a class="anchor-link" href="#File-paths-on-Yale-cluster">&#182;</a></h2><ul>
<li>Genotype files in PLINK format:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/pleiotropy_geneticfiles/UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv</code></li>
<li>Genotype files in bgen format:
<code>SAY/dbgapstg/scratch/UKBiobank/genotype_files/ukb39554_imputeddataset/</code></li>
<li>Summary stats for imputed variants BOLT-LMM:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/results/BOLTLMM_results/results_imputed_data</code></li>
<li>Summary stats for inputed variants FastGWA:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/results/FastGWA_results/results_imputed_data</code></li>
<li>Phenotype files:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/phenotype_files/pleiotropy_R01/phenotypesforanalysis/UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis</code></li>
<li>Relationship file:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/pleiotropy_geneticfiles/unrelated_n307259/UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620</code>
-Other traits to be analyzed:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/phenotype_files/pleiotropy_R01/phenotypesforanalysis/UKB_CAUC_lipidsforanalysis_apolipoproteinAandB,Hba1c_continuousandcategorical,egfrbyCKDEPI,serumcreatinine,UACR_inverseranknorm_110320</code></li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="07/28/20-analysis">07/28/20 analysis<a class="anchor-link" href="#07/28/20-analysis">&#182;</a></h2><p>On the cluster, open up this notebook using the JupyterLab server you set up via the ssh channel, then run the following cells,</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Important-analysis-notes">Important analysis notes<a class="anchor-link" href="#Important-analysis-notes">&#182;</a></h2><p>Corrected measures are being used for LDL (LDL/0.7) and total cholesterol (TC/0.8). Correction used for those on medication only.</p>
<p>Triglycerides and HDL are used without correction.</p>
<p>Covariates are age at recruitment, genetic sex, medication status, smoking (1 &amp; 2), and alcohol (1, 2, 3, 4). Smoking and alcohol are dummy coded</p>
<p>Alcohol: f.1558 (categories never/special ocassion collapsed)
Smoking: f.20116 (never/former/current). Plan is to replace with f.20161 (pack years) once we have that variable.
Medication status: cholesterolmedbyf6153or6177or20003 (It's what was used to adjust the phenotype)</p>
<h3 id="Inverse-rank-normalized-traits">Inverse rank normalized traits<a class="anchor-link" href="#Inverse-rank-normalized-traits">&#182;</a></h3><p><code>/gpfs/gibbs/pi/dewan/data/UKBiobank/phenotype_files/pleiotropy_R01/phenotypesforanalysis/UKBCauc_cholesterolandbloodpressurefields_inverseranknorm_covariatesage,sex,alcohol,smokingpackyears_foranalysis</code></p>
<p>The variables for analysis are:</p>
<ol>
<li>HDL_inverseranknorm</li>
<li>LDL_medcorrected_inverseranknorm</li>
<li>triglycerides_inverseranknorm</li>
<li>cholesterol.f30690_medcorrected_inverseranknorm</li>
<li>diastolic_plus10inmedicated_inverserank</li>
<li>systolic_plus15inmedicated_inverserank</li>
</ol>
<p>Potential covariates:</p>
<ul>
<li>age</li>
<li>sex</li>
<li>smoking_packyears (there is also smoking_dummy1 and smoking_dummy2, which is from before we had the pack years available)</li>
<li>alcohol1, alcohol2, alcohol3, alcohol4</li>
</ul>
<p><strong>The phenotypes (the continuous ones have already been inverse rank normalized, as indicated by the _int at the ends of their names) are:</strong></p>
<ol>
<li>ApoA_int (Apolipoprotein A)</li>
<li>ApoB_int (Apolipoprotein B)</li>
<li>Creatinine_int (Serum Creatinine)</li>
<li>Hba1c_int (continuous Hba1c)</li>
<li>Hba1c_categorical (categorical Hba1c, with values 1-3: 1 is Hba1c &lt;= 42, 2 is Hba1c &gt; 42 &amp; &lt; 48, and 3 is Hba1c &gt;= 48)</li>
<li>eGFR_int (calculated with CKD-EPI equation using serum creatinine (µmol/l)),</li>
<li>UACR_int (urinary albumin-to-creatinine ratio (UACR), calculated with urinary albumin (mg/dL) and urinary creatinine (g/dL))</li>
</ol>
<p>Covariates in the file are:</p>
<ul>
<li>Age</li>
<li>Genetic Sex</li>
<li>Smoking (pack years)</li>
<li>Alcohol (dummy coded, Alcohol1, Alcohol2, Alcohol3, Alcohol4)</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Bash-variables-for-workflow-configuration">Bash variables for workflow configuration<a class="anchor-link" href="#Bash-variables-for-workflow-configuration">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Common variables</span>
<span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">bfile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_Caucasians_phenotypeindepqc120319_updated082020removedwithdrawnindiv</span><span class="o">.</span><span class="n">bed</span>
<span class="n">sampleFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb39554_imputeddataset</span><span class="o">/</span><span class="n">ukb32285_imputedindiv</span><span class="o">.</span><span class="n">sample</span>
<span class="n">bgenFile</span><span class="o">=</span><span class="sb">`echo /gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/ukb39554_imputeddataset/ukb_imp_chr{1..22}_v3.bgen`</span>
<span class="n">unrelated_samples</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">unrelated_n307259</span><span class="o">/</span><span class="n">UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620</span>
<span class="n">formatFile_fastgwa</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">fastGWA_template</span><span class="o">.</span><span class="n">yml</span>
<span class="n">formatFile_bolt</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">boltlmm_template</span><span class="o">.</span><span class="n">yml</span>
<span class="n">formatFile_saige</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">saige_template</span><span class="o">.</span><span class="n">yml</span>
<span class="c1"># Container</span>
<span class="n">container_lmm</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="n">container_marp</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">marp</span><span class="o">.</span><span class="n">sif</span>
<span class="c1"># LMM directories</span>
<span class="n">lmm_dir_fastgwa</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span>
<span class="n">lmm_dir_bolt</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span>
<span class="n">lmm_dir_saige</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">SAIGE_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span>
<span class="n">lmm_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">LMM</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_imp</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">lmm_sbatch_bolt</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_imp</span><span class="o">-</span><span class="n">bolt</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">lmm_sbatch_saige</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_imp</span><span class="o">-</span><span class="n">saige</span><span class="o">.</span><span class="n">sbatch</span>
<span class="c1">#Phenotype file</span>
<span class="n">phenoFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis</span>
<span class="c1">## LMM variables </span>
<span class="n">covarFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis</span>
<span class="n">LDscoresFile</span><span class="o">=~/</span><span class="n">software</span><span class="o">/</span><span class="n">BOLT</span><span class="o">-</span><span class="n">LMM_v2</span><span class="o">.</span><span class="mf">3.4</span><span class="o">/</span><span class="n">tables</span><span class="o">/</span><span class="n">LDSCORE</span><span class="o">.</span><span class="mi">1000</span><span class="n">G_EUR</span><span class="o">.</span><span class="n">tab</span><span class="o">.</span><span class="n">gz</span>
<span class="n">geneticMapFile</span><span class="o">=~/</span><span class="n">software</span><span class="o">/</span><span class="n">BOLT</span><span class="o">-</span><span class="n">LMM_v2</span><span class="o">.</span><span class="mf">3.4</span><span class="o">/</span><span class="n">tables</span><span class="o">/</span><span class="n">genetic_map_hg19_withX</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="s2">&quot;cholesterol.f30690_medcorrected&quot;</span>
<span class="n">covarCol</span><span class="o">=</span><span class="s2">&quot;SEX cholesterolmedbyf6153or6177or20003 smoking{1:2} alcohol{1:4}&quot;</span>
<span class="n">covarMaxLevels</span><span class="o">=</span><span class="mi">10</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">AGE</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">20</span>
<span class="n">bgenMinMAF</span><span class="o">=</span><span class="mf">0.001</span>
<span class="n">bgenMinINFO</span><span class="o">=</span><span class="mf">0.8</span>
<span class="n">lmm_job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">ylim</span><span class="o">=</span><span class="mi">0</span>
<span class="c1">### Specific to FastGWA</span>
<span class="n">grmFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv</span><span class="o">.</span><span class="n">grm</span><span class="o">.</span><span class="n">sp</span>
<span class="c1">### Specific to SAIGE</span>
<span class="n">bgenMinMAC</span><span class="o">=</span><span class="mi">4</span>
<span class="n">trait_type</span><span class="o">=</span><span class="n">quantitative</span>
<span class="n">loco</span><span class="o">=</span><span class="n">TRUE</span>
<span class="n">sampleCol</span><span class="o">=</span><span class="n">IID</span>
<span class="c1"># LD clumping directories</span>
<span class="n">clumping_dir</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span>
<span class="n">clumping_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">LD_Clumping</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">clumping_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_HDL_imp_ldclumping</span><span class="o">.</span><span class="n">sbatch</span>
<span class="c1">## LD clumping variables</span>
<span class="c1"># For sumtastsFiles if more than one provide each path</span>
<span class="n">bfile_ref</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span><span class="n">UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv</span><span class="o">.</span><span class="mf">1210.</span><span class="n">ref_geno</span><span class="o">.</span><span class="n">bed</span>
<span class="c1"># Changes dependending upon which traits are analyzed</span>
<span class="n">sumstatsFiles</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">HDL</span><span class="o">/*.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">ld_sample_size</span><span class="o">=</span><span class="mi">1210</span>
<span class="n">clump_field</span><span class="o">=</span><span class="n">P</span>
<span class="n">clump_p1</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">clump_p2</span><span class="o">=</span><span class="mi">1</span>
<span class="n">clump_r2</span><span class="o">=</span><span class="mf">0.2</span>
<span class="n">clump_kb</span><span class="o">=</span><span class="mi">2000</span>
<span class="n">clump_annotate</span><span class="o">=</span><span class="n">BP</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">20</span>
<span class="n">clump_job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="c1">#clumpFile= </span>
<span class="c1">#clumregionFile=</span>
<span class="c1"># Region extraction directories</span>
<span class="n">extract_dir</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">region_extraction</span><span class="o">/</span><span class="n">HDL</span>
<span class="n">extract_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Region_Extraction</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">extract_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_HDL_imp</span><span class="o">-</span><span class="n">region</span><span class="o">.</span><span class="n">sbatch</span>
<span class="c1">## Region extraction variables</span>
<span class="n">region_file</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">LD_clumping</span><span class="o">/</span><span class="n">HDL</span><span class="o">/*.</span><span class="n">clumped_region</span>
<span class="n">geno_path</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">UKBB_bgenfilepath</span><span class="o">.</span><span class="n">txt</span>
<span class="n">sumstats_path</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">HDL</span><span class="o">/*.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">extract_job_size</span><span class="o">=</span><span class="mi">10</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Bolt-LMM-jobs">Bolt-LMM jobs<a class="anchor-link" href="#Bolt-LMM-jobs">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="HDL">HDL<a class="anchor-link" href="#HDL">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_bolt</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">HDL</span>
<span class="n">lmm_sbatch_bolt</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_HDL_imp</span><span class="o">-</span><span class="n">bolt</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="s2">&quot;HDL&quot;</span>
<span class="n">covarCol</span><span class="o">=</span><span class="s2">&quot;SEX cholesterolmedbyf6153or6177or20003 smoking{1:2} alcohol{1:4}&quot;</span>
<span class="n">qcovarCol</span><span class="o">=</span><span class="n">AGE</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;boltlmm</span>
<span class="s2">    --cwd $lmm_dir_bolt </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_bolt </span>
<span class="s2">    --covarFile $covarFile </span>
<span class="s2">    --LDscoresFile $LDscoresFile </span>
<span class="s2">    --geneticMapFile $geneticMapFile </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_bolt</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-10-13_HDL_imp-bolt.sbatch</span>
INFO: Workflow farnam (ID=39b24b87b30a8a06) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Cholesterol">Cholesterol<a class="anchor-link" href="#Cholesterol">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_bolt</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">cholesterol</span>
<span class="n">lmm_sbatch_bolt</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_cholesterol_imp</span><span class="o">-</span><span class="n">bolt</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="s2">&quot;cholesterol.f30690_medcorrected&quot;</span>
<span class="n">covarCol</span><span class="o">=</span><span class="s2">&quot;SEX cholesterolmedbyf6153or6177or20003 smoking{1:2} alcohol{1:4}&quot;</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">AGE</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;boltlmm</span>
<span class="s2">    --cwd $lmm_dir_bolt </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_bolt </span>
<span class="s2">    --covarFile $covarFile </span>
<span class="s2">    --LDscoresFile $LDscoresFile </span>
<span class="s2">    --geneticMapFile $geneticMapFile </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_bolt</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-12-15_cholesterol_imp-bolt.sbatch</span>
INFO: Workflow farnam (ID=b178fdaaac231b03) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="LDL">LDL<a class="anchor-link" href="#LDL">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_bolt</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">LDL</span>
<span class="n">lmm_sbatch_bolt</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_LDL_imp</span><span class="o">-</span><span class="n">bolt</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="s2">&quot;LDL_medcorrected&quot;</span>
<span class="n">covarCol</span><span class="o">=</span><span class="s2">&quot;SEX cholesterolmedbyf6153or6177or20003 smoking{1:2} alcohol{1:4}&quot;</span>
<span class="n">qcovarCol</span><span class="o">=</span><span class="n">AGE</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;boltlmm</span>
<span class="s2">    --cwd $lmm_dir_bolt </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_bolt </span>
<span class="s2">    --covarFile $covarFile </span>
<span class="s2">    --LDscoresFile $LDscoresFile </span>
<span class="s2">    --geneticMapFile $geneticMapFile </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_bolt</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-12-15_LDL_imp-bolt.sbatch</span>
INFO: Workflow farnam (ID=26fd06bd806de5ef) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Triglycerides">Triglycerides<a class="anchor-link" href="#Triglycerides">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_bolt</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">triglycerides</span>
<span class="n">lmm_sbatch_bolt</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_triglycerides_imp</span><span class="o">-</span><span class="n">bolt</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="s2">&quot;triglycerides&quot;</span>
<span class="n">covarCol</span><span class="o">=</span><span class="s2">&quot;SEX cholesterolmedbyf6153or6177or20003 smoking{1:2} alcohol{1:4}&quot;</span>
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">AGE</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;boltlmm</span>
<span class="s2">    --cwd $lmm_dir_bolt </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_bolt </span>
<span class="s2">    --covarFile $covarFile </span>
<span class="s2">    --LDscoresFile $LDscoresFile </span>
<span class="s2">    --geneticMapFile $geneticMapFile </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_bolt</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="FastGWA-jobs">FastGWA jobs<a class="anchor-link" href="#FastGWA-jobs">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="HDL">HDL<a class="anchor-link" href="#HDL">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">HDL</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_HDL_imp</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis</span>
<span class="n">covarFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="s2">&quot;HDL&quot;</span>
<span class="n">covarCol</span><span class="o">=</span><span class="s2">&quot;SEX cholesterolmedbyf6153or6177or20003 smoking1 smoking2 alcohol1 alcohol2 alcohol3 alcohol4&quot;</span> 
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">AGE</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-09-15_HDL_imp-fastgwa.sbatch</span>
INFO: Workflow farnam (ID=18b620039e084b7d) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Cholesterol">Cholesterol<a class="anchor-link" href="#Cholesterol">&#182;</a></h2><p>Ran please finish creating the script for fastGWA for the remaining traits</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">cholesterol</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_cholesterol_imp</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis</span>
<span class="n">covarFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="s2">&quot;cholesterol.f30690_medcorrected&quot;</span>
<span class="n">covarCol</span><span class="o">=</span><span class="s2">&quot;SEX cholesterolmedbyf6153or6177or20003 smoking1 smoking2 alcohol1 alcohol2 alcohol3 alcohol4&quot;</span> 
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">AGE</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-09-10_cholesterol_imp-fastgwa.sbatch</span>
INFO: Workflow farnam (ID=7cf2c7310ac1d222) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="LDL">LDL<a class="anchor-link" href="#LDL">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">LDL</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_LDL_imp</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis</span>
<span class="n">covarFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="s2">&quot;LDL_medcorrected&quot;</span>
<span class="n">covarCol</span><span class="o">=</span><span class="s2">&quot;SEX cholesterolmedbyf6153or6177or20003 smoking1 smoking2 alcohol1 alcohol2 alcohol3 alcohol4&quot;</span> 
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">AGE</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-09-10_LDL_imp-fastgwa.sbatch</span>
INFO: Workflow farnam (ID=74f114d1a8a102d3) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Triglycerides">Triglycerides<a class="anchor-link" href="#Triglycerides">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[4]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">lmm_dir_fastgwa</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">triglycerides</span>
<span class="n">lmm_sbatch_fastgwa</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_triglycerides_imp</span><span class="o">-</span><span class="n">fastgwa</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis</span>
<span class="n">covarFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="s2">&quot;triglycerides&quot;</span>
<span class="n">covarCol</span><span class="o">=</span><span class="s2">&quot;SEX cholesterolmedbyf6153or6177or20003 smoking1 smoking2 alcohol1 alcohol2 alcohol3 alcohol4&quot;</span> 
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">AGE</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;fastGWA</span>
<span class="s2">    --cwd $lmm_dir_fastgwa </span>
<span class="s2">    --bfile $bfile </span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile </span>
<span class="s2">    --phenoFile $phenoFile </span>
<span class="s2">    --formatFile $formatFile_fastgwa </span>
<span class="s2">    --covarFile $covarFile  </span>
<span class="s2">    --phenoCol $phenoCol </span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --covarMaxLevels $covarMaxLevels </span>
<span class="s2">    --qCovarCol $qCovarCol </span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --bgenMinMAF $bgenMinMAF </span>
<span class="s2">    --bgenMinINFO $bgenMinINFO </span>
<span class="s2">    --job_size $lmm_job_size</span>
<span class="s2">    --grmFile $grmFile</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_fastgwa</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-09-10_triglycerides_imp-fastgwa.sbatch</span>
INFO: Workflow farnam (ID=e14e0004ea8849e5) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="REGENIE-int-blood">REGENIE int-blood<a class="anchor-link" href="#REGENIE-int-blood">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[8]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">bfile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_Caucasians_phenotypeindepqc120319_updated082020removedwithdrawnindiv</span><span class="o">.</span><span class="n">bed</span>
<span class="n">sampleFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb39554_imputeddataset</span><span class="o">/</span><span class="n">ukb32285_imputedindiv</span><span class="o">.</span><span class="n">sample</span>
<span class="n">bgenFile</span><span class="o">=</span><span class="sb">`echo /gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/ukb39554_imputeddataset/ukb_imp_chr{1..22}_v3.bgen`</span>
<span class="n">unrelated_samples</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">unrelated_n307259</span><span class="o">/</span><span class="n">UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620</span>
<span class="n">formatFile_regenie</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">regenie_template</span><span class="o">.</span><span class="n">yml</span>
<span class="c1">#Container</span>
<span class="n">container_lmm</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="n">container_marp</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">marp</span><span class="o">.</span><span class="n">sif</span>
<span class="c1"># Directories</span>
<span class="n">lmm_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">LMM</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">lmm_dir_regenie</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">int_ApoA_ApoB_Creatinine_Hba1c_eGFR_UACR</span>
<span class="n">lmm_sbatch_regenie</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+%</span><span class="n">Y</span><span class="o">-%</span><span class="n">m</span><span class="o">-%</span><span class="n">d</span><span class="p">)</span><span class="n">_lipids_imp</span><span class="o">-</span><span class="n">regenie</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKB_CAUC_lipidsforanalysis_apolipoproteinAandB</span><span class="p">,</span><span class="n">Hba1c_continuousandcategorical</span><span class="p">,</span><span class="n">egfrbyCKDEPI</span><span class="p">,</span><span class="n">serumcreatinine</span><span class="p">,</span><span class="n">UACR_inverseranknorm_110320</span>
<span class="n">covarFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKB_CAUC_lipidsforanalysis_apolipoproteinAandB</span><span class="p">,</span><span class="n">Hba1c_continuousandcategorical</span><span class="p">,</span><span class="n">egfrbyCKDEPI</span><span class="p">,</span><span class="n">serumcreatinine</span><span class="p">,</span><span class="n">UACR_inverseranknorm_110320</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="s2">&quot;ApoA_int ApoB_int Creatinine_int Hba1c_int eGFR_int UACR_int&quot;</span>
<span class="n">covarCol</span><span class="o">=</span><span class="s2">&quot;SEX alcohol1 alcohol2 alcohol3 alcohol4&quot;</span> 
<span class="n">qCovarCol</span><span class="o">=</span><span class="s2">&quot;AGE Packyrs_smoking&quot;</span>
<span class="n">maf_filter</span><span class="o">=</span><span class="mf">0.001</span>
<span class="n">geno_filter</span><span class="o">=</span><span class="mf">0.0</span>
<span class="n">hwe_filter</span><span class="o">=</span><span class="mf">0.0</span>
<span class="n">mind_filter</span><span class="o">=</span><span class="mf">0.0</span>
<span class="n">trait</span><span class="o">=</span>
<span class="n">minMAC</span><span class="o">=</span><span class="mi">4</span>
<span class="n">bsize</span><span class="o">=</span><span class="mi">1000</span>
<span class="n">ylim</span><span class="o">=</span><span class="mi">0</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">22</span>
<span class="n">reverse_log_p</span><span class="o">=</span><span class="bp">True</span>
<span class="n">lowmem_prefix</span><span class="o">=~/</span><span class="n">scratch60</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;regenie</span>
<span class="s2">    --cwd $lmm_dir_regenie</span>
<span class="s2">    --bfile $bfile</span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --bgenFile $bgenFile</span>
<span class="s2">    --phenoFile $phenoFile</span>
<span class="s2">    --formatFile $formatFile_regenie</span>
<span class="s2">    --covarFile $covarFile</span>
<span class="s2">    --phenoCol $phenoCol</span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --qCovarCol $qCovarCol</span>
<span class="s2">    --maf_filter $maf_filter</span>
<span class="s2">    --geno_filter $geno_filter</span>
<span class="s2">    --hwe_filter $hwe_filter</span>
<span class="s2">    --mind_filter $mind_filter</span>
<span class="s2">    --trait $trait</span>
<span class="s2">    --minMAC $minMAC</span>
<span class="s2">    --bsize $bsize</span>
<span class="s2">    --numThreads $numThreads</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --p-filter 1</span>
<span class="s2">    --reverse_log_p $reverse_log_p</span>
<span class="s2">    --lowmem_prefix $lowmem_prefix</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>
    
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
        <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
        <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
        <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_regenie</span> \
        <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-11-18_lipids_imp-regenie.sbatch</span>
INFO: Workflow farnam (ID=24d29053e6048d83) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="REGENIE-int-lipids">REGENIE int-lipids<a class="anchor-link" href="#REGENIE-int-lipids">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">bfile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_Caucasians_phenotypeindepqc120319_updated082020removedwithdrawnindiv</span><span class="o">.</span><span class="n">bed</span>
<span class="n">sampleFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb39554_imputeddataset</span><span class="o">/</span><span class="n">ukb32285_imputedindiv</span><span class="o">.</span><span class="n">sample</span>
<span class="n">genoFile</span><span class="o">=</span><span class="sb">`echo /gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/ukb39554_imputeddataset/ukb_imp_chr{1..22}_v3.bgen`</span>
<span class="n">unrelated_samples</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">unrelated_n307259</span><span class="o">/</span><span class="n">UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620</span>
<span class="n">formatFile_regenie</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">regenie_template</span><span class="o">.</span><span class="n">yml</span>
<span class="c1">#Container</span>
<span class="n">container_lmm</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="n">container_marp</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">marp</span><span class="o">.</span><span class="n">sif</span>
<span class="c1"># Directories</span>
<span class="n">lmm_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">LMM</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">lmm_dir_regenie</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">//</span><span class="n">REGENIE_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">int_HDL_LDL_Triglycerides_Cholesterol</span>
<span class="n">lmm_sbatch_regenie</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+%</span><span class="n">Y</span><span class="o">-%</span><span class="n">m</span><span class="o">-%</span><span class="n">d</span><span class="p">)</span><span class="n">_int_lipids_imp</span><span class="o">-</span><span class="n">regenie</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">phenoFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKBCauc_cholesterolandbloodpressurefields_inverseranknorm</span>
<span class="n">covarFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span><span class="o">/</span><span class="n">UKBCauc_cholesterolandbloodpressurefields_inverseranknorm</span>
<span class="n">phenoCol</span><span class="o">=</span><span class="s2">&quot;HDL_inverseranknorm LDL_medcorrected_inverseranknorm triglycerides_inverseranknorm cholesterol.f30690_medcorrected_inverseranknorm&quot;</span>
<span class="n">covarCol</span><span class="o">=</span><span class="s2">&quot;SEX cholesterolmedbyf6153or6177or20003 smoking_dummy1 smoking_dummy2 alcohol1 alcohol2 alcohol3 alcohol4&quot;</span> 
<span class="n">qCovarCol</span><span class="o">=</span><span class="n">AGE</span>
<span class="n">maf_filter</span><span class="o">=</span><span class="mf">0.001</span>
<span class="n">geno_filter</span><span class="o">=</span><span class="mf">0.0</span>
<span class="n">hwe_filter</span><span class="o">=</span><span class="mf">0.0</span>
<span class="n">mind_filter</span><span class="o">=</span><span class="mf">0.0</span>
<span class="n">trait</span><span class="o">=</span>
<span class="n">minMAC</span><span class="o">=</span><span class="mi">4</span>
<span class="n">bsize</span><span class="o">=</span><span class="mi">1000</span>
<span class="n">ylim</span><span class="o">=</span><span class="mi">0</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">22</span>
<span class="n">reverse_log_p</span><span class="o">=</span><span class="bp">True</span>
<span class="n">lowmem_prefix</span><span class="o">=~/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">regenie_lipids</span>

<span class="n">lmm_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;regenie</span>
<span class="s2">    --cwd $lmm_dir_regenie</span>
<span class="s2">    --bfile $bfile</span>
<span class="s2">    --sampleFile $sampleFile</span>
<span class="s2">    --genoFile $genoFile</span>
<span class="s2">    --phenoFile $phenoFile</span>
<span class="s2">    --formatFile $formatFile_regenie</span>
<span class="s2">    --covarFile $covarFile</span>
<span class="s2">    --phenoCol $phenoCol</span>
<span class="s2">    --covarCol $covarCol </span>
<span class="s2">    --qCovarCol $qCovarCol</span>
<span class="s2">    --maf_filter $maf_filter</span>
<span class="s2">    --geno_filter $geno_filter</span>
<span class="s2">    --hwe_filter $hwe_filter</span>
<span class="s2">    --mind_filter $mind_filter</span>
<span class="s2">    --trait $trait</span>
<span class="s2">    --minMAC $minMAC</span>
<span class="s2">    --bsize $bsize</span>
<span class="s2">    --numThreads $numThreads</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --ylim $ylim</span>
<span class="s2">    --p-filter 1</span>
<span class="s2">    --reverse_log_p $reverse_log_p</span>
<span class="s2">    --lowmem_prefix $lowmem_prefix</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">    --container_marp $container_marp</span>
<span class="s2">&quot;&quot;&quot;</span>
    
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
        <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
        <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">lmm_sos</span> \
        <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">lmm_sbatch_regenie</span> \
        <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$lmm_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-01-08_int_lipids_imp-regenie.sbatch</span>
INFO: Workflow farnam (ID=ba16dcb1d7a543a5) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Hudson-plots:-method-comparisson">Hudson plots: method comparisson<a class="anchor-link" href="#Hudson-plots:-method-comparisson">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">hudson_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Hudson_plot</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">hudson_dir</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">hudson_plots</span><span class="o">/</span><span class="n">pleiotropy</span><span class="o">/</span><span class="n">HDL_fastGWA_boltlmm</span>
<span class="n">hudson_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_HDL_fastGWA_bolt_hudson</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstats_1</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">HDL</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis_HDL</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">sumstats_2</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">HDL</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis_HDL</span><span class="o">.</span><span class="n">boltlmm</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">toptitle</span><span class="o">=</span><span class="s2">&quot;HDL_fastGWA&quot;</span>
<span class="n">bottomtitle</span><span class="o">=</span><span class="s2">&quot;HDL_boltlmm&quot;</span>
<span class="n">highlight_p_top</span><span class="o">=</span><span class="mi">0</span>
<span class="n">highlight_p_bottom</span><span class="o">=</span><span class="mi">0</span>
<span class="n">pval_filter</span><span class="o">=</span><span class="mf">5e-08</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_lmm</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="c1">#highlight_snp=</span>
<span class="n">annotate_snp</span><span class="o">=</span><span class="mi">0</span>
<span class="n">phenocol1</span><span class="o">=</span><span class="s1">&#39;HDL&#39;</span>
<span class="n">phenocol2</span><span class="o">=</span><span class="s1">&#39;HDL&#39;</span>

<span class="n">hudson_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;hudson</span>
<span class="s2">    --cwd $hudson_dir</span>
<span class="s2">    --sumstats_1 $sumstats_1</span>
<span class="s2">    --sumstats_2 $sumstats_2</span>
<span class="s2">    --toptitle $toptitle</span>
<span class="s2">    --bottomtitle $bottomtitle</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --highlight_p_top $highlight_p_top</span>
<span class="s2">    --highlight_p_bottom $highlight_p_bottom</span>
<span class="s2">    --pval_filter $pval_filter</span>
<span class="s2">    --annotate_snp $annotate_snp</span>
<span class="s2">    --phenocol1 $phenocol1</span>
<span class="s2">    --phenocol2 $phenocol2</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">hudson_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">hudson_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$hudson_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-12-16_HDL_fastGWA_bolt_hudson.sbatch</span>
INFO: Workflow farnam (ID=8e27a828cc9b9c9b) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="LDL_fastGWA---LDL_boltlmm">LDL_fastGWA - LDL_boltlmm<a class="anchor-link" href="#LDL_fastGWA---LDL_boltlmm">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[7]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">hudson_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Hudson_plot</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">hudson_dir</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">hudson_plots</span><span class="o">/</span><span class="n">pleiotropy</span><span class="o">/</span><span class="n">LDL_fast_bolt</span>
<span class="n">hudson_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_LDLfast_vs_LDLbolt_hudson</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstats_1</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">LDL</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis_LDL_medcorrected</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">sumstats_2</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">LDL</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis_LDL_medcorrected</span><span class="o">.</span><span class="n">boltlmm</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">toptitle</span><span class="o">=</span><span class="s2">&quot;LDLfast_LDLbolt&quot;</span>
<span class="n">bottomtitle</span><span class="o">=</span><span class="s2">&quot;LDLfast_LDLbolt&quot;</span>
<span class="n">highlight_p_top</span><span class="o">=</span><span class="mi">0</span>
<span class="n">highlight_p_bottom</span><span class="o">=</span><span class="mi">0</span>
<span class="n">pval_filter</span><span class="o">=</span><span class="mf">5e-06</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_lmm</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="c1">#highlight_snp=</span>
<span class="n">annotate_snp</span><span class="o">=</span><span class="mi">0</span>
<span class="n">phenocol1</span><span class="o">=</span><span class="s1">&#39;LDLfastGWA&#39;</span>
<span class="n">phenocol2</span><span class="o">=</span><span class="s1">&#39;LDLbolt&#39;</span>

<span class="n">hudson_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;hudson</span>
<span class="s2">    --cwd $hudson_dir</span>
<span class="s2">    --sumstats_1 $sumstats_1</span>
<span class="s2">    --sumstats_2 $sumstats_2</span>
<span class="s2">    --toptitle $toptitle</span>
<span class="s2">    --bottomtitle $bottomtitle</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --highlight_p_top $highlight_p_top</span>
<span class="s2">    --highlight_p_bottom $highlight_p_bottom</span>
<span class="s2">    --pval_filter $pval_filter</span>
<span class="s2">    --annotate_snp $annotate_snp</span>
<span class="s2">    --phenocol1 $phenocol1</span>
<span class="s2">    --phenocol2 $phenocol2</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">hudson_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">hudson_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$hudson_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-12-03_LDLfast_vs_LDLbolt_hudson.sbatch</span>
INFO: Workflow farnam (ID=630e79f954f1d0f0) is executed successfully with 1 completed step.
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Triglycerides">Triglycerides<a class="anchor-link" href="#Triglycerides">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[8]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">hudson_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Hudson_plot</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">hudson_dir</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">hudson_plots</span><span class="o">/</span><span class="n">pleiotropy</span><span class="o">/</span><span class="n">TG_fast_bolt</span>
<span class="n">hudson_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_LDLfast_vs_LDLbolt_hudson</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstats_1</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">triglycerides</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis_triglycerides</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">sumstats_2</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">triglycerides</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis_triglycerides</span><span class="o">.</span><span class="n">boltlmm</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">toptitle</span><span class="o">=</span><span class="s2">&quot;TGfast_TGbolt&quot;</span>
<span class="n">bottomtitle</span><span class="o">=</span><span class="s2">&quot;TGfast_TGbolt&quot;</span>
<span class="n">highlight_p_top</span><span class="o">=</span><span class="mi">0</span>
<span class="n">highlight_p_bottom</span><span class="o">=</span><span class="mi">0</span>
<span class="n">pval_filter</span><span class="o">=</span><span class="mf">5e-06</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_lmm</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="c1">#highlight_snp=</span>
<span class="n">annotate_snp</span><span class="o">=</span><span class="mi">0</span>
<span class="n">phenocol1</span><span class="o">=</span><span class="s1">&#39;TGfastGWA&#39;</span>
<span class="n">phenocol2</span><span class="o">=</span><span class="s1">&#39;TGbolt&#39;</span>

<span class="n">hudson_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;hudson</span>
<span class="s2">    --cwd $hudson_dir</span>
<span class="s2">    --sumstats_1 $sumstats_1</span>
<span class="s2">    --sumstats_2 $sumstats_2</span>
<span class="s2">    --toptitle $toptitle</span>
<span class="s2">    --bottomtitle $bottomtitle</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --highlight_p_top $highlight_p_top</span>
<span class="s2">    --highlight_p_bottom $highlight_p_bottom</span>
<span class="s2">    --pval_filter $pval_filter</span>
<span class="s2">    --annotate_snp $annotate_snp</span>
<span class="s2">    --phenocol1 $phenocol1</span>
<span class="s2">    --phenocol2 $phenocol2</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">hudson_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">hudson_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$hudson_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-12-03_LDLfast_vs_LDLbolt_hudson.sbatch</span>
INFO: Workflow farnam (ID=272f16dbd8df22ed) is executed successfully with 1 completed step.
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Cholesterol">Cholesterol<a class="anchor-link" href="#Cholesterol">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[9]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">hudson_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Hudson_plot</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">hudson_dir</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">hudson_plots</span><span class="o">/</span><span class="n">pleiotropy</span><span class="o">/</span><span class="n">Cholesterol_fast_bolt</span>
<span class="n">hudson_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_LDLfast_vs_LDLbolt_hudson</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstats_1</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">cholesterol</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis_cholesterol</span><span class="o">.</span><span class="n">f30690_medcorrected</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">sumstats_2</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">cholesterol</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis_cholesterol</span><span class="o">.</span><span class="n">f30690_medcorrected</span><span class="o">.</span><span class="n">boltlmm</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">toptitle</span><span class="o">=</span><span class="s2">&quot;Cholesterolfast_Cholesterolbolt&quot;</span>
<span class="n">bottomtitle</span><span class="o">=</span><span class="s2">&quot;Cholesterolfast_Cholesterolbolt&quot;</span>
<span class="n">highlight_p_top</span><span class="o">=</span><span class="mi">0</span>
<span class="n">highlight_p_bottom</span><span class="o">=</span><span class="mi">0</span>
<span class="n">pval_filter</span><span class="o">=</span><span class="mf">5e-06</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_lmm</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="c1">#highlight_snp=</span>
<span class="n">annotate_snp</span><span class="o">=</span><span class="mi">0</span>
<span class="n">phenocol1</span><span class="o">=</span><span class="s1">&#39;CholesterolfastGWA&#39;</span>
<span class="n">phenocol2</span><span class="o">=</span><span class="s1">&#39;Cholesterolbolt&#39;</span>

<span class="n">hudson_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;hudson</span>
<span class="s2">    --cwd $hudson_dir</span>
<span class="s2">    --sumstats_1 $sumstats_1</span>
<span class="s2">    --sumstats_2 $sumstats_2</span>
<span class="s2">    --toptitle $toptitle</span>
<span class="s2">    --bottomtitle $bottomtitle</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --highlight_p_top $highlight_p_top</span>
<span class="s2">    --highlight_p_bottom $highlight_p_bottom</span>
<span class="s2">    --pval_filter $pval_filter</span>
<span class="s2">    --annotate_snp $annotate_snp</span>
<span class="s2">    --phenocol1 $phenocol1</span>
<span class="s2">    --phenocol2 $phenocol2</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">hudson_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">hudson_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$hudson_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-12-03_LDLfast_vs_LDLbolt_hudson.sbatch</span>
INFO: Workflow farnam (ID=36d3ad1a0f04abe6) is executed successfully with 1 completed step.
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="HDL">HDL<a class="anchor-link" href="#HDL">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[10]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">hudson_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Hudson_plot</span><span class="o">.</span><span class="n">ipynb</span>
<span class="n">hudson_dir</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">hudson_plots</span><span class="o">/</span><span class="n">pleiotropy</span><span class="o">/</span><span class="n">HDL_fast_bolt</span>
<span class="n">hudson_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_LDLfast_vs_LDLbolt_hudson</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">sumstats_1</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">FastGWA_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">HDL</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis_HDL</span><span class="o">.</span><span class="n">fastGWA</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">sumstats_2</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">HDL</span><span class="o">/</span><span class="n">UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis_HDL</span><span class="o">.</span><span class="n">boltlmm</span><span class="o">.</span><span class="n">snp_stats</span><span class="o">.</span><span class="n">gz</span>
<span class="n">toptitle</span><span class="o">=</span><span class="s2">&quot;HDLfast_HDLbolt&quot;</span>
<span class="n">bottomtitle</span><span class="o">=</span><span class="s2">&quot;HDLfast_HDLbolt&quot;</span>
<span class="n">highlight_p_top</span><span class="o">=</span><span class="mi">0</span>
<span class="n">highlight_p_bottom</span><span class="o">=</span><span class="mi">0</span>
<span class="n">pval_filter</span><span class="o">=</span><span class="mf">5e-06</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="n">container_lmm</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="c1">#highlight_snp=</span>
<span class="n">annotate_snp</span><span class="o">=</span><span class="mi">0</span>
<span class="n">phenocol1</span><span class="o">=</span><span class="s1">&#39;HDLfastGWA&#39;</span>
<span class="n">phenocol2</span><span class="o">=</span><span class="s1">&#39;HDLbolt&#39;</span>

<span class="n">hudson_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;hudson</span>
<span class="s2">    --cwd $hudson_dir</span>
<span class="s2">    --sumstats_1 $sumstats_1</span>
<span class="s2">    --sumstats_2 $sumstats_2</span>
<span class="s2">    --toptitle $toptitle</span>
<span class="s2">    --bottomtitle $bottomtitle</span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --highlight_p_top $highlight_p_top</span>
<span class="s2">    --highlight_p_bottom $highlight_p_bottom</span>
<span class="s2">    --pval_filter $pval_filter</span>
<span class="s2">    --annotate_snp $annotate_snp</span>
<span class="s2">    --phenocol1 $phenocol1</span>
<span class="s2">    --phenocol2 $phenocol2</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">hudson_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">hudson_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$hudson_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2020-12-03_LDLfast_vs_LDLbolt_hudson.sbatch</span>
INFO: Workflow farnam (ID=21adef2b434d78ff) is executed successfully with 1 completed step.
</pre>
</div>
</div>

</div>
</div>

</div>
<hr>
&copy 2020 Diana Cornejo at Center for Statistical Genetics, Sergievsky Center, Columbia University VP&S
<p><small>Exported from <a href="http://github.com/dianacornejo/pleiotropy_UKB/blob/804a49e20903d8a6a7eafcb7472453c5ea2dcde8/analysis/cluster_scripts/Lipids_scripts.ipynb"><code>analysis/cluster_scripts/Lipids_scripts.ipynb</code></a> committed by dc2325 on Fri Feb 19 22:22:36 2021 <a href="http://github.com/dianacornejo/pleiotropy_UKB/commit/804a49e20903d8a6a7eafcb7472453c5ea2dcde8">revision 1, 804a49e</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
