<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=analysisDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=analysisArray;
            var docs_map=analysisArrayMap;
            var pos=analysisArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Pleiotropy analysis in UKBB</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Pleiotropy analysis in UKBB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../workflow.html">Workflow</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/dianacornejo/pleiotropy_UKB"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Scripts-for-diverse-analyses">Scripts for diverse analyses<a class="anchor-link" href="#Scripts-for-diverse-analyses">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>This notebook applies the <code>Get_Job_Script.ipynb</code> to automatically generate the sbatch scripts to run in Yale's cluster.</p>
<p>Here the scripts generated are to run:</p>
<ol>
<li>PCA analysis</li>
<li>Detect missingness in plink files</li>
<li>Extract SNPs/Individuals using Plink</li>
<li>Run regenie burden MWE</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="File-paths-on-Yale-cluster">File paths on Yale cluster<a class="anchor-link" href="#File-paths-on-Yale-cluster">&#182;</a></h2><ul>
<li>Genotype files exome data:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/ukb28374_exomedata/exome_data_OCT2020</code></li>
<li>Genotype files in PLINK format:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/pleiotropy_geneticfiles/UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv</code></li>
<li>Genotype files in bgen format:
<code>SAY/dbgapstg/scratch/UKBiobank/genotype_files/ukb39554_imputeddataset/</code></li>
<li>Summary stats for imputed variants BOLT-LMM:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/results/BOLTLMM_results/results_imputed_data</code></li>
<li>Summary stats for inputed variants FastGWA:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/results/FastGWA_results/results_imputed_data</code></li>
<li>Phenotype files:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/phenotype_files/pleiotropy_R01/phenotypesforanalysis/UKB_Caucasiansubset_cholesterolfields_adjbymedstatus_062420_foranalysis</code></li>
<li>Relationship file:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/pleiotropy_geneticfiles/unrelated_n307259/UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620</code></li>
<li>Other traits to be analyzed:
<code>/gpfs/gibbs/pi/dewan/data/UKBiobank/phenotype_files/pleiotropy_R01/phenotypesforanalysis/UKB_CAUC_lipidsforanalysis_apolipoproteinAandB,Hba1c_continuousandcategorical,egfrbyCKDEPI,serumcreatinine,UACR_inverseranknorm_110320</code></li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Common variables</span>
<span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">UKBB_PATH</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span>
<span class="n">USER_PATH</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">project</span>
<span class="c1">#pca_dir=/gpfs/gibbs/pi/dewan/data/UKBiobank/results/pca_exomes</span>
<span class="n">pca_dir</span><span class="o">=~/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">pca</span>
<span class="n">famFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">fam</span>
<span class="c1"># Use a subset of the exomed markers</span>
<span class="c1">#genoFile=`echo $UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_c{1..22}_b0_v1.bed`</span>
<span class="c1">#Use the original bed files for the genotype array on regenie step1</span>
<span class="n">genoFile</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_originalgenotypefilesdownloaded083019</span><span class="o">/</span><span class="n">UKB_genotypedatadownloaded083019</span><span class="o">.</span><span class="n">bed</span>

<span class="n">phenoFile</span><span class="o">=~/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">pca</span><span class="o">/</span><span class="n">cache</span><span class="o">/</span><span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">non_white_white_outliers_11971ind</span><span class="o">.</span><span class="n">pheno</span>
<span class="n">database</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">ukb42495_updatedJune2020</span><span class="o">/</span><span class="n">ukb42495</span><span class="o">.</span><span class="n">tab</span>
<span class="n">ethnia_prefix</span><span class="o">=</span><span class="s1">&#39;non_white_white_outliers_11971ind&#39;</span>
<span class="n">select_ethnia</span><span class="o">=</span><span class="mi">2001</span><span class="p">,</span><span class="mi">3001</span><span class="p">,</span><span class="mi">4001</span><span class="p">,</span><span class="mi">2002</span><span class="p">,</span><span class="mi">3002</span><span class="p">,</span><span class="mi">4002</span><span class="p">,</span><span class="mi">2003</span><span class="p">,</span><span class="mi">3003</span><span class="p">,</span><span class="mi">4003</span><span class="p">,</span><span class="mi">2004</span><span class="p">,</span><span class="mi">3004</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="o">-</span><span class="mi">1</span>
<span class="c1"># Container</span>
<span class="n">container_lmm</span><span class="o">=</span><span class="err">$</span><span class="n">UKBB_PATH</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="c1"># Pipeline</span>
<span class="n">pca_sos</span><span class="o">=</span><span class="err">$</span><span class="n">USER_PATH</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">workflow</span><span class="o">/</span><span class="n">PCA</span><span class="o">.</span><span class="n">ipynb</span>
<span class="c1"># Name of bash script</span>
<span class="c1">#pca_sbatch=../output/$(date +&quot;%Y-%m-%d&quot;)_pca_non_white.sbatch</span>
<span class="n">pca_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_flashpca_non_white_whiteoutliers</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">1</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="c1">#PCA variables</span>
<span class="n">k</span><span class="o">=</span><span class="mi">10</span>
<span class="n">maxiter</span><span class="o">=</span><span class="mi">0</span>
<span class="n">topk</span><span class="o">=</span><span class="mi">10</span>
<span class="n">sigma</span><span class="o">=</span><span class="mi">6</span>
<span class="n">window</span><span class="o">=</span><span class="mi">50</span>
<span class="n">shift</span><span class="o">=</span><span class="mi">5</span>
<span class="n">r2</span><span class="o">=</span><span class="mf">0.5</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="PCA-jobs">PCA jobs<a class="anchor-link" href="#PCA-jobs">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">pca_dir</span><span class="o">=~/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">pca</span><span class="o">/</span><span class="n">african_asian</span><span class="o">/</span><span class="n">genotype_array</span>
<span class="n">ethnia_prefix</span><span class="o">=</span><span class="s1">&#39;african_asian_8308ind&#39;</span>
<span class="n">phenoFile</span><span class="o">=~/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">pca</span><span class="o">/</span><span class="n">african_asian</span><span class="o">/</span><span class="n">genotype_array</span><span class="o">/</span><span class="n">cache</span><span class="o">/</span><span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">african_asian_8308ind</span><span class="o">.</span><span class="n">pheno</span>
<span class="n">pca_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_flashpca_african_asian_genoarray</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">trait_name</span><span class="o">=</span><span class="n">ethnicity</span>

<span class="n">pca_args</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;flashpca</span>
<span class="s2">    --cwd $pca_dir </span>
<span class="s2">    --genoFile $genoFile</span>
<span class="s2">    --famFile $famFile</span>
<span class="s2">    --database $database</span>
<span class="s2">    --ethnia_prefix $ethnia_prefix</span>
<span class="s2">    --select_ethnia $select_ethnia</span>
<span class="s2">    --phenoFile $phenoFile</span>
<span class="s2">    --k $k</span>
<span class="s2">    --maxiter $maxiter</span>
<span class="s2">    --topk $topk</span>
<span class="s2">    --sigma $sigma</span>
<span class="s2">    --window $window</span>
<span class="s2">    --shift $shift</span>
<span class="s2">    --r2 $r2</span>
<span class="s2">    --trait_name $trait_name</span>
<span class="s2">    --numThreads $numThreads </span>
<span class="s2">    --job_size $job_size</span>
<span class="s2">    --container_lmm $container_lmm</span>
<span class="s2">&quot;&quot;&quot;</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">Get_Job_Script</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">farnam</span> \
    <span class="o">--</span><span class="n">template</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">tpl_file</span> \
    <span class="o">--</span><span class="n">workflow</span><span class="o">-</span><span class="nb">file</span> <span class="err">$</span><span class="n">pca_sos</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">script</span> <span class="err">$</span><span class="n">pca_sbatch</span> \
    <span class="o">--</span><span class="n">args</span> <span class="s2">&quot;$pca_args&quot;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">farnam</span>: Configuration for Yale `farnam` cluster
INFO: <span class="ansi-green-fg">farnam</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">farnam</span> output:   <span class="ansi-green-fg">../output/2021-02-19_flashpca_african_asian_genoarray.sbatch</span>
INFO: Workflow farnam (ID=w8966d90f06c312c8) is executed successfully with 1 completed step.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">pca_dir</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">pca_exomes</span>
<span class="n">famFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">fam</span>
<span class="n">bedfiles</span><span class="o">=</span><span class="sb">`echo /gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_c{1..22}_b0_v1.bed`</span>
<span class="n">bimfiles</span><span class="o">=</span><span class="sb">`echo /gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/ukb28374_exomedata/exome_data_OCT2020/UKBexomeOQFE_chr{1..22}.bim`</span>
<span class="n">database</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">ukb42495_updatedJune2020</span><span class="o">/</span><span class="n">ukb42495</span><span class="o">.</span><span class="n">tab</span>
<span class="c1"># Container</span>
<span class="n">container_lmm</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="c1"># Pipeline</span>
<span class="n">pca_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">PCA</span><span class="o">.</span><span class="n">ipynb</span>
<span class="c1"># Name of bash script</span>
<span class="n">pca_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_pca_white</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">1</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>
<span class="c1">#PCA variables</span>
<span class="n">k</span><span class="o">=</span><span class="mi">10</span>
<span class="n">maxiter</span><span class="o">=</span><span class="mi">5</span>
<span class="n">topk</span><span class="o">=</span><span class="mi">10</span>
<span class="n">sigma</span><span class="o">=</span><span class="mi">6</span>
<span class="n">window</span><span class="o">=</span><span class="mi">50</span>
<span class="n">shift</span><span class="o">=</span><span class="mi">5</span>
<span class="n">r2</span><span class="o">=</span><span class="mf">0.5</span>
<span class="n">stand</span><span class="o">=</span><span class="s2">&quot;binom2&quot;</span>
<span class="n">maf_filter</span><span class="o">=</span><span class="mf">0.01</span>
<span class="n">geno_filter</span><span class="o">=</span><span class="mf">0.01</span>
<span class="n">mind_filter</span><span class="o">=</span><span class="mf">0.02</span>

<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">PCA</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">smartpca</span> \
    <span class="o">--</span><span class="n">cwd</span> <span class="err">$</span><span class="n">pca_dir</span> \
    <span class="o">--</span><span class="n">bedfiles</span> <span class="err">$</span><span class="n">bedfiles</span> \
    <span class="o">--</span><span class="n">bimfiles</span> <span class="err">$</span><span class="n">bimfiles</span> \
    <span class="o">--</span><span class="n">famFile</span> <span class="err">$</span><span class="n">famFile</span> \
    <span class="o">--</span><span class="n">database</span> <span class="err">$</span><span class="n">database</span> \
    <span class="o">--</span><span class="n">k</span> <span class="err">$</span><span class="n">k</span> \
    <span class="o">--</span><span class="n">stand</span> <span class="err">$</span><span class="n">stand</span> \
    <span class="o">--</span><span class="n">maxiter</span> <span class="err">$</span><span class="n">maxiter</span> \
    <span class="o">--</span><span class="n">topk</span> <span class="err">$</span><span class="n">topk</span> \
    <span class="o">--</span><span class="n">sigma</span> <span class="err">$</span><span class="n">sigma</span> \
    <span class="o">--</span><span class="n">window</span> <span class="err">$</span><span class="n">window</span> \
    <span class="o">--</span><span class="n">shift</span> <span class="err">$</span><span class="n">shift</span> \
    <span class="o">--</span><span class="n">r2</span> <span class="err">$</span><span class="n">r2</span> \
    <span class="o">--</span><span class="n">maf_filter</span> <span class="err">$</span><span class="n">maf_filter</span>\
    <span class="o">--</span><span class="n">geno_filter</span> <span class="err">$</span><span class="n">geno_filter</span>\
    <span class="o">--</span><span class="n">mind_filter</span> <span class="err">$</span><span class="n">mind_filter</span> \
    <span class="o">--</span><span class="n">numThreads</span> <span class="err">$</span><span class="n">numThreads</span> \
    <span class="o">--</span><span class="n">job_size</span> <span class="err">$</span><span class="n">job_size</span> \
    <span class="o">--</span><span class="n">container_lmm</span> <span class="err">$</span><span class="n">container_lmm</span> \
    <span class="o">-</span><span class="n">s</span> <span class="n">build</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">smartpca</span><span class="o">.</span><span class="n">perl</span> \
    <span class="o">-</span><span class="n">i</span> <span class="n">example</span><span class="o">.</span><span class="n">geno</span> \
    <span class="o">-</span><span class="n">a</span> <span class="n">example</span><span class="o">.</span><span class="n">snp</span> \
    <span class="o">-</span><span class="n">b</span> <span class="n">example</span><span class="o">.</span><span class="n">ind</span> \
    <span class="o">-</span><span class="n">k</span> <span class="mi">2</span> \
    <span class="o">-</span><span class="n">o</span> <span class="n">example</span><span class="o">.</span><span class="n">pca</span> \
    <span class="o">-</span><span class="n">p</span> <span class="n">example</span><span class="o">.</span><span class="n">plot</span> \
    <span class="o">-</span><span class="n">e</span> <span class="n">example</span><span class="o">.</span><span class="n">eval</span> \
    <span class="o">-</span><span class="n">l</span> <span class="n">example</span><span class="o">.</span><span class="n">log</span> \
    <span class="o">-</span><span class="n">m</span> <span class="mi">5</span> \
    <span class="o">-</span><span class="n">t</span> <span class="mi">2</span> \
    <span class="o">-</span><span class="n">s</span> <span class="mf">6.0</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">par</span><span class="o">.</span><span class="n">PACKEDPED</span><span class="o">.</span><span class="n">EIGENSTRAT</span>
<span class="kn">genotypename:</span>    <span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">merged</span><span class="o">.</span><span class="n">bed</span>
<span class="kn">snpname:</span>         <span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">merged</span><span class="o">.</span><span class="n">bim</span>
<span class="kn">indivname:</span>       <span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">merged</span><span class="o">.</span><span class="n">fam</span>
<span class="kn">outputformat:</span>    <span class="n">EIGENSTRAT</span>
<span class="kn">genotypeoutname:</span> <span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">merged</span><span class="o">.</span><span class="n">eigenstratgeno</span>
<span class="kn">snpoutname:</span>      <span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">merged</span><span class="o">.</span><span class="n">snp</span>
<span class="kn">indivoutname:</span>    <span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">merged</span><span class="o">.</span><span class="n">ind</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#SBATCH --partition general</span>
<span class="c1">#SBATCH --nodes 1</span>
<span class="c1">#SBATCH --ntasks-per-node 1</span>
<span class="c1">#SBATCH --cpus-per-task 1</span>
<span class="c1">#SBATCH --mem 60G</span>
<span class="c1">#SBATCH --time 5-0:00:00</span>
<span class="c1">#SBATCH --job-name ../output/2020-12-01_pca_white</span>
<span class="c1">#SBATCH --output ../output/2020-12-01_pca_white-%J.out</span>
<span class="c1">#SBATCH --error ../output/2020-12-01_pca_white-%J.log</span>
<span class="n">module</span> <span class="n">load</span> <span class="n">EIGENSOFT</span><span class="o">/</span><span class="mf">7.2</span><span class="o">.</span><span class="mi">1</span><span class="o">-</span><span class="n">foss</span><span class="o">-</span><span class="mi">2018</span><span class="n">b</span>
<span class="n">smartpca</span><span class="o">.</span><span class="n">perl</span> <span class="o">-</span><span class="n">i</span> <span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">merged</span><span class="o">.</span><span class="n">bed</span> <span class="o">-</span><span class="n">a</span> <span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">merged</span><span class="o">.</span><span class="n">pedsnp</span> <span class="o">-</span><span class="n">b</span> <span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">merged</span><span class="o">.</span><span class="n">pedind</span> <span class="o">-</span><span class="n">o</span> <span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">merged</span><span class="o">.</span><span class="n">pca</span> <span class="o">-</span><span class="n">p</span> <span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">merged</span><span class="o">.</span><span class="n">plot</span> <span class="o">-</span><span class="n">e</span> <span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">eval</span> <span class="o">-</span><span class="n">l</span> <span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">merged</span><span class="o">.</span><span class="n">log</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Running-plink-missing-pipeline">Running plink missing pipeline<a class="anchor-link" href="#Running-plink-missing-pipeline">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[36]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">pca_dir</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">pca_exomes</span>
<span class="n">famFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">fam</span>
<span class="n">bedfiles</span><span class="o">=</span><span class="sb">`echo /gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_c{1..22}_b0_v1.bed`</span>
<span class="n">bimfiles</span><span class="o">=</span><span class="sb">`echo /gpfs/gibbs/pi/dewan/data/UKBiobank/genotype_files/ukb28374_exomedata/exome_data_OCT2020/UKBexomeOQFE_chr{1..22}.bim`</span>
<span class="c1"># Container</span>
<span class="n">container_lmm</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="n">container_marp</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">marp</span><span class="o">.</span><span class="n">sif</span>
<span class="c1"># Pipeline</span>
<span class="n">plink_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">workflow</span><span class="o">/</span><span class="n">plink_missing</span><span class="o">.</span><span class="n">ipynb</span>
<span class="c1"># Name of bash script</span>
<span class="n">pca_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_plink_miss</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">1</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>


<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">workflow</span><span class="o">/</span><span class="n">plink_missing</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">missing</span>\
    <span class="o">--</span><span class="n">cwd</span> <span class="err">$</span><span class="n">pca_dir</span> \
    <span class="o">--</span><span class="n">bedfiles</span> <span class="err">$</span><span class="n">bedfiles</span> \
    <span class="o">--</span><span class="n">bimfiles</span> <span class="err">$</span><span class="n">bimfiles</span> \
    <span class="o">--</span><span class="n">famFile</span> <span class="err">$</span><span class="n">famFile</span> \
    <span class="o">--</span><span class="n">numThreads</span> <span class="err">$</span><span class="n">numThreads</span> \
    <span class="o">--</span><span class="n">job_size</span> <span class="err">$</span><span class="n">job_size</span> \
    <span class="o">--</span><span class="n">container_lmm</span> <span class="err">$</span><span class="n">container_lmm</span> \
    <span class="o">--</span><span class="n">container_marp</span> <span class="err">$</span><span class="n">container_marp</span> \
    <span class="o">-</span><span class="n">s</span> <span class="n">build</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Running <span class="ansi-green-fg">missing_1</span>: Genotype and sample missingness for exome files
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=0) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=1) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=2) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=3) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=4) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=5) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=6) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=7) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=8) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=9) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=10) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=11) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=12) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=13) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=14) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=15) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=16) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=17) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=18) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=19) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=20) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: Step <span class="ansi-green-fg">missing_1</span> (index=21) is <span class="ansi-green-fg">ignored</span> with signature constructed
INFO: <span class="ansi-green-fg">missing_1</span> output:   <span class="ansi-green-fg">/gpfs/gibbs/pi/dewan/data/UKBiobank/results/pca_exomes/ukb23155_c1_b0_v1.imiss /gpfs/gibbs/pi/dewan/data/UKBiobank/results/pca_exomes/ukb23155_c1_b0_v1.lmiss... (88 items in 22 groups)</span>
INFO: Running <span class="ansi-green-fg">missing_2</span>: Generate plots to understand patterns of missing data
1783906 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=0) is <span class="ansi-green-fg">completed</span>.
1310712 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=1) is <span class="ansi-green-fg">completed</span>.
1039941 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=2) is <span class="ansi-green-fg">completed</span>.
717783 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=3) is <span class="ansi-green-fg">completed</span>.
790760 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=4) is <span class="ansi-green-fg">completed</span>.
887003 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=5) is <span class="ansi-green-fg">completed</span>.
858981 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=6) is <span class="ansi-green-fg">completed</span>.
654090 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=7) is <span class="ansi-green-fg">completed</span>.
774556 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=8) is <span class="ansi-green-fg">completed</span>.
733427 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=9) is <span class="ansi-green-fg">completed</span>.
1059628 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=10) is <span class="ansi-green-fg">completed</span>.
950532 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=11) is <span class="ansi-green-fg">completed</span>.
320631 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=12) is <span class="ansi-green-fg">completed</span>.
557237 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=13) is <span class="ansi-green-fg">completed</span>.
620095 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=14) is <span class="ansi-green-fg">completed</span>.
875192 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=15) is <span class="ansi-green-fg">completed</span>.
1047224 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=16) is <span class="ansi-green-fg">completed</span>.
286694 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=17) is <span class="ansi-green-fg">completed</span>.
1210652 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=18) is <span class="ansi-green-fg">completed</span>.
459900 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=19) is <span class="ansi-green-fg">completed</span>.
195726 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=20) is <span class="ansi-green-fg">completed</span>.
414980 variants loaded from .bim file.
INFO: <span class="ansi-green-fg">missing_2</span> (index=21) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_2</span> output:   <span class="ansi-green-fg">/gpfs/gibbs/pi/dewan/data/UKBiobank/results/pca_exomes/ukb23155_c1_b0_v1.missing.md /gpfs/gibbs/pi/dewan/data/UKBiobank/results/pca_exomes/ukb23155_c1_b0_v1.imiss.png... (88 items in 22 groups)</span>
INFO: Running <span class="ansi-green-fg">missing_3</span>: Generate markdown file
INFO: <span class="ansi-green-fg">missing_3</span> (index=0) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=1) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=2) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=3) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=4) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=5) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=6) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=7) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=8) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=9) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=10) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=11) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=12) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=13) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=14) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=15) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=16) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=17) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=18) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=19) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=20) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_3</span> (index=21) is <span class="ansi-green-fg">completed</span>.
INFO: Running <span class="ansi-green-fg">missing_4</span>: Generate analysis report: HTML file, and optionally PPTX file
INFO: <span class="ansi-green-fg">missing_4</span> is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">missing_4</span> output:   <span class="ansi-green-fg">/gpfs/gibbs/pi/dewan/data/UKBiobank/results/pca_exomes/ukb23155.merged.html</span>
INFO: Workflow missing (ID=w0e3348cee7472d97) is executed successfully with 3 completed steps, 45 completed substeps, 1 ignored step and 22 ignored substeps.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Extracting-individuals-for-a-particular-snp-plink">Extracting individuals for a particular snp plink<a class="anchor-link" href="#Extracting-individuals-for-a-particular-snp-plink">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[37]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tpl_file</span><span class="o">=../</span><span class="n">farnam</span><span class="o">.</span><span class="n">yml</span>
<span class="n">pca_dir</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">plink_extract</span>
<span class="n">famFile</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_s200631</span><span class="o">.</span><span class="n">fam</span>
<span class="n">bedfiles</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">ukb23155_c12_b0_v1</span><span class="o">.</span><span class="n">bed</span>
<span class="n">bimfiles</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb28374_exomedata</span><span class="o">/</span><span class="n">exome_data_OCT2020</span><span class="o">/</span><span class="n">UKBexomeOQFE_chr12</span><span class="o">.</span><span class="n">bim</span>
<span class="n">snp_list</span><span class="o">=/</span><span class="n">home</span><span class="o">/</span><span class="n">dc2325</span><span class="o">/</span><span class="n">scratch60</span><span class="o">/</span><span class="n">plink_extract</span><span class="o">/</span><span class="n">snp</span><span class="o">.</span><span class="n">txt</span>
<span class="c1"># Container</span>
<span class="n">container_lmm</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
<span class="n">container_marp</span><span class="o">=/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">marp</span><span class="o">.</span><span class="n">sif</span>
<span class="c1"># Pipeline</span>
<span class="n">plink_sos</span><span class="o">=~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">plink_extract</span><span class="o">.</span><span class="n">ipynb</span>
<span class="c1"># Name of bash script</span>
<span class="n">pca_sbatch</span><span class="o">=../</span><span class="n">output</span><span class="o">/</span><span class="err">$</span><span class="p">(</span><span class="n">date</span> <span class="o">+</span><span class="s2">&quot;%Y-%m-</span><span class="si">%d</span><span class="s2">&quot;</span><span class="p">)</span><span class="n">_plink_miss</span><span class="o">.</span><span class="n">sbatch</span>
<span class="n">numThreads</span><span class="o">=</span><span class="mi">1</span>
<span class="n">job_size</span><span class="o">=</span><span class="mi">1</span>


<span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">UKBB_GWAS_dev</span><span class="o">/</span><span class="n">plink_extract</span><span class="o">.</span><span class="n">ipynb</span>  \
    <span class="o">--</span><span class="n">cwd</span> <span class="err">$</span><span class="n">pca_dir</span> \
    <span class="o">--</span><span class="n">bedfiles</span> <span class="err">$</span><span class="n">bedfiles</span> \
    <span class="o">--</span><span class="n">bimfiles</span> <span class="err">$</span><span class="n">bimfiles</span> \
    <span class="o">--</span><span class="n">famFile</span> <span class="err">$</span><span class="n">famFile</span> \
    <span class="o">--</span><span class="n">snp_list</span> <span class="err">$</span><span class="n">snp_list</span> \
    <span class="o">--</span><span class="n">numThreads</span> <span class="err">$</span><span class="n">numThreads</span> \
    <span class="o">--</span><span class="n">job_size</span> <span class="err">$</span><span class="n">job_size</span> \
    <span class="o">--</span><span class="n">container_lmm</span> <span class="err">$</span><span class="n">container_lmm</span> \
    <span class="o">--</span><span class="n">container_marp</span> <span class="err">$</span><span class="n">container_marp</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre><span class="ansi-red-intense-fg">ERROR</span>: <span class="ansi-red-intense-fg">Failed to locate /home/dc2325/project/UKBB_GWAS_dev/plink_extract.ipynb.sos</span>

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Regenie-burden">Regenie burden<a class="anchor-link" href="#Regenie-burden">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[23]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">genoFile</span><span class="o">=</span><span class="sb">`echo /gpfs/gibbs/pi/dewan/data/UKBiobank/MWE/burden/ukb23155_c{21..22}_b0_v1.plink.exome.filtered.bed`</span>
<span class="n">sos</span> <span class="n">dryrun</span> <span class="o">~/</span><span class="n">project</span><span class="o">/</span><span class="n">bioworkflows</span><span class="o">/</span><span class="n">GWAS</span><span class="o">/</span><span class="n">LMM</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">regenie_burden</span> \
    <span class="o">--</span><span class="n">cwd</span> <span class="n">output</span> \
    <span class="o">--</span><span class="n">bfile</span> <span class="n">genotypes21_22</span><span class="o">.</span><span class="n">bed</span> \
    <span class="o">--</span><span class="n">genoFile</span> <span class="err">$</span><span class="n">genoFile</span> \
    <span class="o">--</span><span class="n">sampleFile</span> \
    <span class="o">--</span><span class="n">phenoFile</span> <span class="n">burden</span><span class="o">/</span><span class="n">phenotype_burden</span><span class="o">.</span><span class="n">txt</span>\
    <span class="o">--</span><span class="n">phenoCol</span> <span class="n">ASTHMA</span>\
    <span class="o">--</span><span class="n">covarCol</span> <span class="n">SEX</span> \
    <span class="o">--</span><span class="n">qCovarCol</span> <span class="n">AGE</span> \
    <span class="o">--</span><span class="n">numThreads</span> <span class="mi">8</span> \
    <span class="o">--</span><span class="n">bsize</span> <span class="mi">10</span> \
    <span class="o">--</span><span class="n">anno_file</span> <span class="n">burden</span><span class="o">/</span><span class="n">annotation_file</span><span class="o">.</span><span class="n">txt</span>\
    <span class="o">--</span><span class="n">set_list</span> <span class="n">burden</span><span class="o">/</span><span class="n">set_list_file</span><span class="o">.</span><span class="n">txt</span> \
    <span class="o">--</span><span class="n">mask_file</span> <span class="n">burden</span><span class="o">/</span><span class="n">mask_file</span><span class="o">.</span><span class="n">txt</span> \
    <span class="o">--</span><span class="n">keep_gene</span> <span class="n">burden</span><span class="o">/</span><span class="n">keep_file</span><span class="o">.</span><span class="n">txt</span>\
    <span class="o">--</span><span class="n">aaf_bins</span> <span class="mf">0.05</span> \
    <span class="o">--</span><span class="n">trait</span> <span class="n">bt</span> \
    <span class="o">--</span><span class="n">build_mask</span> <span class="nb">max</span> \
    <span class="o">--</span><span class="n">container_lmm</span> <span class="o">/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">gibbs</span><span class="o">/</span><span class="n">pi</span><span class="o">/</span><span class="n">dewan</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">lmm</span><span class="o">.</span><span class="n">sif</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>INFO: Checking <span class="ansi-green-fg">regenie_burden</span>: Run regenie for burden tests
HINT: singularity exec  /gpfs/gibbs/pi/dewan/data/UKBiobank/lmm.sif /bin/bash /gpfs/ysm/project/dewan/dc2325/UKBB_GWAS_dev/analysis/tmph24c8d72/singularity_run_193934.sh
set -e
regenie \
  --step 2 \
  --bed /gpfs/gibbs/pi/dewan/data/UKBiobank/MWE/burden/ukb23155_c21_b0_v1.plink.exome.filtered \
  --phenoFile output/phenotype_burden.regenie_phenotype \
  --covarFile output/phenotype_burden.regenie_covar \
  --phenoColList ASTHMA \
  --bt
  --firth --approx \
  --pred output/phenotype_burden_ASTHMA.regenie_pred.list \
  --anno-file burden/annotation_file.txt \
  --set-list burden/set_list_file.txt \
  --extract-sets burden/keep_file.txt\
  --mask-def burden/mask_file.txt \
  --aaf-bins 0.05 \
  --write-mask \
  --build-mask \
  --bsize 10 \
  --check-burden-files \
  --gz \
  --out  output/cache/ukb23155_c21_b0_v1.plink.exome.filtered.burden


INFO: <span class="ansi-green-fg">regenie_burden</span> (index=0) is <span class="ansi-green-fg">completed</span>.
HINT: singularity exec  /gpfs/gibbs/pi/dewan/data/UKBiobank/lmm.sif /bin/bash /gpfs/ysm/project/dewan/dc2325/UKBB_GWAS_dev/analysis/tmpeg8za31c/singularity_run_193934.sh
set -e
regenie \
  --step 2 \
  --bed /gpfs/gibbs/pi/dewan/data/UKBiobank/MWE/burden/ukb23155_c22_b0_v1.plink.exome.filtered \
  --phenoFile output/phenotype_burden.regenie_phenotype \
  --covarFile output/phenotype_burden.regenie_covar \
  --phenoColList ASTHMA \
  --bt
  --firth --approx \
  --pred output/phenotype_burden_ASTHMA.regenie_pred.list \
  --anno-file burden/annotation_file.txt \
  --set-list burden/set_list_file.txt \
  --extract-sets burden/keep_file.txt\
  --mask-def burden/mask_file.txt \
  --aaf-bins 0.05 \
  --write-mask \
  --build-mask \
  --bsize 10 \
  --check-burden-files \
  --gz \
  --out  output/cache/ukb23155_c22_b0_v1.plink.exome.filtered.burden


INFO: <span class="ansi-green-fg">regenie_burden</span> (index=1) is <span class="ansi-green-fg">completed</span>.
INFO: <span class="ansi-green-fg">regenie_burden</span> output:   <span class="ansi-green-fg">output/cache/ukb23155_c21_b0_v1.plink.exome.filtered_ASTHMAburden.regenie.gz output/cache/ukb23155_c22_b0_v1.plink.exome.filtered_ASTHMAburden.regenie.gz in 2 groups</span>
INFO: Workflow regenie_burden (ID=w7723835731215d60) is tested successfully with 1 completed step and 2 completed substeps.

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span>
</pre></div>

    </div>
</div>
</div>

</div>
<hr>
&copy 2020 Diana Cornejo at Center for Statistical Genetics, Sergievsky Center, Columbia University VP&S
<p><small>Exported from <a href="http://github.com/dianacornejo/pleiotropy_UKB/blob/804a49e20903d8a6a7eafcb7472453c5ea2dcde8/analysis/cluster_scripts/PCA_scripts.ipynb"><code>analysis/cluster_scripts/PCA_scripts.ipynb</code></a> committed by dc2325 on Fri Feb 19 22:22:36 2021 <a href="http://github.com/dianacornejo/pleiotropy_UKB/commit/804a49e20903d8a6a7eafcb7472453c5ea2dcde8">revision 1, 804a49e</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
