{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d3b610ed-89f6-4730-b43c-7f906a2e6dd5",
   "metadata": {},
   "source": [
    "# Regenie interaction for Hearing impairment traits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1cd3be57-bc42-484a-98df-25392cd12877",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Common variables Columbia's cluster\n",
    "UKBB_PATH=$HOME/UKBiobank\n",
    "UKBB_yale=$HOME/UKBiobank_Yale_transfer\n",
    "USER_PATH=$HOME/working\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "container_marp=$HOME/containers/marp.sif\n",
    "container_annovar=$HOME/containers/gatk4-annovar.sif\n",
    "hearing_pheno_path=$UKBB_PATH/phenotype_files/hearing_impairment\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "formatFile_fastgwa=$USER_PATH/UKBB_GWAS_dev/data/fastGWA_template.yml\n",
    "formatFile_bolt=$USER_PATH/UKBB_GWAS_dev/data/boltlmm_template.yml\n",
    "formatFile_saige=$USER_PATH/UKBB_GWAS_dev/data/saige_template.yml\n",
    "formatFile_regenie=$USER_PATH/UKBB_GWAS_dev/data/regenie_template.yml\n",
    "# Workflows\n",
    "lmm_sos=$USER_PATH/bioworkflows/GWAS/LMM.ipynb\n",
    "anno_sos=$USER_PATH/bioworkflows/variant-annotation/annovar.ipynb\n",
    "clumping_sos=$USER_PATH/bioworkflows/GWAS/LD_Clumping.ipynb\n",
    "extract_sos=$USER_PATH/bioworkflows/GWAS/Region_Extraction.ipynb\n",
    "snptogene_sos=$USER_PATH/UKBB_GWAS_dev/workflow/snptogene.ipynb\n",
    "\n",
    "# LMM directories for array data data\n",
    "lmm_imp_dir_fastgwa=$UKBB_PATH/results/FastGWA_results/results_array_data\n",
    "lmm_imp_dir_bolt=$UKBB_PATH/results/BOLTLMM_results/results_array_data\n",
    "lmm_imp_dir_saige=$UKBB_PATH/results/SAIGE_results/results_array_data\n",
    "lmm_imp_dir_regenie=$UKBB_PATH/results/REGENIE_results/results_array_data\n",
    "\n",
    "# LMM directories for exome data\n",
    "lmm_exome_dir_fastgwa=$UKBB_PATH/results/FastGWA_results/results_array_data\n",
    "lmm_exome_dir_bolt=$UKBB_PATH/results/BOLTLMM_results/results_array_data\n",
    "lmm_exome_dir_saige=$UKBB_PATH/results/SAIGE_results/results_array_data\n",
    "lmm_exome_dir_regenie=$UKBB_PATH/results/REGENIE_results/results_array_data\n",
    "### Specific to REGENIE\n",
    "bsize=1000\n",
    "lowmem=~/project/guangyou/interaction/scratch60/\n",
    "lowmem_dir=~/project/guangyou/interaction/scratch60/predictions\n",
    "trait=bt\n",
    "minMAC=4\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "reverse_log_p=True\n",
    "numThreads=2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b16c64bc-da15-4fc9-8a39-42acc50caf00",
   "metadata": {},
   "source": [
    "# MWE\n",
    "\n",
    "### f3393 first 1000 samples and first 1000 interactions (excluding 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7558698b-8259-44bd-951a-f3afd2555934",
   "metadata": {},
   "outputs": [],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=/home/gl2776/working/interaction/MWE/regenie_mwe\n",
    "lmm_sbatch_regenie=/home/gl2776/working/interaction/MWE/mwe-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=/home/gl2776/working/interaction/MWE/mwe.phe\n",
    "phenoCol=f3393\n",
    "covarCol=sex\n",
    "qCovarCol=\"age PC1 PC2 xb\"\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=/home/gl2776/working/interaction/MWE/mwe.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "anno_file=\n",
    "label_annotate=SNP\n",
    "known_vars=\n",
    "new_vars=\n",
    "lmm_job_size=1\n",
    "ylim=0\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait \n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --numThreads $numThreads\n",
    "    --anno_file $anno_file\n",
    "    --label_annotate $label_annotate\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32a19e2b-5586-4f6c-9bae-f65f9dbfac25",
   "metadata": {},
   "source": [
    "## f3393 all samples and first 10000 interactions (excluding 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ebf78dd6-e3c7-4023-8582-cb17ffa226a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=/home/gl2776/project/guangyou/interaction/110921_f3393\n",
    "lmm_sbatch_regenie=/home/gl2776/project/guangyou/interaction/110921_f3393/f3393-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=/home/gl2776/project/guangyou/interaction/phenotype/f3393.phe\n",
    "phenoCol=f3393\n",
    "covarCol=sex\n",
    "qCovarCol=\"age PC1 PC2 xb\"\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.xx.1.1.exclu4.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "anno_file=\n",
    "label_annotate=SNP\n",
    "known_vars=\n",
    "new_vars=\n",
    "lmm_job_size=1\n",
    "ylim=0\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait \n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --numThreads $numThreads\n",
    "    --anno_file $anno_file\n",
    "    --label_annotate $label_annotate\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eb76ca97-215e-41e7-96db-7268dcc3a3a7",
   "metadata": {},
   "source": [
    "## case only f3393 y ~ x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d2703fb0-47b0-42fb-8bf4-6c50e12b5e1b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/project/guangyou/interaction/111221_f3393_caseonly_y/f3393_caseonly_y-regenie_2021-11-13.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wb3ad4be56b6c3305) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=/home/gl2776/project/guangyou/interaction/111221_f3393_caseonly_y\n",
    "lmm_sbatch_regenie=/home/gl2776/project/guangyou/interaction/111221_f3393_caseonly_y/f3393_caseonly_y-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=/home/gl2776/project/guangyou/interaction/phenotype/f3393.caseonly.phe\n",
    "phenoCol=f3393\n",
    "covarCol=sex\n",
    "qCovarCol=\"age PC1 PC2 xb\"\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.caseonly.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "anno_file=\n",
    "label_annotate=SNP\n",
    "known_vars=\n",
    "new_vars=\n",
    "lmm_job_size=1\n",
    "ylim=0\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait \n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --numThreads $numThreads\n",
    "    --anno_file $anno_file\n",
    "    --label_annotate $label_annotate\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32a13249-f51e-43a7-92e9-a1b2e0861e04",
   "metadata": {},
   "source": [
    "## case only f3393 x1 ~ x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6435e6f3-e90f-4114-bf35-6b2b71214733",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/project/guangyou/interaction/111221_f3393_caseonly_x1/f3393_caseonly_x1-regenie_2021-11-14.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=web068279b214c6bd) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=/home/gl2776/project/guangyou/interaction/111221_f3393_caseonly_x1\n",
    "lmm_sbatch_regenie=/home/gl2776/project/guangyou/interaction/111221_f3393_caseonly_x1/f3393_caseonly_x1-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.caseonly.x\n",
    "phenoCol=snp1\n",
    "covarCol=\n",
    "qCovarCol=\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.caseonly.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "anno_file=\n",
    "label_annotate=SNP\n",
    "known_vars=\n",
    "new_vars=\n",
    "lmm_job_size=1\n",
    "ylim=0\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "trait=\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --numThreads $numThreads\n",
    "    --anno_file $anno_file\n",
    "    --label_annotate $label_annotate\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b729d9db-869b-4131-bc07-5289c86ce71c",
   "metadata": {},
   "source": [
    "## case only f3393 xs ~ x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2a738338-aabe-4a11-9e86-6e5fc37a997b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/project/guangyou/interaction/111221_f3393_caseonly_xs/f3393_caseonly_xs-regenie_2021-11-14.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wf8da67a69991ee74) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=/home/gl2776/project/guangyou/interaction/111221_f3393_caseonly_xs\n",
    "lmm_sbatch_regenie=/home/gl2776/project/guangyou/interaction/111221_f3393_caseonly_xs/f3393_caseonly_xs-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.caseonly.x\n",
    "phenoCol=`echo snp{1..4020}`\n",
    "covarCol=\n",
    "qCovarCol=\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.caseonly.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "anno_file=\n",
    "label_annotate=SNP\n",
    "known_vars=\n",
    "new_vars=\n",
    "lmm_job_size=1\n",
    "ylim=0\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "trait=\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --numThreads $numThreads\n",
    "    --anno_file $anno_file\n",
    "    --label_annotate $label_annotate\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e15e10be-835f-4e68-8388-dd79fb64aee5",
   "metadata": {},
   "source": [
    "## case only f2247 xs ~ x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c12af853-94cf-484c-9f5f-0d9c0f094880",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/project/guangyou/interaction/111221_f2247_caseonly_xs/f2247_caseonly_xs-regenie_2021-11-14.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w9d3956e65f2f7bfe) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=/home/gl2776/project/guangyou/interaction/111221_f2247_caseonly_xs\n",
    "lmm_sbatch_regenie=/home/gl2776/project/guangyou/interaction/111221_f2247_caseonly_xs/f2247_caseonly_xs-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=/home/gl2776/project/guangyou/interaction/bfile/f2247/f2247.caseonly.x\n",
    "phenoCol=`echo snp{1..5252}`\n",
    "covarCol=\n",
    "qCovarCol=\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=/home/gl2776/project/guangyou/interaction/bfile/f2247/f2247.caseonly.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "anno_file=\n",
    "label_annotate=SNP\n",
    "known_vars=\n",
    "new_vars=\n",
    "lmm_job_size=1\n",
    "ylim=0\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "trait=\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --numThreads $numThreads\n",
    "    --anno_file $anno_file\n",
    "    --label_annotate $label_annotate\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d0af0359-28f0-488f-bb0a-1cc779564be1",
   "metadata": {},
   "source": [
    "## case only f2257 xs ~ x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8c6eb5ee-718a-479e-b499-38e55e58350b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/project/guangyou/interaction/111221_f2257_caseonly_xs/f2257_caseonly_xs-regenie_2021-11-14.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wc2b2ce546bb0a4a9) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=/home/gl2776/project/guangyou/interaction/111221_f2257_caseonly_xs\n",
    "lmm_sbatch_regenie=/home/gl2776/project/guangyou/interaction/111221_f2257_caseonly_xs/f2257_caseonly_xs-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=/home/gl2776/project/guangyou/interaction/bfile/f2257/f2257.caseonly.x\n",
    "phenoCol=`echo snp{1..5252}`\n",
    "covarCol=\n",
    "qCovarCol=\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=/home/gl2776/project/guangyou/interaction/bfile/f2257/f2257.caseonly.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "anno_file=\n",
    "label_annotate=SNP\n",
    "known_vars=\n",
    "new_vars=\n",
    "lmm_job_size=1\n",
    "ylim=0\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "trait=\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --numThreads $numThreads\n",
    "    --anno_file $anno_file\n",
    "    --label_annotate $label_annotate\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3eaa6b53-07e7-4280-b317-9363f5392a83",
   "metadata": {},
   "source": [
    "## Benchmark of the resources Regenie takes by running a few SNPs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6169087d-04ab-4c21-912c-e286983519a7",
   "metadata": {},
   "source": [
    "## case only f3393 x1 ~ x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "11b03c43-e290-4215-bce9-2e90748cb9fc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/project/guangyou/interaction/111521_f3393_caseonly_x1/f3393_caseonly_x1-regenie_2021-11-15.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w18fc9a94722be400) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=/home/gl2776/project/guangyou/interaction/111521_f3393_caseonly_x1\n",
    "lmm_sbatch_regenie=/home/gl2776/project/guangyou/interaction/111521_f3393_caseonly_x1/f3393_caseonly_x1-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.caseonly.x\n",
    "phenoCol=snp1\n",
    "covarCol=\n",
    "qCovarCol=\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.caseonly.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "anno_file=\n",
    "label_annotate=SNP\n",
    "known_vars=\n",
    "new_vars=\n",
    "lmm_job_size=1\n",
    "ylim=0\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "trait=\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --numThreads $numThreads\n",
    "    --anno_file $anno_file\n",
    "    --label_annotate $label_annotate\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f47a6315-c0b0-4927-bc6c-d7b96209b656",
   "metadata": {},
   "source": [
    "## case only f3393 x{1..3} ~ x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a473c769-b6f7-4d11-a95a-07756ec9b791",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/project/guangyou/interaction/111521_f3393_caseonly_x3/f3393_caseonly_x3-regenie_2021-11-15.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w2f132f6074811567) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=/home/gl2776/project/guangyou/interaction/111521_f3393_caseonly_x3\n",
    "lmm_sbatch_regenie=/home/gl2776/project/guangyou/interaction/111521_f3393_caseonly_x3/f3393_caseonly_x3-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.caseonly.x\n",
    "phenoCol=`echo snp{1..3}`\n",
    "covarCol=\n",
    "qCovarCol=\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.caseonly.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "anno_file=\n",
    "label_annotate=SNP\n",
    "known_vars=\n",
    "new_vars=\n",
    "lmm_job_size=1\n",
    "ylim=0\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "trait=\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --numThreads $numThreads\n",
    "    --anno_file $anno_file\n",
    "    --label_annotate $label_annotate\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed499d50-d63c-43a2-8b7f-d33851d7b135",
   "metadata": {},
   "source": [
    "## case only f3393 x{1..10} ~ x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b0d77d4d-984c-4727-b871-62725cb80c15",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/project/guangyou/interaction/111521_f3393_caseonly_x10/f3393_caseonly_x10-regenie_2021-11-15.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w451c8dccb171a08c) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=/home/gl2776/project/guangyou/interaction/111521_f3393_caseonly_x10\n",
    "lmm_sbatch_regenie=/home/gl2776/project/guangyou/interaction/111521_f3393_caseonly_x10/f3393_caseonly_x10-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.caseonly.x\n",
    "phenoCol=`echo snp{1..10}`\n",
    "covarCol=\n",
    "qCovarCol=\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.caseonly.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "anno_file=\n",
    "label_annotate=SNP\n",
    "known_vars=\n",
    "new_vars=\n",
    "lmm_job_size=1\n",
    "ylim=0\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "trait=\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --numThreads $numThreads\n",
    "    --anno_file $anno_file\n",
    "    --label_annotate $label_annotate\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "df0dfabb-2a2c-45b2-b2aa-4f17361da376",
   "metadata": {},
   "source": [
    "# Regenie interaction: Correct the model\n",
    "\n",
    "logit(p) = x1x2+x.+sex+age+g+e "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0e66bf34-68d1-4cc3-8997-a513526a908a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/project/guangyou/interaction/111821_f3393/f3393_hearing_aid-regenie_interaction_2021-11-18.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w8e77a7f7f6fefcc6) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=~/project/guangyou/interaction/111821_f3393\n",
    "lmm_sbatch_regenie=$lmm_dir_regenie/f3393_hearing_aid-regenie_interaction_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.addx.phe\n",
    "phenoCol=f3393\n",
    "covarCol=sex\n",
    "s=\"age PC1 PC2\"\n",
    "for i in {1..4020}\n",
    "do\n",
    "t=\" snp\"\n",
    "s=\"$s$t$i\"\n",
    "done\n",
    "qCovarCol=`echo $s`\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.xx.1.1.exclu4.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "anno_file=\n",
    "label_annotate=SNP\n",
    "known_vars=\n",
    "new_vars=\n",
    "lmm_job_size=1\n",
    "ylim=0\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait \n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --numThreads $numThreads\n",
    "    --anno_file $anno_file\n",
    "    --label_annotate $label_annotate\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92419e44-872c-4a24-80c0-aa44b5dec064",
   "metadata": {},
   "source": [
    "# Simulation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "490ae0bc-3791-4d3b-a62d-f0129654dfe1",
   "metadata": {},
   "outputs": [],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=/home/gl2776/working/interaction/simulation/112421_caseonly\n",
    "lmm_sbatch_regenie=$lmm_dir_regenie/simulation_caseonly-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=/home/gl2776/working/interaction/simulation/simulation.phe\n",
    "phenoCol=`echo snp{1..10}`\n",
    "covarCol=\n",
    "qCovarCol=\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=/home/gl2776/working/interaction/simulation/simulation.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "anno_file=\n",
    "label_annotate=SNP\n",
    "known_vars=\n",
    "new_vars=\n",
    "lmm_job_size=1\n",
    "ylim=0\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "trait=\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "lmm_sos=~/working/LMM-caseonly.ipynb\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --numThreads $numThreads\n",
    "    --anno_file $anno_file\n",
    "    --label_annotate $label_annotate\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "301d1192-1f00-45a9-b206-af969a23ca60",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "1ad89b46-1e7b-4b69-9a11-2789257d99d3",
   "metadata": {},
   "source": [
    "# Regenie interaction MWE (new version)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48fd2c39-72c1-4a4b-bc8d-bb1c9fcbcdf4",
   "metadata": {},
   "outputs": [],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=/home/gl2776/working/interaction/newv/112421_regenie_interaction_pipeline_mwe\n",
    "lmm_sbatch_regenie=$lmm_dir_regenie/MWE-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=~/UKBiobank/MWE_regenie/1000G_data/1000G.EUR.pheno.x\n",
    "phenoCol=x\n",
    "covarCol=sex\n",
    "qCovarCol=\n",
    "bfile=~/UKBiobank/MWE_regenie/1000G_data/1000G.EUR.filtered.pruned.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=~/UKBiobank/MWE_regenie/1000G_data/1000G.EUR.filtered.pruned.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "lmm_job_size=10\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "minMAC=5\n",
    "maf_filter=0.001\n",
    "geno_filter=0.1\n",
    "hwe_filter=5e-08\n",
    "mind_filter=0.1\n",
    "regenie_sos=~/working/interaction/newv/Regenie_interaction.ipynb\n",
    "regenieFile=/mnt/mfs/statgen/guangyou/containers/regenie.v.2.3\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --regenieFile $regenieFile\n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait \n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $regenie_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd23e073-86e3-4b0c-b462-eee7c178c2df",
   "metadata": {},
   "source": [
    "# Regenie interaction new version f3393"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "09d4e0bb-3533-46d2-b8dd-11e744014ed6",
   "metadata": {},
   "outputs": [],
   "source": [
    "## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "lmm_dir_regenie=/home/gl2776/project/guangyou/interaction/newv/112421_f3393\n",
    "lmm_sbatch_regenie=$lmm_dir_regenie/f3393_newversion-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "#Use the phenotype file for the 500K individuals\n",
    "phenoFile=$hearing_pheno_path/fulldb_500K/100521_UKBB_Hearing_aid_f3393_expandedwhite_15601cases_237318ctrl_500k_PC1_PC2\n",
    "phenoCol=f3393\n",
    "covarCol=sex\n",
    "qCovarCol=\"age PC1 PC2\"\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the same bfile here for the association\n",
    "genoFile=/home/gl2776/project/guangyou/interaction/bfile/f3393/f3393.clumped.bed\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "lmm_job_size=20\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "regenie_sos=~/working/interaction/newv/Regenie_interaction.ipynb\n",
    "regenieFile=/mnt/mfs/statgen/guangyou/containers/regenie.v.2.3\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --regenieFile $regenieFile\n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Calysto Bash",
   "language": "bash",
   "name": "calysto_bash"
  },
  "language_info": {
   "file_extension": ".sh",
   "help_links": [
    {
     "text": "MetaKernel Magics",
     "url": "https://metakernel.readthedocs.io/en/latest/source/README.html"
    }
   ],
   "mimetype": "text/x-sh",
   "name": "bash",
   "version": "0.2.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
