<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<title>Pleiotropy analysis in UKBB</title>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>
<style type="text/css">
  div.input_prompt {display: none;}
  div.output_html {
     font-family: "PT Mono", monospace;
     font-size: 10.0pt;
     color: #353535;
     padding-bottom: 25px;
 }
  pre:not([class]) {
    background-color: white;
  }
</style>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'] ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>

</head>

<body>
<style type = "text/css">
@font-face {
 font-family: 'Droid Sans';
 font-weight: normal;
 font-style: normal;
 src: local('Droid Sans'), url('fonts/droid-sans.ttf') format('truetype');
}
@font-face {
 font-family: 'Fira Code';
 font-weight: normal;
 font-style: normal;
 src: local('Fira Code'), url('fonts/firacode.otf') format('opentype');
}
@font-face {
 font-family: 'PT Mono';
 font-weight: normal;
 font-style: normal;
 src: local('PT Mono'), url('fonts/ptmono.ttf') format('truetype');
}

body {
  font-family: "Droid Sans";
  font-size: 160%;
  padding-top: 66px;
  padding-bottom: 40px;
}

h1, h2, h3, h4, h5, h6 {
  margin-top: 20px;
 }

a.anchor-link:link {
  text-decoration: none;
  padding: 0px 20px;
  visibility: hidden;
}

h1:hover .anchor-link,
h2:hover .anchor-link,
h3:hover .anchor-link,
h4:hover .anchor-link,
h5:hover .anchor-link,
h6:hover .anchor-link {
  visibility: hidden;
}

.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Pleiotropy analysis in UKBB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="./index.html">Overview</a>
</li>
        
<li>
  <a href="./workflow.html">Workflow</a>
</li>
        
<li>
  <a href="./analysis.html">Analysis</a>
</li>
        
      </ul>
    
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/dianacornejo/pleiotropy_UKB"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Pleiotropy-Project-UKBB">Pleiotropy Project UKBB<a class="anchor-link" href="#Pleiotropy-Project-UKBB">&#182;</a></h1><h2 id="Overview">Overview<a class="anchor-link" href="#Overview">&#182;</a></h2><p>This project aims to elucidate variants which play a role in pleiotropy for complex diseases and traits with high public health significance.</p>
<p>We investigate phenotypes of strong public health significance, for which there is support for potential pleiotropy.
For example, there may exist pleiotropic relationships between asthma-adiposity, asthma-T2D, adiposity-T2D,
 blood lipids-adiposity, CVD-Alzhemier's disease, and age related hearing impairment (ARHI)-tinnitus. A formal evaluation of pleiotropy for these phenotypes may
reveal novel loci and shed light on the mechanistic relationship between these variants and phenotypes.</p>
<p>We use UK Biobank data for these phenotypes, and perform genome-wide association scan for both genotyped and imputed variants,
analyzing association for both common and rare variants using linear mixed model (LMM).
We then look for evidence of pleiotropy in regions where more than one phenotype have detectable association signals, focusing on
the following pairs of traits: asthma-adiposity, T2D-adiposity, asthma-T2D, LDL-HDL, LDL-TG, HDL-TG, adiposity-LDL, adiposity-HDL,
adiposity-TG, T2G-LDL, T2D-HDL, T2D-TG and ARHI-tinnitus. Methods for fine-mapping, pleiotropy
and mediation analysis will be applied to these regions. We use TOPMed as a secondary replication dataset.</p>
<h2 id="Goals">Goals<a class="anchor-link" href="#Goals">&#182;</a></h2><ol>
<li><strong>Extension of methods for rare and imputed variants and implementation in SEQSpark</strong></li>
<li><strong>Detect pleiotropic effects for a variety of diseases and complex traits using rare variant and imputed data</strong></li>
<li><strong>Determine tissue types most relevant to complex trait/disease etiology</strong></li>
<li><strong>Analysis of tissue-specific expression levels with phenotype data to biologically validate pleiotropic effects</strong></li>
</ol>
<p>The main goal here is to analyze imputed and rare variants for association with different traits (asthma, BMI, waist-to-hip ratio, diabetes, waist circumference). First through a single variant analysis using Linear Mixed Models.</p>
<h2 id="Association-scans-using-LMM">Association scans using LMM<a class="anchor-link" href="#Association-scans-using-LMM">&#182;</a></h2><p>FIXME: add links and status</p>
<h2 id="Candidate-region-analysis">Candidate region analysis<a class="anchor-link" href="#Candidate-region-analysis">&#182;</a></h2><p>For given regions of interest across multiple phenotypes, a number of statistical workflows will be developed for fine-mapping, pleiotropy, and mediation analysis. Candidate regions will be examined with a number of published &amp; in-house methods looking to reveal
the possible causal relationship between genetic variants and multiple phenotypes. The approach developed can also be applied to looking at a number of molecular phenotypes
such as gene expression, splicing, methylation, etc.</p>
<p>Additionally, some of these methods can be generated to analyzing rare variants in the context of multiple phenotypes.</p>
<p>FIXME: add links to relevant issues on this page:</p>
<p><a href="https://github.com/statgenetics/UKBB_GWAS_dev/issues">https://github.com/statgenetics/UKBB_GWAS_dev/issues</a></p>
<h1 id="Hearing-impairment-Project-UKBB">Hearing impairment Project UKBB<a class="anchor-link" href="#Hearing-impairment-Project-UKBB">&#182;</a></h1><h2 id="Documents"><strong>Documents</strong><a class="anchor-link" href="#Documents">&#182;</a></h2><ol>
<li><p>Analisis plan for each phenotype can be found <a href="https://docs.google.com/document/d/1cpxTzElpsEkwmBDjnMBHg2wW7CL1AcG_b0_0wE_k5rQ/edit">here</a></p>
</li>
<li><p>Exclusion criteria applied to the data set is <a href="https://docs.google.com/spreadsheets/d/12L7Cx4Ov8FppGVmG0DxL9uG-lVRHM5QJSea0nORyirQ/edit#gid=0">here</a></p>
</li>
</ol>
<p><strong>Location of the files on Yale's cluster:</strong></p>
<p><strong>Hard-called genotype files:</strong> /SAY/dbgapstg/scratch/UKBiobank/genotype_files/pleiotropy_geneticfiles/UKB_Caucasians_phenotypeindepqc120319_updated082020removedwithdrawnindiv.bed</p>
<p><strong>Imputed genotypes:</strong> /SAY/dbgapstg/scratch/UKBiobank/genotype_files/ukb39554_imputeddataset/</p>
<p><strong>Unrelated samples:</strong> /SAY/dbgapstg/scratch/UKBiobank/genotype_files/pleiotropy_geneticfiles/unrelated_n307259/UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620</p>
<p><strong>Results:</strong> /SAY/dbgapstg/scratch/UKBiobank/results/</p>
<p><strong>Phenotype files:</strong> /SAY/dbgapstg/scratch/UKBiobank/phenotype_files/</p>
<p><strong>Location of the files on Columbias's cluster:</strong></p>
<p><strong>Results:</strong> /mnt/mfs/statgen/UKBiobank/results</p>

</div>
</div>
</div>
<hr>
&copy 2020 Diana Cornejo at Center for Statistical Genetics, Sergievsky Center, Columbia University VP&S
<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

</div>

<script>
// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
