<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=workflowDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=workflowArray;
            var docs_map=workflowArrayMap;
            var pos=workflowArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Pleiotropy analysis in UKBB</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Pleiotropy analysis in UKBB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../workflow.html">Workflow</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/dianacornejo/pleiotropy_UKB"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="BOLT-LMM-analyses-for-UK-Biobank-data">BOLT-LMM analyses for UK Biobank data<a class="anchor-link" href="#BOLT-LMM-analyses-for-UK-Biobank-data">&#182;</a></h1><p>For binary and quantitative traits.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Aim">Aim<a class="anchor-link" href="#Aim">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>To perfom genetic association analysis for quantitative traits (waist-to-hip ratio, waist circumference, BMI, blood lipids) using BOLT-LMM software and UK Biobank imputed data of ~500K invidivuals</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Method-and-workflow-overview">Method and workflow overview<a class="anchor-link" href="#Method-and-workflow-overview">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<ol>
<li>Download data from UKB: phenotype, genotypes files (<code>.fam</code>,<code>.bed</code>, <code>.bim</code>) and imputed genotypes (<code>.bgen</code>, <code>.bgi</code>, <code>.sample</code>)</li>
<li>Install BOLT-LMM in Yale's Farnam cluster. To install for source code follow instructions here: <a href="https://data.broadinstitute.org/alkesgroup/BOLT-LMM/#x1-70002.2">https://data.broadinstitute.org/alkesgroup/BOLT-LMM/#x1-70002.2</a></li>
<li>Run BOLT-LMM analysis to obtain summary statistics for association analysis</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Input-data">Input data<a class="anchor-link" href="#Input-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<ol>
<li>Genotype file for constructing the GRM (genetic relationship matrix) formated as a plink binary file <code>(.bed/.bim/.fam)</code> <code>--bfile=prefix</code></li>
<li>Reference genetic maps <code>--geneticMapFile=tables/genetic_map_hg##.txt.gz</code></li>
<li>Imputed genotype dosages in <code>.bgen</code> format specify the options with  <code>--bgenFile</code> and <code>--sampleFile</code></li>
<li>Phenotype file (white space delimited file with column headers, first two columns should be FID and IID) specify files by options <code>--phenoFile</code> and the phenotype to be analized by <code>--phenoCol</code></li>
<li>Covariates file (same format as phenoFile) specify them by <code>--covarFile</code> for qualitative covariates use <code>--covarCol</code> and for quantitative <code>--qCovarCol</code>. Use <code>--covarMaxLevels</code> to specify the number of categories of a qualitative covariate. To specify an array of quantitative covariates use <code>--qCovarCol=PC{1:20}</code></li>
</ol>
<p>Note: reference genome used <strong>GRCh37/hg19</strong></p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Output-data">Output data<a class="anchor-link" href="#Output-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>For a complete description on bolt commands go to: <a href="http://manpages.ubuntu.com/manpages/eoan/en/man1/bolt.1.html">http://manpages.ubuntu.com/manpages/eoan/en/man1/bolt.1.html</a></p>

<pre><code>--statsFile arg
  output file for assoc stats at PLINK genotypes

--statsFileBgenSnps arg
  output file for assoc stats at BGEN-format genotypes</code></pre>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Global-parameter-setting">Global parameter setting<a class="anchor-link" href="#Global-parameter-setting">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="p">[</span><span class="k">global</span><span class="p">]</span>
<span class="c1"># the output directory for generated files</span>
<span class="kn">parameter:</span> <span class="n">cwd</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="s1">&#39;~/scratch60/2020-04_bolt/INT-WHR&#39;</span><span class="p">)</span>
<span class="c1"># Path to sample file</span>
<span class="kn">parameter:</span> <span class="n">sampleFile</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="s1">&#39;/SAY/dbgapstg/scratch/UKBiobank/genotype_files/ukb39554_imputeddataset/ukb32285_imputedindiv.sample&#39;</span><span class="p">)</span>
<span class="c1"># Genotype files in plink binary this is used for computing the GRM</span>
<span class="kn">parameter:</span> <span class="n">bfile</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="s1">&#39;/SAY/dbgapstg/scratch/UKBiobank/genotype_files/pleiotropy_geneticfiles/UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv&#39;</span><span class="p">)</span>
<span class="c1"># Path to bgen files </span>
<span class="kn">parameter:</span> <span class="n">bgenFile</span> <span class="o">=</span> <span class="n">paths</span><span class="p">([</span><span class="sa">f</span><span class="s1">&#39;/SAY/dbgapstg/scratch/UKBiobank/genotype_files/ukb39554_imputeddataset/ukb_imp_chr</span><span class="si">{</span><span class="n">x</span><span class="o">+</span><span class="mi">1</span><span class="si">}</span><span class="s1">_v3.bgen&#39;</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">22</span><span class="p">)])</span>
<span class="c1"># Phenotype file for quantitative trait (BMI)</span>
<span class="kn">parameter:</span> <span class="n">phenoFile</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="s1">&#39;/SAY/dbgapstg/scratch/UKBiobank/phenotype_files/pleiotropy_R01/phenotypesforanalysis/&#39;</span><span class="p">)</span> 
<span class="c1"># Phenotype to be analyzed (specify the column)</span>
<span class="kn">parameter:</span> <span class="n">phenoCol</span> <span class="o">=</span> <span class="s1">&#39;rankNorm_WHR&#39;</span>
<span class="c1"># Covariate file path</span>
<span class="kn">parameter:</span> <span class="n">covarFile</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="s1">&#39;/SAY/dbgapstg/scratch/UKBiobank/phenotype_files/pleiotropy_R01/phenotypesforanalysis/UKB_caucasians_BMIwaisthip_AsthmaAndT2D_withagesex_033120&#39;</span><span class="p">)</span>
<span class="c1"># Qualitative covariates to be used in the analysis</span>
<span class="kn">parameter:</span> <span class="n">covarCol1</span> <span class="o">=</span> <span class="s1">&#39;SEX&#39;</span>
<span class="c1"># Maximum categories of covariates allowed </span>
<span class="kn">parameter:</span> <span class="n">covarMaxLevels</span> <span class="o">=</span> <span class="mi">10</span>
<span class="c1"># Quantitative covariates to be used in the analysis</span>
<span class="kn">parameter:</span> <span class="n">qCovarCol2</span> <span class="o">=</span> <span class="s1">&#39;AGE&#39;</span>
<span class="c1"># Quantitative covariates to be used in the analysis</span>
<span class="kn">parameter:</span> <span class="n">qCovarCol3</span> <span class="o">=</span> <span class="s1">&#39;BMI&#39;</span>
<span class="c1">#parameter: qCovarCol3 = &#39;PC{1:10}&#39; # if we are going to use PC as covariates uncomment</span>
<span class="c1"># Path to LDscore file for european population</span>
<span class="kn">parameter:</span> <span class="n">LDscoresFile</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="s1">&#39;~/software/BOLT-LMM_v2.3.4/tables/LDSCORE.1000G_EUR.tab.gz&#39;</span><span class="p">)</span>
<span class="c1"># Path to genetic map file used to interpolate genetic map coordinates from SNP physical (base pair) positions</span>
<span class="kn">parameter:</span> <span class="n">geneticMapFile</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="s1">&#39;~/software/BOLT-LMM_v2.3.4/tables/genetic_map_hg19_withX.txt.gz&#39;</span><span class="p">)</span>
<span class="c1"># Specific number of threads to use</span>
<span class="kn">parameter:</span> <span class="n">numThreads</span> <span class="o">=</span> <span class="mi">20</span>
<span class="c1"># Minimum MAF to be used</span>
<span class="kn">parameter:</span> <span class="n">bgenMinMAF</span> <span class="o">=</span> <span class="mf">0.001</span>
<span class="c1"># Mimimum info score to be used</span>
<span class="kn">parameter:</span> <span class="n">bgenMinINFO</span> <span class="o">=</span> <span class="mf">0.8</span>
<span class="c1"># For cluster jobs, number commands to run per job</span>
<span class="kn">parameter:</span> <span class="n">job_size</span> <span class="o">=</span> <span class="mi">1</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Running-BOLT-LMM">Running BOLT-LMM<a class="anchor-link" href="#Running-BOLT-LMM">&#182;</a></h2><p>On Yale Farnam cluster,</p>
<ul>
<li>INT-BMI with age and sex as covariates</li>
</ul>

<pre><code>PHEN_DIR=~/project/phenotypes_UKB/
sos run ~/project/pleiotropy_UKB/worfkflow/BoltLMM.ipynb bolt -c ~/project/pleiotropy_UKB/farnam.yml -q farnam -J 40 \
   --phenoFile $PHEN_DIR/UKB_caucasians_BMIwaisthip_AsthmaAndT2D_INT-BMI_withagesex_041720 -s build &amp;&gt; sos-submission-INT-BMI.log</code></pre>
<ul>
<li>INT-WAIST with age, sex and BMI as covariates</li>
</ul>

<pre><code>PHEN_DIR=~/project/phenotypes_UKB/
sos run ~/project/pleiotropy_UKB/worfkflow/BoltLMM.ipynb bolt -c ~/project/pleiotropy_UKB/farnam.yml -q farnam -J 40 \
   --phenoFile $PHEN_DIR/UKB_caucasians_BMIwaisthip_AsthmaAndT2D_INT-WAIST_withagesex_042020 -s build &amp;&gt; sos-submission-INT-WAIST.log</code></pre>
<ul>
<li>INT-WHR with age, sex and BMI as covariates</li>
</ul>

<pre><code>PHEN_DIR=~/project/phenotypes_UKB/
sos run ~/project/pleiotropy_UKB/worfkflow/BoltLMM.ipynb bolt -c ~/project/pleiotropy_UKB/farnam.yml -q farnam -J 40 \
   --phenoFile $PHEN_DIR/UKB_caucasians_BMIwaisthip_AsthmaAndT2D_INT-WHR_withagesex_042020 -s build &amp;&gt; sos-submission-INT-WHR.log</code></pre>
<p>On a local computer with for example 8 threads,</p>

<pre><code>sos run BoltLMM.ipynb bolt -q none -j 8</code></pre>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Notes-about-how-the-analyses-where-done">Notes about how the analyses where done<a class="anchor-link" href="#Notes-about-how-the-analyses-where-done">&#182;</a></h3><ol>
<li>Run INT-BMI with covariates AGE and SEX</li>
<li>Run INT-WHR with covariates AGE and SEX, repeated on 05-07-2020 with BMI as covariate</li>
<li>Run INT-WAIST with covariates AGE, SEX and BMI</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Workflow-implementation-details">Workflow implementation details<a class="anchor-link" href="#Workflow-implementation-details">&#182;</a></h2><p><strong>A note for developers</strong>: it is important to have input and output for each step. Input files and output files are best derived from one another.</p>
<p>BOLT-LMM software computes statistics for testing association between phenotypes and genotypes using a linear mixed model</p>

<pre><code>--bfile = accepts genotype files in PLINK binary format (.fam, .bed, .bim)
--geneticMapFile = Oxford-format file for interpolating genetic distances: tables/genetic_map_hg##.txt.gz
--phenoFile = phenotype file (header required; FII and IID must be first two columns)
--phenoCol = phenotype columns header
--covarFile = covariate file (header required; FII and IID must be first two columns)
--covarCol = categorical covariate column(s); for &gt;1, use multiple --covarCol and/or {i:j} expansion
--qcovarCol = quantitative covariate column(s); for  &gt;1, use multiple --qCovarCol and/or {i:j} expansion
--lmm = compute assoc stats under the inf model and with Bayesian non-inf prior (VB approx), if power gain expected
--modelSnps = file(s) listing SNPs to use in model (i.e., GRM) (default: use all non-excluded SNPs)
--LDscoresFile = LD Scores for calibration of Bayesian assoc stats: tables/LDSCORE.1000G_EUR.tab.g
--numThreads = number of computational threads
--statsFile = output file for assoc stats at PLINK genotypes
--bgenFile = file(s) containing Oxford BGEN-format genotypes to test for association
--sampleFile = file containing Oxford sample file corresponding to BGEN file(s)
--bgenMinMAF = MAF threshold on Oxford BGEN-format genotypes; lower-MAF SNPs will be ignored
--bgenMinINFO = INFO threshold on Oxford BGEN-format genotypes; lower-INFO SNPs will be ignored
--statsFileBgenSNPs = output file for assoc stats at BGEN-format genotypes</code></pre>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="p">[</span><span class="n">bolt</span><span class="p">]</span>
<span class="kn">depends:</span> <span class="n">executable</span><span class="p">(</span><span class="s2">&quot;bolt&quot;</span><span class="p">)</span>
<span class="kn">input:</span> <span class="n">bgenFile</span><span class="p">,</span> <span class="n">group_by</span> <span class="o">=</span> <span class="mi">1</span>
<span class="kn">output:</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">cwd</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">_input</span><span class="si">:</span><span class="s1">bn</span><span class="si">}</span><span class="s1">.</span><span class="si">{</span><span class="n">phenoFile</span><span class="si">:</span><span class="s1">b</span><span class="si">}</span><span class="s1">.boltlmm.snp_stats.gz&#39;</span>
<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">trunk_size</span> <span class="o">=</span> <span class="n">job_size</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;48h&#39;</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;60G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="n">numThreads</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">step_name</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">_output</span><span class="si">:</span><span class="s1">bn</span><span class="si">}</span><span class="s1">&#39;</span>
<span class="kn">bash:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="kp">workdir</span> <span class="o">=</span> <span class="n">cwd</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">_output</span><span class="si">:</span><span class="s1">n</span><span class="si">}</span><span class="s1">.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">_output</span><span class="si">:</span><span class="s1">n</span><span class="si">}</span><span class="s1">.stdout&#39;</span>
 <span class="n">bolt</span> \
    <span class="o">--</span><span class="n">bfile</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">bfile</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">phenoFile</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">phenoFile</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">phenoCol</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">phenoCol</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">covarFile</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">covarFile</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">covarCol</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">covarCol1</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">covarMaxLevels</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">covarMaxLevels</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">qCovarCol</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">qCovarCol2</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">qCovarCol</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">qCovarCol3</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">LDscoresFile</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">LDscoresFile</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">geneticMapFile</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">geneticMapFile</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">lmm</span> \
    <span class="o">--</span><span class="n">statsFile</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">nn</span><span class="p">}</span><span class="o">.</span><span class="n">stats</span><span class="o">.</span><span class="n">gz</span> \
    <span class="o">--</span><span class="n">numThreads</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">numThreads</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">bgenFile</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">bgenMinMAF</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">bgenMinMAF</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">bgenMinINFO</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">bgenMinINFO</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">sampleFile</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">sampleFile</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">statsFileBgenSnps</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">verboseStats</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Results">Results<a class="anchor-link" href="#Results">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="o">%</span><span class="n">cd</span> <span class="o">~/</span><span class="n">scratch60</span><span class="o">/</span><span class="mi">2020</span><span class="o">-</span><span class="mi">04</span><span class="n">_bolt</span>
</pre></div>

    </div>
</div>
</div>

</div>
<hr>
&copy 2020 Diana Cornejo at Center for Statistical Genetics, Sergievsky Center, Columbia University VP&S
<p><small>Exported from <a href="http://github.com/dianacornejo/pleiotropy_UKB/blob/b71252c72a6e33b2d774a2f46829bbafe5593ff6/workflow/BoltLMM.ipynb"><code>workflow/BoltLMM.ipynb</code></a> committed by dianacornejo on Fri May 29 17:26:44 2020 <a href="http://github.com/dianacornejo/pleiotropy_UKB/commit/b71252c72a6e33b2d774a2f46829bbafe5593ff6">revision 5, b71252c</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
