<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=workflowDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=workflowArray;
            var docs_map=workflowArrayMap;
            var pos=workflowArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Pleiotropy analysis in UKBB</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Pleiotropy analysis in UKBB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../workflow.html">Workflow</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/dianacornejo/pleiotropy_UKB"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Principal-component-analysis">Principal component analysis<a class="anchor-link" href="#Principal-component-analysis">&#182;</a></h1><p>The intention of this notebook is to generate the PCA analysis and plots for the exomed samples 200K.</p>
<p>Steps to generate a PCA include removing related individuals, pruning variants in linkage disequilibrium (LD), and excluding outlier samples that can suggest poor genotyping quality or distant relatedness (also restrict to individuals of homogeneous ancestry).</p>
<p>Pitfalls</p>
<ol>
<li>Some of the PCs may capture LD structure rather than population structure (decrease in power to detect associations in these regions of high LD)</li>
<li>When projecting a new study dataset to the PCA space computed from a reference dataset: projected PCs are shrunk toward 0 in the new dataset </li>
<li>PC scores may capture outliers that are due to family structure, population structure or other reasons; it might be beneficial to detect and remove these individuals to maximize the population structure captured by PCA (in the case of removing a few outliers) or to restrict analyses to genetically homogeneous samples</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Command-interface">Command interface<a class="anchor-link" href="#Command-interface">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">sos</span> <span class="n">run</span> <span class="n">PCA</span><span class="o">.</span><span class="n">ipynb</span> <span class="o">-</span><span class="n">h</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>usage: sos run PCA.ipynb [workflow_name | -t targets] [options] [workflow_options]
  workflow_name:        Single or combined workflows defined in this script
  targets:              One or more targets to generate
  options:              Single-hyphen sos parameters (see &#34;sos run -h&#34; for details)
  workflow_options:     Double-hyphen workflow-specific parameters

Workflows:
  filter_samples
  filter
  pca
  flashpca

Global Workflow Options:
  --cwd VAL (as path, required)
                        the output directory for generated files
  --genoFile  paths

                        Plink binary files
  --famFile VAL (as path, required)
                        The fam file associated to the bed files
  --database VAL (as path, required)
                        The database to extract info from
  --ethnia-prefix VAL (as str, required)
                        The prefix for the output files
  --phenoFile  path(f&#39;{cwd}/cache/{famFile:bn}.{ethnia_prefix}.pheno&#39;)

                        The phenotypic file
  --job-size 1 (as int)
                        For cluster jobs, number commands to run per job
  --numThreads 1 (as int)
                        Number of threads
  --eigensoft-module &#39;\nmodule load EIGENSOFT/7.2.1-foss-2018b\necho &#34;Module Eigensoft v.7.2.1 loaded&#34;\n{cmd}\n&#39;
                        Load Eigensoft module from cluster
  --container-lmm &#39;statisticalgenetics/lmm:1.7&#39;
                        Software container option

Sections
  filter_samples:       Filter individuals from ancestries different than
                        British, Irish, Other white background, prefer not to
                        answer, do not know
    Workflow Options:
      --select-ethnia  (as list)
  filter_1:             Filter SNPs with MAF&gt;1% for PCA analysis, select
                        individuals and merge bed into one file
    Workflow Options:
      --maf-filter 0.01 (as float)
      --geno-filter 0.01 (as float)
                        Maximum missingess per-variant
      --mind-filter 0.02 (as float)
                        Maximum missingness per-sample
      --hwe-filter 0.0 (as float)
  filter_2:             Merge all the .bed files into one bed file for input to
                        eigensoft
  filter_3:             LD prunning window=50, shift-window every 5 SNPS, r2=0.5
    Workflow Options:
      --window 50 (as int)
      --shift 5 (as int)
      --r2 0.5 (as float)
  pca_1:                Run pca analysis using Eigenstrat: the program suports
                        plink files here called PACKEDPED format smartpca.perl:
                        run PCA on input genotype data (calls smartpca)
    Workflow Options:
      --k VAL (as int, required)
                        Number of Principal Components to output
      --maxiter 0 (as int)
                        Maximum number of iterations for outlier removal.
                        Default 0 turns off outlier removal
      --topk 10 (as int)
                        Number of principal components along which to remove
                        outliers during each outlier removal iteration. Default
                        is 10
      --sigma 6 (as int)
                        Number of standard deviations which an individual must
                        exceed, along one of topk top principal components, in
                        order to be removed as an outlier. Default is 6
  flashpca:             Run PCA analysis using flashpca
    Workflow Options:
      --k VAL (as int, required)
                        Number of Principal Components to output. Default is 10
      --trait-name VAL (as str, required)
                        Name of the trait in the phenoFile (format FID, IID,
                        father, mother,trait )
      --stand binom2
                        How to standardize X before PCA

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="PCA-analysis-pipeline">PCA analysis pipeline<a class="anchor-link" href="#PCA-analysis-pipeline">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="p">[</span><span class="k">global</span><span class="p">]</span>
<span class="c1"># the output directory for generated files</span>
<span class="kn">parameter:</span> <span class="n">cwd</span> <span class="o">=</span> <span class="n">path</span>
<span class="c1"># Plink binary files</span>
<span class="kn">parameter:</span> <span class="n">genoFile</span> <span class="o">=</span> <span class="n">paths</span>
<span class="c1"># The fam file associated to the bed files</span>
<span class="kn">parameter:</span> <span class="n">famFile</span> <span class="o">=</span> <span class="n">path</span> 
<span class="c1"># The database to extract info from</span>
<span class="kn">parameter:</span> <span class="n">database</span> <span class="o">=</span> <span class="n">path</span>
<span class="c1"># The prefix for the output files</span>
<span class="kn">parameter:</span> <span class="n">ethnia_prefix</span> <span class="o">=</span> <span class="nb">str</span>
<span class="c1"># The phenotypic file</span>
<span class="kn">parameter:</span> <span class="n">phenoFile</span> <span class="o">=</span> <span class="n">path</span><span class="p">(</span><span class="n">f</span><span class="s1">&#39;{cwd}/cache/{famFile:bn}.{ethnia_prefix}.pheno&#39;</span><span class="p">)</span>
<span class="c1"># For cluster jobs, number commands to run per job</span>
<span class="kn">parameter:</span> <span class="n">job_size</span> <span class="o">=</span> <span class="mi">1</span>
<span class="c1"># Number of threads</span>
<span class="kn">parameter:</span> <span class="n">numThreads</span> <span class="o">=</span> <span class="mi">1</span>
<span class="c1"># Merge data</span>
<span class="kn">parameter:</span> <span class="n">merge</span> <span class="o">=</span> <span class="bp">True</span>
<span class="c1"># Load Eigensoft module from cluster</span>
<span class="kn">parameter:</span> <span class="n">eigensoft_module</span> <span class="o">=</span> <span class="s1">&#39;&#39;&#39;</span>
<span class="s1">module load EIGENSOFT/7.2.1-foss-2018b</span>
<span class="s1">echo &quot;Module Eigensoft v.7.2.1 loaded&quot;</span>
<span class="s1">{cmd}</span>
<span class="s1">&#39;&#39;&#39;</span>
<span class="c1"># Software container option</span>
<span class="kn">parameter:</span> <span class="n">container_lmm</span> <span class="o">=</span> <span class="s1">&#39;statisticalgenetics/lmm:1.8&#39;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Filter individuals depending on their ancestry</span>
<span class="p">[</span><span class="n">filter_samples</span><span class="p">]</span>
<span class="kn">parameter:</span> <span class="n">select_ethnia</span> <span class="o">=</span> <span class="p">[</span> <span class="p">]</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s1">&#39;{cwd}/cache/{famFile:bn}.{ethnia_prefix}&#39;</span><span class="p">,</span> <span class="n">phenoFile</span>
<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;10h&#39;</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;40G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="n">numThreads</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{step_name}_{_output[0]:bn}&#39;</span>
<span class="kn">R:</span> <span class="n">container</span><span class="o">=</span><span class="n">container_lmm</span><span class="p">,</span> <span class="n">expand</span><span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]:nn}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]:nn}.stdout&#39;</span>
    <span class="c1">#Load libraries</span>
    <span class="n">library</span><span class="p">(</span><span class="s1">&#39;dplyr&#39;</span><span class="p">)</span>
    <span class="c1"># This database is the one from June 2020 and contains a subset of variables with the PCs</span>
    <span class="n">fam</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">famFile</span><span class="p">:</span><span class="n">r</span><span class="p">},</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39; &#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="n">F</span><span class="p">)</span>
    <span class="n">colnames</span><span class="p">(</span><span class="n">fam</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;FID&quot;</span><span class="p">,</span><span class="s2">&quot;IID&quot;</span><span class="p">,</span><span class="s2">&quot;fatherID&quot;</span><span class="p">,</span> <span class="s2">&quot;motherID&quot;</span><span class="p">,</span> <span class="s2">&quot;sex&quot;</span><span class="p">,</span> <span class="s2">&quot;phenotype&quot;</span><span class="p">)</span>
    <span class="n">cat</span><span class="p">(</span><span class="s2">&quot;There are&quot;</span><span class="p">,</span><span class="n">nrow</span><span class="p">(</span><span class="n">fam</span><span class="p">),</span><span class="s2">&quot;individuals with exomes.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">bd</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">database</span><span class="p">:</span><span class="n">r</span><span class="p">},</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="n">T</span><span class="p">)</span>
    <span class="n">cat</span><span class="p">(</span><span class="s2">&quot;The size of the full database is&quot;</span><span class="p">,</span><span class="n">dim</span><span class="p">(</span><span class="n">bd</span><span class="p">),</span><span class="s2">&quot;.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="c1"># Assign individual ID column to bd f.eid</span>
    <span class="n">names</span><span class="p">(</span><span class="n">bd</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s2">&quot;IID&quot;</span>
    <span class="c1"># Select the 200K individuals with exomes from the full db</span>
    <span class="n">exomed_IID</span> <span class="o">&lt;-</span> <span class="n">bd</span><span class="p">[</span><span class="n">bd</span><span class="err">$</span><span class="n">IID</span> <span class="o">%</span><span class="ow">in</span><span class="o">%</span> <span class="n">fam</span><span class="err">$</span><span class="n">IID</span><span class="p">,]</span>
    <span class="n">cat</span><span class="p">(</span><span class="s2">&quot;The number of selected individuals is&quot;</span><span class="p">,</span><span class="n">nrow</span><span class="p">(</span><span class="n">exomed_IID</span><span class="p">),</span><span class="s2">&quot;.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="c1"># Filter db based on ethnicity variable</span>
    <span class="n">ethnicity</span> <span class="o">&lt;-</span> <span class="n">exomed_IID</span> <span class="o">%&gt;%</span>
          <span class="n">select</span><span class="p">(</span><span class="n">IID</span><span class="p">,</span> <span class="n">starts_with</span><span class="p">(</span><span class="s2">&quot;f.21000&quot;</span><span class="p">))</span>
    <span class="n">ethnicity</span> <span class="o">&lt;-</span> <span class="n">ethnicity</span> <span class="o">%&gt;%</span>
          <span class="n">mutate_all</span><span class="p">(</span><span class="n">na_if</span><span class="p">,</span><span class="o">-</span><span class="mi">3</span><span class="p">)</span> <span class="o">%&gt;%</span>
          <span class="n">mutate_all</span><span class="p">(</span><span class="n">na_if</span><span class="p">,</span><span class="o">-</span><span class="mi">1</span><span class="p">)</span>
    <span class="c1"># Function to extract all the available answers for 3 visits and put them in one list</span>
    <span class="n">f</span><span class="o">&lt;-</span><span class="n">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
      <span class="n">visit</span><span class="o">&lt;-</span><span class="n">c</span><span class="p">()</span>
      <span class="k">for</span> <span class="p">(</span><span class="n">i</span> <span class="ow">in</span> <span class="mi">2</span><span class="p">:</span><span class="mi">4</span><span class="p">){</span>
        <span class="k">if</span> <span class="p">(</span><span class="err">!</span><span class="ow">is</span><span class="o">.</span><span class="n">na</span><span class="p">(</span><span class="n">x</span><span class="p">[</span><span class="n">i</span><span class="p">]))</span>
        <span class="p">{</span><span class="n">visit</span><span class="o">&lt;-</span><span class="n">c</span><span class="p">(</span><span class="n">visit</span><span class="p">,</span><span class="n">x</span><span class="p">[</span><span class="n">i</span><span class="p">])}</span>
      <span class="p">}</span>
      <span class="k">if</span><span class="p">(</span><span class="ow">is</span><span class="o">.</span><span class="n">null</span><span class="p">(</span><span class="n">visit</span><span class="p">)){</span><span class="n">visit</span><span class="o">=-</span><span class="mi">99</span><span class="p">}</span>
      <span class="k">else</span><span class="p">{</span><span class="n">visit</span><span class="o">=</span><span class="k">as</span><span class="o">.</span><span class="n">numeric</span><span class="p">(</span><span class="n">visit</span><span class="p">)}</span>
      <span class="k">return</span> <span class="p">(</span><span class="n">visit</span><span class="p">)</span>
    <span class="p">}</span>

    <span class="c1"># Apply the above function and remove NAs</span>
    <span class="n">ethnicity</span><span class="err">$</span><span class="n">visit</span><span class="o">&lt;-</span><span class="nb">apply</span><span class="p">(</span><span class="n">ethnicity</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="n">f</span><span class="p">)</span>
    <span class="c1"># Filter out individuals wih missing values in ethnicity: 212 ind total</span>
    <span class="n">ethnicity</span> <span class="o">&lt;-</span> <span class="n">ethnicity</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="err">!</span><span class="ow">is</span><span class="o">.</span><span class="n">na</span><span class="p">(</span><span class="n">visit</span><span class="p">))</span>
    <span class="n">cat</span><span class="p">(</span><span class="s2">&quot;There are&quot;</span><span class="p">,</span><span class="n">nrow</span><span class="p">(</span><span class="n">ethnicity</span><span class="p">),</span><span class="s2">&quot;individuals without missing values for ethnicity.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="c1"># Identify the unique available codings in f.21000</span>
    <span class="n">code</span><span class="o">&lt;-</span><span class="n">union</span><span class="p">(</span><span class="n">union</span><span class="p">(</span><span class="n">unique</span><span class="p">(</span><span class="n">ethnicity</span><span class="err">$</span><span class="n">f</span><span class="o">.</span><span class="mf">21000.0</span><span class="o">.</span><span class="mi">0</span><span class="p">),</span><span class="n">unique</span><span class="p">(</span><span class="n">ethnicity</span><span class="err">$</span><span class="n">f</span><span class="o">.</span><span class="mf">21000.1</span><span class="o">.</span><span class="mi">0</span><span class="p">)),</span><span class="n">unique</span><span class="p">(</span><span class="n">ethnicity</span><span class="err">$</span><span class="n">f</span><span class="o">.</span><span class="mf">21000.2</span><span class="o">.</span><span class="mi">0</span><span class="p">))</span>
    <span class="c1"># Codes to keep white individuals</span>
    <span class="c1">#useful_code&lt;-c(1001,1002,1003,1,-3,-1)</span>
    <span class="n">useful_code</span><span class="o">&lt;-</span><span class="n">c</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="s1">&#39;,&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> &#39;</span> <span class="o">%</span> <span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">select_ethnia</span> <span class="k">if</span> <span class="n">x</span> <span class="ow">is</span> <span class="ow">not</span> <span class="bp">None</span><span class="p">])})</span>
    <span class="c1"># the rest that don’t have the combinations above can be set as NA</span>
    <span class="n">useless_code</span><span class="o">&lt;-</span><span class="n">code</span><span class="p">[</span><span class="err">!</span><span class="n">code</span> <span class="o">%</span><span class="ow">in</span><span class="o">%</span> <span class="n">useful_code</span><span class="p">]</span> 
    <span class="n">useless_code</span><span class="o">&lt;-</span><span class="n">useless_code</span><span class="p">[</span><span class="o">-</span><span class="n">which</span><span class="p">(</span><span class="ow">is</span><span class="o">.</span><span class="n">na</span><span class="p">(</span><span class="n">useless_code</span><span class="p">))]</span> <span class="c1"># remove NA here in the vector</span>
    <span class="c1"># Function to get the final code for ethnicity</span>
    <span class="n">f</span><span class="o">&lt;-</span><span class="n">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
      <span class="n">l</span><span class="o">=</span><span class="n">length</span><span class="p">(</span><span class="n">unique</span><span class="p">(</span><span class="n">x</span><span class="err">$</span><span class="n">visit</span><span class="p">))</span>
      <span class="k">if</span> <span class="p">(</span><span class="n">l</span><span class="o">==</span><span class="mi">1</span><span class="p">){</span> <span class="c1"># only one value available</span>
        <span class="n">result</span><span class="o">=</span><span class="n">unique</span><span class="p">(</span><span class="n">x</span><span class="err">$</span><span class="n">visit</span><span class="p">)</span>
      <span class="p">}</span>
      <span class="k">else</span><span class="p">{</span> <span class="c1"># more then one value available</span>
        <span class="n">l</span><span class="o">=</span><span class="n">length</span><span class="p">(</span><span class="n">x</span><span class="err">$</span><span class="n">visit</span><span class="p">)</span>
        <span class="k">for</span> <span class="p">(</span><span class="n">i</span> <span class="ow">in</span> <span class="mi">1</span><span class="p">:</span><span class="n">l</span><span class="p">){</span>
          <span class="k">if</span> <span class="p">(</span><span class="n">x</span><span class="err">$</span><span class="n">visit</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">%</span><span class="ow">in</span><span class="o">%</span> <span class="n">useless_code</span><span class="p">){</span><span class="n">result</span><span class="o">=</span><span class="mi">8000</span><span class="p">;</span> <span class="k">break</span><span class="p">}</span> <span class="c1"># inconsistent ones with conbination not wanted</span>
          <span class="k">else</span> <span class="p">{</span><span class="n">result</span><span class="o">=</span><span class="mi">9000</span><span class="p">}</span> <span class="c1"># inconsistent ones with right conbination</span>
        <span class="p">}</span>
      <span class="p">}</span>
      <span class="k">return</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
    <span class="p">}</span>

    <span class="c1"># Apply the above function and remove NAs</span>
    <span class="n">ethnicity</span><span class="err">$</span><span class="n">new_ethnicity</span><span class="o">&lt;-</span><span class="nb">apply</span><span class="p">(</span><span class="n">ethnicity</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="n">f</span><span class="p">)</span>
    <span class="c1"># Filter by NA presence</span>
    <span class="n">ethnicity_noNA</span><span class="o">&lt;-</span><span class="n">ethnicity</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="err">!</span><span class="ow">is</span><span class="o">.</span><span class="n">na</span><span class="p">(</span><span class="n">new_ethnicity</span><span class="p">))</span>
    <span class="n">cat</span><span class="p">(</span><span class="s2">&quot;There are&quot;</span><span class="p">,</span><span class="n">nrow</span><span class="p">(</span><span class="n">ethnicity_noNA</span><span class="p">),</span><span class="s2">&quot;individuals consistent for f.21000.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">ethnicity_isNA</span> <span class="o">&lt;-</span> <span class="n">ethnicity</span> <span class="o">%&gt;%</span>
      <span class="nb">filter</span><span class="p">(</span><span class="ow">is</span><span class="o">.</span><span class="n">na</span><span class="p">(</span><span class="n">new_ethnicity</span><span class="p">))</span>
    <span class="n">cat</span><span class="p">(</span><span class="s2">&quot;There are&quot;</span><span class="p">,</span><span class="n">nrow</span><span class="p">(</span><span class="n">ethnicity_isNA</span><span class="p">),</span><span class="s2">&quot;individuals inconsistent for f.21000.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="c1"># keep only certain individuals</span>
    <span class="err">$</span><span class="p">{</span><span class="n">ethnia_prefix</span><span class="p">}</span> <span class="o">&lt;-</span> <span class="n">ethnicity_noNA</span> <span class="o">%&gt;%</span>
        <span class="nb">filter</span><span class="p">(</span><span class="n">new_ethnicity</span> <span class="o">%</span><span class="ow">in</span><span class="o">%</span> <span class="n">useful_code</span><span class="p">)</span> <span class="o">%&gt;%</span>
        <span class="n">mutate</span><span class="p">(</span><span class="n">FID</span> <span class="o">=</span> <span class="n">IID</span><span class="p">)</span> <span class="o">%&gt;%</span>
        <span class="n">select</span><span class="p">(</span><span class="n">FID</span><span class="p">,</span><span class="n">IID</span><span class="p">)</span>
    <span class="n">cat</span><span class="p">(</span><span class="s2">&quot;After excluding ethnic backgrounds different than  ${ethnia_prefix}, the number of individuals is&quot;</span><span class="p">,</span><span class="n">nrow</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">ethnia_prefix</span><span class="p">}),</span><span class="s2">&quot;.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="c1"># Write the selected individuals to a txt file</span>
    <span class="n">write</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">ethnia_prefix</span><span class="p">},</span><span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">row</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">col</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">F</span><span class="p">)</span>
    <span class="c1"># Create the phenotype file</span>
    <span class="n">pheno</span> <span class="o">&lt;-</span> <span class="n">ethnicity_noNA</span> <span class="o">%&gt;%</span>
        <span class="nb">filter</span><span class="p">(</span><span class="n">new_ethnicity</span> <span class="o">%</span><span class="ow">in</span><span class="o">%</span> <span class="n">useful_code</span><span class="p">)</span> <span class="o">%&gt;%</span>
        <span class="n">mutate</span><span class="p">(</span><span class="n">ethnicity</span> <span class="o">=</span> <span class="n">new_ethnicity</span><span class="p">)</span> <span class="o">%&gt;%</span>
        <span class="n">select</span><span class="p">(</span><span class="n">IID</span><span class="p">,</span><span class="n">ethnicity</span><span class="p">)</span>
    <span class="c1"># Merge the two data frames</span>
    <span class="n">famfile</span> <span class="o">&lt;-</span><span class="n">merge</span><span class="p">(</span><span class="n">fam</span><span class="p">,</span> <span class="n">pheno</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span><span class="s2">&quot;IID&quot;</span><span class="p">,</span> <span class="nb">all</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span>
    <span class="n">cat</span><span class="p">(</span><span class="s2">&quot;The famfile has &quot;</span><span class="p">,</span><span class="n">nrow</span><span class="p">(</span><span class="n">famfile</span><span class="p">),</span><span class="s2">&quot;individuals.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">write</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="n">famfile</span><span class="p">,</span><span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">1</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">row</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">col</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">F</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Filter SNPs with MAF&gt;1% for PCA analysis, select individuals and merge bed into one file</span>
<span class="p">[</span><span class="n">filter_1</span><span class="p">]</span>
<span class="kn">parameter:</span> <span class="n">maf_filter</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="c1">#Maximum missingess per-variant</span>
<span class="kn">parameter:</span> <span class="n">geno_filter</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="c1">#Maximum missingness per-sample</span>
<span class="kn">parameter:</span> <span class="n">mind_filter</span> <span class="o">=</span> <span class="mf">0.02</span>
<span class="kn">parameter:</span> <span class="n">hwe_filter</span> <span class="o">=</span> <span class="mf">0.0</span>
<span class="kn">input:</span> <span class="n">genoFile</span><span class="p">,</span> <span class="n">group_by</span><span class="o">=</span><span class="mi">1</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s1">&#39;{cwd}/cache/{_input:bn}.filtered.bed&#39;</span>
<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;10h&#39;</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;30G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="n">numThreads</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{step_name}_{_output:bn}&#39;</span>
<span class="kn">bash:</span> <span class="n">container</span><span class="o">=</span><span class="n">container_lmm</span><span class="p">,</span> <span class="n">expand</span><span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:n}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:n}.stdout&#39;</span>
    <span class="n">plink2</span> \
      <span class="o">--</span><span class="n">bfile</span> <span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">:</span><span class="n">n</span><span class="p">}</span> \
      <span class="err">$</span><span class="p">{(</span><span class="s1">&#39;--maf </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">maf_filter</span><span class="p">)</span> <span class="k">if</span> <span class="n">maf_filter</span> <span class="o">&gt;</span> <span class="mi">0</span> <span class="k">else</span> <span class="s1">&#39;&#39;</span><span class="p">}</span> <span class="err">$</span><span class="p">{(</span><span class="s1">&#39;--geno </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">geno_filter</span><span class="p">)</span> <span class="k">if</span> <span class="n">geno_filter</span> <span class="o">&gt;</span> <span class="mi">0</span> <span class="k">else</span> <span class="s1">&#39;&#39;</span><span class="p">}</span> <span class="err">$</span><span class="p">{(</span><span class="s1">&#39;--hwe </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">hwe_filter</span><span class="p">)</span> <span class="k">if</span> <span class="n">hwe_filter</span> <span class="o">&gt;</span> <span class="mi">0</span> <span class="k">else</span> <span class="s1">&#39;&#39;</span><span class="p">}</span> <span class="err">$</span><span class="p">{(</span><span class="s1">&#39;--mind </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mind_filter</span><span class="p">)</span> <span class="k">if</span> <span class="n">mind_filter</span> <span class="o">&gt;</span> <span class="mi">0</span> <span class="k">else</span> <span class="s1">&#39;&#39;</span><span class="p">}</span> \
      <span class="o">--</span><span class="n">keep</span> <span class="err">$</span><span class="p">{</span><span class="n">cwd</span><span class="p">}</span><span class="o">/</span><span class="n">cache</span><span class="o">/</span><span class="err">$</span><span class="p">{</span><span class="n">famFile</span><span class="p">:</span><span class="n">bn</span><span class="p">}</span><span class="o">.</span><span class="err">$</span><span class="p">{</span><span class="n">ethnia_prefix</span><span class="p">}</span> \
      <span class="o">--</span><span class="n">make</span><span class="o">-</span><span class="n">bed</span> \
      <span class="o">--</span><span class="n">threads</span> <span class="err">$</span><span class="p">{</span><span class="n">numThreads</span><span class="p">}</span> \
      <span class="o">--</span><span class="n">out</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">n</span><span class="p">}</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Merge all the .bed files into one bed file for input to eigensoft</span>
<span class="p">[</span><span class="n">filter_2</span><span class="p">:</span><span class="n">skip</span><span class="o">=</span><span class="ow">not</span> <span class="n">merge</span><span class="p">]</span>
<span class="kn">input:</span> <span class="n">group_by</span> <span class="o">=</span> <span class="s1">&#39;all&#39;</span>
<span class="kn">output:</span> <span class="n">bfile_merge</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{cwd}/cache/{famFile:bn}.filtered.merged.bed&#39;</span>
<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">trunk_size</span> <span class="o">=</span> <span class="n">job_size</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;48h&#39;</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;60G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="n">numThreads</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{step_name}_{_output:bn}&#39;</span>
<span class="kn">bash:</span> <span class="n">container</span><span class="o">=</span><span class="n">container_lmm</span><span class="p">,</span> <span class="n">expand</span><span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:n}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:n}.stdout&#39;</span>
    <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="err">$</span><span class="p">{</span><span class="s1">&#39; &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)[:</span><span class="o">-</span><span class="mi">4</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">_input</span><span class="p">[</span><span class="mi">1</span><span class="p">:]])}</span> <span class="o">|</span> <span class="n">sed</span> <span class="s1">&#39;s/ /</span><span class="se">\n</span><span class="s1">/g&#39;</span> <span class="o">&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">n</span><span class="p">}</span><span class="o">.</span><span class="n">merge_list</span>
    <span class="n">plink</span> \
    <span class="o">--</span><span class="n">bfile</span> <span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">n</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">merge</span><span class="o">-</span><span class="nb">list</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">n</span><span class="p">}</span><span class="o">.</span><span class="n">merge_list</span> \
    <span class="o">--</span><span class="n">make</span><span class="o">-</span><span class="n">bed</span> \
    <span class="o">--</span><span class="n">out</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">n</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">threads</span> <span class="err">$</span><span class="p">{</span><span class="n">numThreads</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">memory</span> <span class="mi">48000</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># LD prunning window=50, shift-window every 10 SNPS, r2=0.1</span>
<span class="p">[</span><span class="n">filter_3</span><span class="p">]</span>
<span class="kn">parameter:</span> <span class="n">window</span> <span class="o">=</span> <span class="mi">50</span>
<span class="kn">parameter:</span> <span class="n">shift</span> <span class="o">=</span> <span class="mi">10</span>
<span class="kn">parameter:</span> <span class="n">r2</span> <span class="o">=</span> <span class="mf">0.1</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s1">&#39;{cwd}/cache/{famFile:bn}.filtered.merged.prune.in&#39;</span><span class="p">,</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{famFile:bn}.filtered.merged.prune.bed&#39;</span>
<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">trunk_size</span> <span class="o">=</span> <span class="n">job_size</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;48h&#39;</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;60G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="n">numThreads</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{step_name}_{_output[0]:bn}&#39;</span>
<span class="kn">bash:</span> <span class="n">container</span><span class="o">=</span><span class="n">container_lmm</span><span class="p">,</span> <span class="n">expand</span><span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]:n}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]:n}.stdout&#39;</span>
    <span class="n">plink</span> \
    <span class="o">--</span><span class="n">bfile</span> <span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">:</span><span class="n">n</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">indep</span><span class="o">-</span><span class="n">pairwise</span> <span class="err">$</span><span class="p">{</span><span class="n">window</span><span class="p">}</span> <span class="err">$</span><span class="p">{</span><span class="n">shift</span><span class="p">}</span> <span class="err">$</span><span class="p">{</span><span class="n">r2</span><span class="p">}</span>  \
    <span class="o">--</span><span class="n">out</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">nn</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">threads</span> <span class="err">$</span><span class="p">{</span><span class="n">numThreads</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">memory</span> <span class="mi">48000</span>
    
    <span class="n">plink</span> \
    <span class="o">--</span><span class="n">bfile</span> <span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">:</span><span class="n">n</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">extract</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span> \
    <span class="o">--</span><span class="n">make</span><span class="o">-</span><span class="n">bed</span> \
    <span class="o">--</span><span class="n">out</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">1</span><span class="p">]:</span><span class="n">n</span><span class="p">}</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Run pca analysis using Eigenstrat: the program suports plink files here called PACKEDPED format</span>
<span class="c1"># smartpca.perl: run PCA on input genotype data (calls smartpca)</span>
<span class="p">[</span><span class="n">smartpca</span><span class="p">]</span>
<span class="c1"># Number of Principal Components to output</span>
<span class="kn">parameter:</span> <span class="n">k</span> <span class="o">=</span> <span class="nb">int</span>
<span class="c1"># Maximum number of iterations for outlier removal. Default 0 turns off outlier removal</span>
<span class="kn">parameter:</span> <span class="n">maxiter</span> <span class="o">=</span> <span class="mi">0</span>
<span class="c1"># Number of principal components along which to remove outliers during each outlier removal iteration. Default is 10</span>
<span class="kn">parameter:</span> <span class="n">topk</span> <span class="o">=</span> <span class="mi">10</span>
<span class="c1"># Number of standard deviations which an individual must exceed, along one of topk top principal components, in order to be removed as an outlier. Default is 6</span>
<span class="kn">parameter:</span> <span class="n">sigma</span> <span class="o">=</span> <span class="mi">6</span>
<span class="kn">input:</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{famFile:bn}.filtered.merged.bed&#39;</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{_input:bn}.parfile&#39;</span> <span class="p">,</span>  <span class="n">f</span><span class="s1">&#39;{cwd}/{_input:bn}.pca&#39;</span>
<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;24h&#39;</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;80G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="n">numThreads</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{step_name}_{_output:bn}&#39;</span>
<span class="kn">bash:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]}.stdout&#39;</span><span class="p">,</span>  <span class="n">template</span> <span class="o">=</span> <span class="s1">&#39;{cmd}&#39;</span> <span class="k">if</span> <span class="n">executable</span><span class="p">(</span><span class="s1">&#39;smartpca.perl&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">target_exists</span><span class="p">()</span> <span class="k">else</span> <span class="n">eigensoft_module</span>
  <span class="c1">#Create the parfile</span>
  <span class="nb">set</span> <span class="o">-</span><span class="n">e</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;genotypename: ${_input}&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;snpname: ${_input:n}.bim&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;indivname: ${_input:n}.eigenstrat.fam&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;outputformat: EIGENSTRAT&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;fastmode: YES&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;genotypeoutname: ${_input:n}.geno&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;snpoutname: ${_input:n}.snp&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;indivoutname: ${_input:n}.ind&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;evectoutname: ${_input:n}.evectout&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;evaloutname: ${_input:n}.evalout&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span> 
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;numoutevec: ${k}&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span> 
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;numoutlieriter: ${maxiter}&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span> 
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;numoutlierevec: ${topk}&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span> 
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;outliersigmathresh: ${sigma}&quot;</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span> 
  

<span class="kn">bash:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]}.stdout&#39;</span><span class="p">,</span>  <span class="n">template</span> <span class="o">=</span> <span class="s1">&#39;{cmd}&#39;</span> <span class="k">if</span> <span class="n">executable</span><span class="p">(</span><span class="s1">&#39;smartpca.perl&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">target_exists</span><span class="p">()</span> <span class="k">else</span> <span class="n">eigensoft_module</span>
   <span class="n">convertf</span> <span class="o">-</span><span class="n">p</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
   <span class="c1">#smartpca -p ${_output[0]}</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Run PCA analysis using flashpca</span>
<span class="p">[</span><span class="n">flashpca</span><span class="p">]</span>
<span class="c1"># Number of Principal Components to output. Default is 10</span>
<span class="kn">parameter:</span> <span class="n">k</span> <span class="o">=</span> <span class="nb">int</span>
<span class="c1"># Name of the trait in the phenoFile (format FID, IID, father, mother,trait )</span>
<span class="kn">parameter:</span> <span class="n">trait_name</span> <span class="o">=</span> <span class="nb">str</span>
<span class="c1"># How to standardize X before PCA</span>
<span class="kn">parameter:</span> <span class="n">stand</span> <span class="o">=</span> <span class="s2">&quot;binom2&quot;</span>
<span class="kn">depends:</span> <span class="n">phenoFile</span>
<span class="kn">input:</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{famFile:bn}.filtered.merged.prune.bed&#39;</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{_input[0]:bn}.pca&#39;</span><span class="p">,</span>
        <span class="n">f</span><span class="s1">&#39;{cwd}/{_input[0]:bn}.pc1vpc2.png&#39;</span><span class="p">,</span>
        <span class="n">f</span><span class="s1">&#39;{cwd}/{_input[0]:bn}.pc3vpc4.png&#39;</span><span class="p">,</span>
        <span class="n">f</span><span class="s1">&#39;{cwd}/{_input[0]:bn}.pc5vpc6.png&#39;</span> 
<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;10h&#39;</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;30G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="n">numThreads</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{step_name}_{_output[0]:bn}&#39;</span>
<span class="kn">R:</span> <span class="n">expand</span><span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]:n}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]:n}.stdout&#39;</span>
    <span class="c1"># Load required libraries</span>
    <span class="n">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
    <span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
    <span class="n">library</span><span class="p">(</span><span class="n">flashpcaR</span><span class="p">)</span>
    <span class="c1"># Read the PLINK binary files</span>
    <span class="n">fn</span> <span class="o">&lt;-</span> <span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">:</span><span class="n">nr</span><span class="p">}</span>
    <span class="c1"># Do the PCA computation</span>
    <span class="n">f</span> <span class="o">&lt;-</span> <span class="n">flashpca</span><span class="p">(</span><span class="n">fn</span><span class="p">,</span> <span class="n">ndim</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">k</span><span class="p">},</span> <span class="n">stand</span><span class="o">=</span><span class="s2">&quot;${stand}&quot;</span><span class="p">,</span> <span class="n">do_loadings</span><span class="o">=</span><span class="n">TRUE</span><span class="p">,</span> <span class="n">check_geno</span><span class="o">=</span><span class="n">TRUE</span><span class="p">)</span>
    <span class="c1"># Save the generated matrices to files</span>
    <span class="n">write</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="n">f</span><span class="err">$</span><span class="n">values</span><span class="p">,</span><span class="s1">&#39;${_output[0]:n}.values&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot; &quot;</span><span class="p">,</span> <span class="n">row</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">col</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span> 
    <span class="n">write</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="n">f</span><span class="err">$</span><span class="n">vectors</span><span class="p">,</span><span class="s1">&#39;${_output[0]:n}.vectors&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot; &quot;</span><span class="p">,</span> <span class="n">row</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">TRUE</span><span class="p">,</span> <span class="n">col</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span>
    <span class="n">write</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="n">f</span><span class="err">$</span><span class="n">projection</span><span class="p">,</span><span class="s1">&#39;${_output[0]:n}.projection&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot; &quot;</span><span class="p">,</span> <span class="n">row</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">TRUE</span><span class="p">,</span> <span class="n">col</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span>
    <span class="n">write</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="n">f</span><span class="err">$</span><span class="n">loadings</span><span class="p">,</span><span class="s1">&#39;${_output[0]:n}.loadings&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot; &quot;</span><span class="p">,</span> <span class="n">row</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">col</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span>
    <span class="n">write</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="n">f</span><span class="err">$</span><span class="n">scale</span><span class="p">,</span><span class="s1">&#39;${_output[0]:n}.scale&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot; &quot;</span><span class="p">,</span> <span class="n">row</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">col</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span>
    <span class="c1"># Use the projection file to generate plot</span>
    <span class="n">pca</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="s1">&#39;${_output[0]:n}.projection&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot; &quot;</span><span class="p">)</span>
    <span class="n">colnames</span><span class="p">(</span><span class="n">pca</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;ID&quot;</span><span class="p">,</span><span class="s2">&quot;PC1&quot;</span><span class="p">,</span><span class="s2">&quot;PC2&quot;</span><span class="p">,</span><span class="s2">&quot;PC3&quot;</span><span class="p">,</span><span class="s2">&quot;PC4&quot;</span><span class="p">,</span><span class="s2">&quot;PC5&quot;</span><span class="p">,</span><span class="s2">&quot;PC6&quot;</span><span class="p">,</span><span class="s2">&quot;PC7&quot;</span><span class="p">,</span><span class="s2">&quot;PC8&quot;</span><span class="p">,</span><span class="s2">&quot;PC9&quot;</span><span class="p">,</span><span class="s2">&quot;PC10&quot;</span><span class="p">)</span>
    <span class="n">pca</span><span class="err">$</span><span class="n">IID</span> <span class="o">&lt;-</span> <span class="n">sapply</span><span class="p">(</span><span class="n">strsplit</span><span class="p">(</span><span class="k">as</span><span class="o">.</span><span class="n">character</span><span class="p">(</span><span class="n">pca</span><span class="err">$</span><span class="n">ID</span><span class="p">),</span><span class="s1">&#39;:&#39;</span><span class="p">),</span> <span class="s2">&quot;[&quot;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span>
    <span class="c1"># Read fam file with phenotypes</span>
    <span class="n">pheno</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">phenoFile</span><span class="p">:</span><span class="n">r</span><span class="p">},</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span> <span class="p">)</span>
    <span class="n">colnames</span><span class="p">(</span><span class="n">pheno</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;FID&quot;</span><span class="p">,</span> <span class="s2">&quot;IID&quot;</span><span class="p">,</span> <span class="s2">&quot;father&quot;</span><span class="p">,</span> <span class="s2">&quot;mother&quot;</span><span class="p">,</span><span class="s2">&quot;sex&quot;</span><span class="p">,</span> <span class="s2">&quot;pheno&quot;</span><span class="p">,</span> <span class="s2">&quot;${trait_name}&quot;</span><span class="p">)</span>
    <span class="n">pca_final</span> <span class="o">&lt;-</span><span class="n">merge</span><span class="p">(</span><span class="n">pheno</span><span class="p">,</span> <span class="n">pca</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span><span class="s2">&quot;IID&quot;</span><span class="p">,</span> <span class="nb">all</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span>

    <span class="n">write</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="n">pca_final</span><span class="p">,</span><span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">row</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">col</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">TRUE</span><span class="p">)</span>
  
    <span class="n">png</span><span class="p">(</span><span class="s1">&#39;${_output[1]}&#39;</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="mi">6</span><span class="p">,</span> <span class="n">height</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span> <span class="n">unit</span><span class="o">=</span><span class="s1">&#39;in&#39;</span><span class="p">,</span> <span class="n">res</span><span class="o">=</span><span class="mi">300</span><span class="p">)</span>
    <span class="n">ggplot</span><span class="p">(</span><span class="n">pca_final</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">PC1</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">PC2</span><span class="p">))</span> <span class="o">+</span> <span class="n">geom_point</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">trait_name</span><span class="p">},</span> <span class="n">shape</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">trait_name</span><span class="p">}),</span> <span class="n">size</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span> <span class="o">+</span> <span class="n">labs</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;PC1 vs PC2 exomed subset ${ethnia_prefix}&quot;</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="s2">&quot;PC1&quot;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s2">&quot;PC2&quot;</span><span class="p">)</span> <span class="o">+</span> <span class="n">theme_classic</span><span class="p">()</span>
    <span class="n">dev</span><span class="o">.</span><span class="n">off</span><span class="p">()</span>
  
    <span class="n">png</span><span class="p">(</span><span class="s1">&#39;${_output[2]}&#39;</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="mi">6</span><span class="p">,</span> <span class="n">height</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span> <span class="n">unit</span><span class="o">=</span><span class="s1">&#39;in&#39;</span><span class="p">,</span> <span class="n">res</span><span class="o">=</span><span class="mi">300</span><span class="p">)</span>
    <span class="n">ggplot</span><span class="p">(</span><span class="n">pca_final</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">PC3</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">PC4</span><span class="p">))</span> <span class="o">+</span> <span class="n">geom_point</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">trait_name</span><span class="p">},</span> <span class="n">shape</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">trait_name</span><span class="p">}),</span> <span class="n">size</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span> <span class="o">+</span> <span class="n">labs</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;PC3 vs PC4 exomed subset ${ethnia_prefix}&quot;</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s2">&quot;PC3&quot;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s2">&quot;PC4&quot;</span><span class="p">)</span> <span class="o">+</span> <span class="n">theme_classic</span><span class="p">()</span>
    <span class="n">dev</span><span class="o">.</span><span class="n">off</span><span class="p">()</span>
    
    <span class="n">png</span><span class="p">(</span><span class="s1">&#39;${_output[3]}&#39;</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="mi">6</span><span class="p">,</span> <span class="n">height</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span> <span class="n">unit</span><span class="o">=</span><span class="s1">&#39;in&#39;</span><span class="p">,</span> <span class="n">res</span><span class="o">=</span><span class="mi">300</span><span class="p">)</span>
    <span class="n">ggplot</span><span class="p">(</span><span class="n">pca_final</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">PC5</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">PC6</span><span class="p">))</span> <span class="o">+</span> <span class="n">geom_point</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">trait_name</span><span class="p">},</span> <span class="n">shape</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">trait_name</span><span class="p">}),</span> <span class="n">size</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span> <span class="o">+</span> <span class="n">labs</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;P5 vs PC6 exomed subset ${ethnia_prefix}&quot;</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="s2">&quot;PC5&quot;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s2">&quot;PC6&quot;</span><span class="p">)</span> <span class="o">+</span> <span class="n">theme_classic</span><span class="p">()</span>
    <span class="n">dev</span><span class="o">.</span><span class="n">off</span><span class="p">()</span>
</pre></div>

    </div>
</div>
</div>

</div>
<hr>
&copy 2020 Diana Cornejo at Center for Statistical Genetics, Sergievsky Center, Columbia University VP&S
<p><small>Exported from <a href="http://github.com/dianacornejo/pleiotropy_UKB/blob/804a49e20903d8a6a7eafcb7472453c5ea2dcde8/workflow/PCA.ipynb"><code>workflow/PCA.ipynb</code></a> committed by dc2325 on Fri Feb 19 22:22:36 2021 <a href="http://github.com/dianacornejo/pleiotropy_UKB/commit/804a49e20903d8a6a7eafcb7472453c5ea2dcde8">revision 3, 804a49e</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
