<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=workflowDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=workflowArray;
            var docs_map=workflowArrayMap;
            var pos=workflowArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Pleiotropy analysis in UKBB</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Pleiotropy analysis in UKBB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../workflow.html">Workflow</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/dianacornejo/pleiotropy_UKB"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Sample-and-variant-missigness-statistics">Sample and variant missigness statistics<a class="anchor-link" href="#Sample-and-variant-missigness-statistics">&#182;</a></h1><p>Understand patterns of missing data PLINK exome files UKBB</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Aim">Aim<a class="anchor-link" href="#Aim">&#182;</a></h2><p>To generate missingness statistics for exome data of the UKBB and create a pptx/html report in order to decide the filtering parameters that will be used in other downstream analyses.</p>
<p>FIXME: the report created using marp needs to be organized to fit in the graphs within one slide and to allow better visualization of the tables</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Running-the-pipeline">Running the pipeline<a class="anchor-link" href="#Running-the-pipeline">&#182;</a></h2><p>In the bash terminal, set the variables and then run:</p>

<pre><code>sos run ~/project/UKBB_GWAS_dev/plink_missing.ipynb \
    --cwd $cwd \
    --bedfiles $bedfiles \
    --bimfiles $bimfiles \
    --famFile $famFile \
    --numThreads $numThreads \
    --job_size $job_size \
    --container_lmm $container_lmm \
    --container_marp $container_marp</code></pre>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="On-Yale's-cluster">On Yale's cluster<a class="anchor-link" href="#On-Yale's-cluster">&#182;</a></h1>
<pre><code>UKBB_PATH=/gpfs/gibbs/pi/dewan/data/UKBiobank
tpl_file=../farnam.yml
pca_dir=$UKBB_PATH/results/pca_exomes
famFile=$UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_s200631.fam
bedfiles=`echo $UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/ukb23155_c{1..22}_b0_v1.bed`
bimfiles=`echo $UKBB_PATH/genotype_files/ukb28374_exomedata/exome_data_OCT2020/UKBexomeOQFE_chr{1..22}.bim`
# Container
container_lmm=$UKBB_PATH/lmm.sif
container_marp=$UKBB_PATH/marp.sif
# Pipeline
plink_sos=~/project/UKBB_GWAS_dev/workflow/plink_missing.ipynb
# Name of bash script
plink_sbatch=../output/$(date +"%Y-%m-%d")_plink_miss.sbatch
numThreads=1
job_size=1

plink_args="""missing \
    --cwd $pca_dir \
    --bedfiles $bedfiles \
    --bimfiles $bimfiles \
    --famFile $famFile \
    --numThreads $numThreads \
    --job_size $job_size \
    --container_lmm $container_lmm \
    --container_marp $container_marp \
    -s build
"""

sos run ~/project/bioworkflows/GWAS/Get_Job_Script.ipynb farnam \
    --template-file $tpl_file \
    --workflow-file $plink_sos \
    --to-script $plink_sbatch \
    --args "$plink_args"</code></pre>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Command-interface">Command interface<a class="anchor-link" href="#Command-interface">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">sos</span> <span class="n">run</span> <span class="n">plink_missing</span><span class="o">.</span><span class="n">ipynb</span> <span class="o">-</span><span class="n">h</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="p">[</span><span class="k">global</span><span class="p">]</span>
<span class="c1"># the output directory for generated files</span>
<span class="kn">parameter:</span> <span class="n">cwd</span> <span class="o">=</span> <span class="n">path</span>
<span class="c1"># BED Plink files for exome data</span>
<span class="kn">parameter:</span> <span class="n">genoFile</span> <span class="o">=</span> <span class="n">paths</span>
<span class="c1"># For cluster jobs, number commands to run per job</span>
<span class="kn">parameter:</span> <span class="n">job_size</span> <span class="o">=</span> <span class="mi">1</span>
<span class="c1"># Number of threads</span>
<span class="kn">parameter:</span> <span class="n">numThreads</span> <span class="o">=</span> <span class="mi">1</span>
<span class="c1"># Software container option</span>
<span class="kn">parameter:</span> <span class="n">container_lmm</span> <span class="o">=</span> <span class="s1">&#39;statisticalgenetics/lmm:1.8&#39;</span>
<span class="kn">parameter:</span> <span class="n">container_marp</span> <span class="o">=</span> <span class="s1">&#39;gaow/marp&#39;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Genotype and sample missingness for exome files</span>
<span class="p">[</span><span class="n">missing_1</span><span class="p">]</span>
<span class="kn">input:</span> <span class="n">genoFile</span><span class="p">,</span> <span class="n">group_by</span><span class="o">=</span><span class="mi">1</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{_input:bn}.imiss&#39;</span><span class="p">,</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{_input:bn}.lmiss&#39;</span><span class="p">,</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{_input:bn}.frq&#39;</span><span class="p">,</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{_input:bn}.log&#39;</span>
<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;10h&#39;</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;30G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="n">numThreads</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{step_name}_{_output:bn}&#39;</span>
<span class="kn">bash:</span> <span class="n">container</span><span class="o">=</span><span class="n">container_lmm</span><span class="p">,</span> <span class="n">expand</span><span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]:n}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]:n}.stdout&#39;</span>
    <span class="n">plink</span> \
      <span class="o">--</span><span class="n">bfile</span> <span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">:</span><span class="n">n</span><span class="p">}</span> \
      <span class="o">--</span><span class="n">missing</span> \
      <span class="o">--</span><span class="n">freq</span> \
      <span class="o">--</span><span class="n">threads</span> <span class="err">$</span><span class="p">{</span><span class="n">numThreads</span><span class="p">}</span> \
      <span class="o">--</span><span class="n">out</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">n</span><span class="p">}</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Generate plots to understand patterns of missing data</span>
<span class="p">[</span><span class="n">missing_2</span><span class="p">]</span>
<span class="c1"># Filter out variants with missing call rate higher that this value</span>
<span class="kn">parameter:</span> <span class="n">geno</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="c1"># Filter out samples with missing rate higher than this value</span>
<span class="kn">parameter:</span> <span class="n">mind</span> <span class="o">=</span> <span class="mf">0.02</span>
<span class="c1"># Filter based on minor allele frequency</span>
<span class="kn">parameter:</span> <span class="n">maf</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="n">sep</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="se">\n\n</span><span class="s1">---</span><span class="se">\n</span><span class="s1">&#39;</span>
<span class="kn">input:</span> <span class="n">group_by</span><span class="o">=</span><span class="mi">4</span>
<span class="kn">output:</span> <span class="n">analysis_summary</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{_input[0]:bn}.missing.md&#39;</span><span class="p">,</span> 
        <span class="n">sample_plot</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{_input[0]:bn}.imiss.png&#39;</span><span class="p">,</span>  
        <span class="n">variant_plot</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{_input[0]:bn}.lmiss.png&#39;</span><span class="p">,</span>  
        <span class="n">maf_plot</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{_input[0]:bn}.maf.png&#39;</span>
<span class="kn">task:</span> <span class="kp">trunk_workers</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="kp">walltime</span> <span class="o">=</span> <span class="s1">&#39;10h&#39;</span><span class="p">,</span> <span class="kp">mem</span> <span class="o">=</span> <span class="s1">&#39;30G&#39;</span><span class="p">,</span> <span class="kp">cores</span> <span class="o">=</span> <span class="n">numThreads</span><span class="p">,</span> <span class="kp">tags</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{step_name}_{_output[0]:bn}&#39;</span>   
<span class="kn">R:</span> <span class="n">expand</span><span class="o">=</span> <span class="s2">&quot;${ }&quot;</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]:n}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]:n}.stdout&#39;</span>
    <span class="c1">#load libraries</span>
    <span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
    <span class="n">library</span><span class="p">(</span><span class="n">scales</span><span class="p">)</span>
    <span class="c1">#read data sample missingness</span>
    <span class="n">smiss</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">header</span><span class="o">=</span><span class="n">T</span><span class="p">)</span>
    <span class="c1">#colnames(smiss) &lt;- c(&quot;FID&quot;,&quot;IID&quot;,&quot;MISS_PHENO&quot;, &quot;N_MISS&quot;, &quot;N_GENO&quot;,&quot;F_MISS&quot;)</span>
    <span class="n">out</span> <span class="o">=</span> <span class="n">capture</span><span class="o">.</span><span class="n">output</span><span class="p">(</span><span class="k">as</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">summary</span><span class="p">(</span><span class="n">smiss</span><span class="err">$</span><span class="n">F_MISS</span><span class="p">)))</span>
    <span class="n">smiss_filter</span> <span class="o">=</span> <span class="n">length</span><span class="p">(</span><span class="n">which</span><span class="p">(</span><span class="n">smiss</span><span class="err">$</span><span class="n">F_MISS</span><span class="o">&gt;</span><span class="err">$</span><span class="p">{</span><span class="n">mind</span><span class="p">}))</span>
    <span class="c1">#read data variant missingness</span>
    <span class="n">vmiss</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">1</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">header</span><span class="o">=</span><span class="n">T</span><span class="p">)</span>
    <span class="c1">#colnames(vmiss) &lt;- c(&quot;CHR&quot;,&quot;SNP&quot;,&quot;N_MISS&quot;, &quot;N_GENO&quot;, &quot;F_MISS&quot;)</span>
    <span class="n">out_1</span> <span class="o">=</span> <span class="n">capture</span><span class="o">.</span><span class="n">output</span><span class="p">(</span><span class="k">as</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">summary</span><span class="p">(</span><span class="n">vmiss</span><span class="err">$</span><span class="n">F_MISS</span><span class="p">)))</span>
    <span class="n">vmiss_filter</span> <span class="o">=</span> <span class="n">length</span><span class="p">(</span><span class="n">which</span><span class="p">(</span><span class="n">vmiss</span><span class="err">$</span><span class="n">F_MISS</span><span class="o">&gt;</span><span class="err">$</span><span class="p">{</span><span class="n">geno</span><span class="p">}))</span>
    <span class="c1">#look at the allele frequency</span>
    <span class="n">afreq</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">2</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">header</span><span class="o">=</span><span class="n">T</span><span class="p">)</span>
    <span class="c1">#colnames(afreq) &lt;- c(&quot;CHR&quot;,&quot;SNP&quot;,&quot;A1&quot;, &quot;A2&quot;, &quot;MAF&quot;, &quot;NCHROBS&quot;)</span>
    <span class="n">out_2</span> <span class="o">=</span> <span class="n">capture</span><span class="o">.</span><span class="n">output</span><span class="p">(</span><span class="k">as</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">summary</span><span class="p">(</span><span class="n">afreq</span><span class="err">$</span><span class="n">MAF</span><span class="p">)))</span>
    <span class="n">afreq_filter</span> <span class="o">=</span> <span class="n">length</span><span class="p">(</span><span class="n">which</span><span class="p">(</span><span class="n">afreq</span><span class="err">$</span><span class="n">MAF</span><span class="o">&gt;</span><span class="err">$</span><span class="p">{</span><span class="n">maf</span><span class="p">}))</span>
    <span class="c1"># create plots</span>
    <span class="n">png</span><span class="p">(</span><span class="s1">&#39;${_output[1]}&#39;</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span> <span class="n">height</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span> <span class="n">unit</span><span class="o">=</span><span class="s1">&#39;in&#39;</span><span class="p">,</span> <span class="n">res</span><span class="o">=</span><span class="mi">300</span><span class="p">)</span>
    <span class="n">smiss_plot</span> <span class="o">&lt;-</span> <span class="n">ggplot</span><span class="p">(</span><span class="n">smiss</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">F_MISS</span><span class="p">))</span> <span class="o">+</span> <span class="n">geom_histogram</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">y</span> <span class="o">=</span> <span class="n">stat</span><span class="p">(</span><span class="n">count</span><span class="p">)</span> <span class="o">/</span> <span class="nb">sum</span><span class="p">(</span><span class="n">count</span><span class="p">)),</span><span class="n">binwidth</span><span class="o">=</span><span class="mf">0.001</span><span class="p">)</span>  <span class="o">+</span> <span class="n">labs</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;Histogram of sample missing call rate</span><span class="se">\n</span><span class="s2"> for ${_input[0]:bn}&quot;</span><span class="p">,</span> <span class="n">x</span> <span class="o">=</span><span class="s2">&quot;Proportion of missing variants&quot;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s2">&quot;Frequency&quot;</span><span class="p">)</span> <span class="o">+</span> <span class="n">scale_y_continuous</span><span class="p">(</span><span class="n">breaks</span><span class="o">=</span><span class="n">seq</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">by</span><span class="o">=</span><span class="mf">0.2</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="n">scales</span><span class="p">::</span><span class="n">percent</span><span class="p">)</span> <span class="o">+</span> <span class="n">theme</span><span class="p">(</span><span class="n">plot</span><span class="o">.</span><span class="n">title</span> <span class="o">=</span> <span class="n">element_text</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="mi">10</span><span class="p">))</span> <span class="o">+</span> <span class="n">theme_classic</span><span class="p">()</span>
    <span class="k">print</span><span class="p">(</span><span class="n">smiss_plot</span><span class="p">)</span>
    <span class="n">dev</span><span class="o">.</span><span class="n">off</span><span class="p">()</span>
    
    <span class="n">png</span><span class="p">(</span><span class="s1">&#39;${_output[2]}&#39;</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span> <span class="n">height</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span> <span class="n">unit</span><span class="o">=</span><span class="s1">&#39;in&#39;</span><span class="p">,</span> <span class="n">res</span><span class="o">=</span><span class="mi">300</span><span class="p">)</span>
    <span class="n">vmiss_plot</span> <span class="o">&lt;-</span> <span class="n">ggplot</span><span class="p">(</span><span class="n">vmiss</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">F_MISS</span><span class="p">))</span> <span class="o">+</span> <span class="n">geom_histogram</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">y</span> <span class="o">=</span> <span class="n">stat</span><span class="p">(</span><span class="n">count</span><span class="p">)</span> <span class="o">/</span> <span class="nb">sum</span><span class="p">(</span><span class="n">count</span><span class="p">)),</span> <span class="n">bins</span><span class="o">=</span><span class="mi">100</span><span class="p">)</span> <span class="o">+</span> <span class="n">labs</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;Plot of variant missingness rate</span><span class="se">\n</span><span class="s2"> for ${_input[0]:bn}&quot;</span><span class="p">,</span> <span class="n">x</span> <span class="o">=</span><span class="s2">&quot;SNP missing call rate&quot;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s2">&quot;Frequency&quot;</span><span class="p">)</span> <span class="o">+</span> <span class="n">scale_y_continuous</span><span class="p">(</span><span class="n">breaks</span><span class="o">=</span><span class="n">seq</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">by</span><span class="o">=</span><span class="mf">0.2</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="n">scales</span><span class="p">::</span><span class="n">percent</span><span class="p">)</span> <span class="o">+</span> <span class="n">theme</span><span class="p">(</span><span class="n">plot</span><span class="o">.</span><span class="n">title</span> <span class="o">=</span> <span class="n">element_text</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="mi">10</span><span class="p">))</span> <span class="o">+</span> <span class="n">theme_classic</span><span class="p">()</span>
    <span class="k">print</span><span class="p">(</span><span class="n">vmiss_plot</span><span class="p">)</span>
    <span class="n">dev</span><span class="o">.</span><span class="n">off</span><span class="p">()</span>
  
    <span class="n">png</span><span class="p">(</span><span class="s1">&#39;${_output[3]}&#39;</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span> <span class="n">height</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span> <span class="n">unit</span><span class="o">=</span><span class="s1">&#39;in&#39;</span><span class="p">,</span> <span class="n">res</span><span class="o">=</span><span class="mi">300</span><span class="p">)</span>
    <span class="n">freq_plot</span> <span class="o">&lt;-</span> <span class="n">ggplot</span><span class="p">(</span><span class="n">afreq</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">MAF</span><span class="p">))</span> <span class="o">+</span> <span class="n">geom_histogram</span><span class="p">(</span><span class="n">bins</span><span class="o">=</span><span class="mi">100</span><span class="p">)</span> <span class="o">+</span> <span class="n">xlim</span><span class="p">(</span><span class="n">c</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mf">0.5</span><span class="p">))</span>  <span class="o">+</span> <span class="n">scale_y_log10</span><span class="p">(</span><span class="n">breaks</span> <span class="o">=</span> <span class="n">trans_breaks</span><span class="p">(</span><span class="s2">&quot;log10&quot;</span><span class="p">,</span> <span class="n">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="mi">10</span><span class="o">^</span><span class="n">x</span><span class="p">),</span><span class="n">labels</span> <span class="o">=</span> <span class="n">trans_format</span><span class="p">(</span><span class="s2">&quot;log10&quot;</span><span class="p">,</span> <span class="n">math_format</span><span class="p">(</span><span class="mi">10</span><span class="o">^.</span><span class="n">x</span><span class="p">)))</span> <span class="o">+</span> <span class="n">labs</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;Minor allele frequency distribution</span><span class="se">\n</span><span class="s2"> for ${_input[0]:bn}&quot;</span><span class="p">,</span> <span class="n">x</span> <span class="o">=</span><span class="s2">&quot;MAF&quot;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s2">&quot;Number of SNPs&quot;</span><span class="p">)</span> <span class="o">+</span> <span class="n">theme</span><span class="p">(</span><span class="n">plot</span><span class="o">.</span><span class="n">title</span> <span class="o">=</span> <span class="n">element_text</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="mi">10</span><span class="p">))</span> <span class="o">+</span> <span class="n">theme_classic</span><span class="p">()</span>
    <span class="k">print</span><span class="p">(</span><span class="n">freq_plot</span><span class="p">)</span>
    <span class="n">dev</span><span class="o">.</span><span class="n">off</span><span class="p">()</span>
  
    <span class="n">write</span><span class="p">(</span><span class="s1">&#39;# ${_input[0]:b} result summary</span><span class="se">\n</span><span class="s1">## Sample missingness summary:</span><span class="se">\n</span><span class="s1">```&#39;</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">append</span> <span class="o">=</span> <span class="n">T</span><span class="p">)</span>
    <span class="n">cat</span><span class="p">(</span><span class="s1">&#39;##The number of samples with a proportion of missing SNPs higher than ${mind} is:&#39;</span><span class="p">,</span><span class="n">smiss_filter</span><span class="p">,</span><span class="s2">&quot;.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">,</span> <span class="nb">file</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">append</span> <span class="o">=</span> <span class="n">T</span> <span class="p">)</span> 
    <span class="n">write</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="n">out</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">append</span> <span class="o">=</span> <span class="n">T</span><span class="p">)</span> 
    <span class="n">cat</span><span class="p">(</span><span class="s1">&#39;##The number of variants missing in more than ${geno} of the samples is:&#39;</span><span class="p">,</span><span class="n">vmiss_filter</span><span class="p">,</span><span class="s2">&quot;.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">,</span> <span class="nb">file</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">append</span> <span class="o">=</span> <span class="n">T</span> <span class="p">)</span>
    <span class="n">write</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="n">out_1</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">append</span> <span class="o">=</span> <span class="n">T</span><span class="p">)</span> 
    <span class="n">cat</span><span class="p">(</span><span class="s1">&#39;##The number of variants with maf&gt;${maf} is:&#39;</span><span class="p">,</span><span class="n">afreq_filter</span><span class="p">,</span><span class="s2">&quot;.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">,</span> <span class="nb">file</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">append</span> <span class="o">=</span> <span class="n">T</span> <span class="p">)</span>
    <span class="n">write</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="n">out_2</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">append</span> <span class="o">=</span> <span class="n">T</span><span class="p">)</span>
    <span class="n">write</span><span class="p">(</span><span class="s2">&quot;```${sep}&quot;</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">append</span> <span class="o">=</span> <span class="n">T</span><span class="p">)</span>
  
<span class="kn">bash:</span> <span class="n">expand</span><span class="o">=</span><span class="bp">True</span>
  <span class="nb">set</span> <span class="o">-</span><span class="n">e</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;### Number of analyzed variants for {_output[0]:bn}</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">&gt;&gt;</span> <span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">cat</span>  <span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">nn</span><span class="p">}</span><span class="o">.</span><span class="n">log</span> <span class="o">|</span> <span class="n">grep</span> <span class="o">-</span><span class="n">w</span> <span class="s2">&quot;variants&quot;</span> <span class="o">&amp;&amp;</span> <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;{sep}&quot;</span> <span class="o">&gt;&gt;</span> <span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;#</span><span class="se">\n</span><span class="s2">#</span><span class="se">\n</span><span class="s2">### Sample missingness rate plot for {_output[1]:bn}</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">&gt;&gt;</span> <span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;![]({_output[1]:b}){sep}&quot;</span> <span class="o">&gt;&gt;</span> <span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;#</span><span class="se">\n</span><span class="s2">#</span><span class="se">\n</span><span class="s2">### Variant missingess rate plot for {_output[2]:bn}</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">&gt;&gt;</span> <span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;![]({_output[2]:b}){sep}&quot;</span> <span class="o">&gt;&gt;</span> <span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;### Minor allele frequency plot for {_output[3]:bn}</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">&gt;&gt;</span> <span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;![]({_output[3]:b}){sep}&quot;</span> <span class="o">&gt;&gt;</span> <span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;### Result files</span><span class="se">\n</span><span class="s2">\`\`\`&quot;</span> <span class="o">&gt;&gt;</span> <span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">ls</span> <span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">nn</span><span class="p">}</span><span class="o">.*</span> <span class="o">|</span> <span class="n">grep</span> <span class="o">-</span><span class="n">vP</span> <span class="s1">&#39;stderr|stdout&#39;</span><span class="o">&gt;&gt;</span> <span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
  <span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;\`\`\`&quot;</span> <span class="o">&gt;&gt;</span> <span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[35]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Generate the merged markdown file</span>
<span class="p">[</span><span class="n">missing_3</span><span class="p">]</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s1">&#39;{cwd}/{str(_input[0]).rsplit(&#39;</span><span class="n">_</span><span class="s1">&#39;,3)[0]}.merged.md&#39;</span>
<span class="kn">bash:</span><span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;${ }&quot;</span>
    <span class="n">echo</span> <span class="s1">&#39;&#39;&#39;---</span>
<span class="s1">    theme: base-theme</span>
<span class="s1">    style: |</span>
<span class="s1">     img {</span>
<span class="s1">       #display: block;</span>
<span class="s1">       height: 60%;</span>
<span class="s1">       width: 60%;</span>
<span class="s1">       #margin-top: 22%;</span>
<span class="s1">    }</span>
<span class="s1">    ---    </span>
<span class="s1">    &#39;&#39;&#39;</span> <span class="o">&gt;</span>  <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">}</span>
    
  <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="err">$</span><span class="p">(</span><span class="n">ls</span> <span class="o">-</span><span class="n">rt</span> <span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">nn</span><span class="p">}</span><span class="o">.*</span> <span class="o">|</span> <span class="n">grep</span> <span class="s2">&quot;missing.md&quot;</span><span class="p">);</span><span class="n">do</span> <span class="n">cat</span> <span class="err">$</span><span class="n">i</span> <span class="o">&gt;&gt;</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">};</span><span class="n">done</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># Generate analysis report: HTML file, and optionally PPTX file</span>
<span class="p">[</span><span class="n">missing_4</span><span class="p">]</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s1">&#39;{_input:n}.html&#39;</span>
<span class="kn">sh:</span> <span class="n">container</span><span class="o">=</span><span class="n">container_marp</span><span class="p">,</span> <span class="n">expand</span> <span class="o">=</span> <span class="bp">True</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:n}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output:n}.stdout&#39;</span>
    <span class="n">node</span> <span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">marp</span><span class="o">/.</span><span class="n">cli</span><span class="o">/</span><span class="n">marp</span><span class="o">-</span><span class="n">cli</span><span class="o">.</span><span class="n">js</span> <span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">n</span><span class="p">}</span><span class="o">.</span><span class="n">md</span> <span class="o">-</span><span class="n">o</span> <span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">a</span><span class="p">}</span> \
        <span class="o">--</span><span class="n">title</span> <span class="s1">&#39;Sample and variant missingness UKBB {_output:bn}&#39;</span> \
        <span class="o">--</span><span class="n">allow</span><span class="o">-</span><span class="n">local</span><span class="o">-</span><span class="n">files</span>
    <span class="n">node</span> <span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">marp</span><span class="o">/.</span><span class="n">cli</span><span class="o">/</span><span class="n">marp</span><span class="o">-</span><span class="n">cli</span><span class="o">.</span><span class="n">js</span> <span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">n</span><span class="p">}</span><span class="o">.</span><span class="n">md</span> <span class="o">-</span><span class="n">o</span> <span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">an</span><span class="p">}</span><span class="o">.</span><span class="n">pptx</span> \
        <span class="o">--</span><span class="n">title</span> <span class="s1">&#39;Sample and variant missingness UKBB {_output:bn}&#39;</span> \
        <span class="o">--</span><span class="n">allow</span><span class="o">-</span><span class="n">local</span><span class="o">-</span><span class="n">files</span>
</pre></div>

    </div>
</div>
</div>

</div>
<hr>
&copy 2020 Diana Cornejo at Center for Statistical Genetics, Sergievsky Center, Columbia University VP&S
<p><small>Exported from <a href="http://github.com/dianacornejo/pleiotropy_UKB/blob/804a49e20903d8a6a7eafcb7472453c5ea2dcde8/workflow/plink_missing.ipynb"><code>workflow/plink_missing.ipynb</code></a> committed by dc2325 on Fri Feb 19 22:22:36 2021 <a href="http://github.com/dianacornejo/pleiotropy_UKB/commit/804a49e20903d8a6a7eafcb7472453c5ea2dcde8">revision 2, 804a49e</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
